<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="robots" content="noodp" />
        <meta http-equiv="X-UA-Compatible" content="IE=edge, chrome=1">
        <title>Nanopore demultiplexing, QC, mapping and downstream analysi pipeline - MAJOR Computational Biology Research Lab</title><meta name="Description" content="MAJOR Computational Biology Research Lab"><meta property="og:title" content="Nanopore demultiplexing, QC, mapping and downstream analysi pipeline" />
<meta property="og:description" content="Introduction nfcore/ontworkflow is a bioinformatics analysis pipeline for Nanopore DNA/RNA sequencing data that can be used to perform basecalling, demultiplexing, QC, mapping and downstream analysis.
The pipeline is built using Nextflow, a workflow tool to run tasks across multiple compute infrastructures in a very portable manner. It uses Docker/Singularity containers making installation trivial and results highly reproducible. The Nextflow DSL2 implementation of this pipeline uses one container per process which makes it much easier to maintain and update software dependencies." />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://cb-majorbiolab.work/posts/ont-workflow/" />
<meta property="og:image" content="http://cb-majorbiolab.work/posts/ont-workflow/ONT_pipe.png"/>
<meta property="article:published_time" content="2021-12-12T17:29:59+08:00" />
<meta property="article:modified_time" content="2021-12-12T16:29:59+08:00" />
<meta name="twitter:card" content="summary_large_image"/>
<meta name="twitter:image" content="http://cb-majorbiolab.work/posts/ont-workflow/ONT_pipe.png"/>
<meta name="twitter:title" content="Nanopore demultiplexing, QC, mapping and downstream analysi pipeline"/>
<meta name="twitter:description" content="Introduction nfcore/ontworkflow is a bioinformatics analysis pipeline for Nanopore DNA/RNA sequencing data that can be used to perform basecalling, demultiplexing, QC, mapping and downstream analysis.
The pipeline is built using Nextflow, a workflow tool to run tasks across multiple compute infrastructures in a very portable manner. It uses Docker/Singularity containers making installation trivial and results highly reproducible. The Nextflow DSL2 implementation of this pipeline uses one container per process which makes it much easier to maintain and update software dependencies."/>
<meta name="application-name" content="MAJOR Computational Biology Research Lab">
<meta name="apple-mobile-web-app-title" content="MAJOR Computational Biology Research Lab"><meta name="theme-color" content="#ffffff"><meta name="msapplication-TileColor" content="#da532c"><link rel="icon" href="/images/MBIL.svg"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5"><link rel="manifest" href="/site.webmanifest"><link rel="canonical" href="http://cb-majorbiolab.work/posts/ont-workflow/" /><link rel="prev" href="http://cb-majorbiolab.work/posts/circrna-pipeline/" /><link rel="stylesheet" href="/lib/normalize/normalize.min.css"><link rel="stylesheet" href="/css/style.min.css"><link rel="stylesheet" href="/lib/fontawesome-free/all.min.css"><link rel="stylesheet" href="/lib/animate/animate.min.css"><script type="application/ld+json">
    {
        "@context": "http://schema.org",
        "@type": "BlogPosting",
        "headline": "Nanopore demultiplexing, QC, mapping and downstream analysi pipeline",
        "inLanguage": "en",
        "mainEntityOfPage": {
            "@type": "WebPage",
            "@id": "http:\/\/cb-majorbiolab.work\/posts\/ont-workflow\/"
        },"image": [{
                            "@type": "ImageObject",
                            "url": "http:\/\/cb-majorbiolab.work\/posts\/ont-workflow\/ONT_pipe.png",
                            "width":  913 ,
                            "height":  947 
                        }],"genre": "posts","keywords": "DNA, workflow, RNA","wordcount":  889 ,
        "url": "http:\/\/cb-majorbiolab.work\/posts\/ont-workflow\/","datePublished": "2021-12-12T17:29:59+08:00","dateModified": "2021-12-12T16:29:59+08:00","publisher": {
            "@type": "Organization",
            "name": "美吉生物信息团队"},"author": {
                "@type": "Person",
                "name": "Meng Luo"
            },"description": ""
    }
    </script></head>
    <body header-desktop="" header-mobile=""><script type="text/javascript">(window.localStorage && localStorage.getItem('theme') ? localStorage.getItem('theme') === 'dark' : ('' === 'auto' ? window.matchMedia('(prefers-color-scheme: dark)').matches : '' === 'dark')) && document.body.setAttribute('theme', 'dark');</script>

        <div id="mask"></div><div class="wrapper"><header class="desktop" id="header-desktop">
    <div class="header-wrapper">
        <div class="header-title">
            <a href="/" title="MAJOR Computational Biology Research Lab"><img
        class="lazyload logo"
        src="/svg/loading.min.svg"
        data-src="/images/MBIL.svg"
        data-srcset="/images/MBIL.svg, /images/MBIL.svg 1.5x, /images/MBIL.svg 2x"
        data-sizes="auto"
        alt="/images/MBIL.svg"
        title="/images/MBIL.svg" /></a>
        </div>
        <div class="menu">
            <div class="menu-inner"><a class="menu-item" href="/posts/" title="Home"> Home </a><a class="menu-item" href="/workshop/" title=" "> Workshop/Talk </a><a class="menu-item" href="/publication/" title="Publications"> Publications </a><a class="menu-item" href="/team/" title="Team"> Team </a><a class="menu-item" href="/resource/" title="Resource"> Resource </a><a class="menu-item" href="/software/" title="Software"> Software </a><a class="menu-item" href="/ne/" title="News/Events"> News/Events </a><a class="menu-item" href="/joinus/" title="Join/Contact"> Join/Contact </a><a class="menu-item" href="https://github.com/MajorBio" title="GitHub" rel="noopener noreffer" target="_blank"><i class='fab fa-github fa-fw'></i>  </a><span class="menu-item delimiter"></span><span class="menu-item search" id="search-desktop">
                        <input type="text" placeholder="搜索文章标题或内容..." id="search-input-desktop">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-desktop" title="搜索">
                            <i class="fas fa-search fa-fw"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-desktop" title="清空">
                            <i class="fas fa-times-circle fa-fw"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-desktop">
                            <i class="fas fa-spinner fa-fw fa-spin"></i>
                        </span>
                    </span><a href="javascript:void(0);" class="menu-item theme-switch" title="切换主题">
                    <i class="fas fa-adjust fa-fw"></i>
                </a>
            </div>
        </div>
    </div>
</header><header class="mobile" id="header-mobile">
    <div class="header-container">
        <div class="header-wrapper">
            <div class="header-title">
                <a href="/" title="MAJOR Computational Biology Research Lab"><img
        class="lazyload logo"
        src="/svg/loading.min.svg"
        data-src="/images/MBIL.svg"
        data-srcset="/images/MBIL.svg, /images/MBIL.svg 1.5x, /images/MBIL.svg 2x"
        data-sizes="auto"
        alt="/images/MBIL.svg"
        title="/images/MBIL.svg" /></a>
            </div>
            <div class="menu-toggle" id="menu-toggle-mobile">
                <span></span><span></span><span></span>
            </div>
        </div>
        <div class="menu" id="menu-mobile"><div class="search-wrapper">
                    <div class="search mobile" id="search-mobile">
                        <input type="text" placeholder="搜索文章标题或内容..." id="search-input-mobile">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-mobile" title="搜索">
                            <i class="fas fa-search fa-fw"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-mobile" title="清空">
                            <i class="fas fa-times-circle fa-fw"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-mobile">
                            <i class="fas fa-spinner fa-fw fa-spin"></i>
                        </span>
                    </div>
                    <a href="javascript:void(0);" class="search-cancel" id="search-cancel-mobile">
                        取消
                    </a>
                </div><a class="menu-item" href="/posts/" title="Home">Home</a><a class="menu-item" href="/workshop/" title=" ">Workshop/Talk</a><a class="menu-item" href="/publication/" title="Publications">Publications</a><a class="menu-item" href="/team/" title="Team">Team</a><a class="menu-item" href="/resource/" title="Resource">Resource</a><a class="menu-item" href="/software/" title="Software">Software</a><a class="menu-item" href="/ne/" title="News/Events">News/Events</a><a class="menu-item" href="/joinus/" title="Join/Contact">Join/Contact</a><a class="menu-item" href="https://github.com/MajorBio" title="GitHub" rel="noopener noreffer" target="_blank"><i class='fab fa-github fa-fw'></i></a><a href="javascript:void(0);" class="menu-item theme-switch" title="切换主题">
                <i class="fas fa-adjust fa-fw"></i>
            </a></div>
    </div>
</header>
<div class="search-dropdown desktop">
    <div id="search-dropdown-desktop"></div>
</div>
<div class="search-dropdown mobile">
    <div id="search-dropdown-mobile"></div>
</div><main class="main">
                <div class="container"><div class="toc" id="toc-auto">
            <h2 class="toc-title">目录</h2>
            <div class="toc-content" id="toc-content-auto"></div>
        </div><article class="page single"><h1 class="single-title animated flipInX">Nanopore demultiplexing, QC, mapping and downstream analysi pipeline</h1><div class="post-meta">
            <div class="post-meta-line"><span class="post-author"><a href="https://github.com/czheluo" title="Author" target="_blank" rel="noopener noreffer author" class="author"><i class="fas fa-user-circle fa-fw"></i>Meng Luo</a></span>&nbsp;<span class="post-category">收录于 <a href="/categories/nanopore/"><i class="far fa-folder fa-fw"></i>Nanopore</a>&nbsp;<a href="/categories/pipeline/"><i class="far fa-folder fa-fw"></i>Pipeline</a>&nbsp;<a href="/categories/workflow/"><i class="far fa-folder fa-fw"></i>workflow</a></span></div>
            <div class="post-meta-line"><i class="far fa-calendar-alt fa-fw"></i>&nbsp;<time datetime="2021-12-12">2021-12-12</time>&nbsp;<i class="fas fa-pencil-alt fa-fw"></i>&nbsp;约 889 字&nbsp;
                <i class="far fa-clock fa-fw"></i>&nbsp;预计阅读 5 分钟&nbsp;<span id="/posts/ont-workflow/" class="leancloud_visitors" data-flag-title="Nanopore demultiplexing, QC, mapping and downstream analysi pipeline">
                        <i class="far fa-eye fa-fw"></i>&nbsp;<span class=leancloud-visitors-count></span>&nbsp;次阅读
                    </span>&nbsp;</div>
        </div><div class="featured-image"><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/posts/ont-workflow/ONT_pipe.png"
        data-srcset="/posts/ont-workflow/ONT_pipe.png, /posts/ont-workflow/ONT_pipe.png 1.5x, /posts/ont-workflow/ONT_pipe.png 2x"
        data-sizes="auto"
        alt="/posts/ont-workflow/ONT_pipe.png"
        title="/posts/ont-workflow/ONT_pipe.png" /></div><div class="details toc" id="toc-static"  kept="">
                <div class="details-summary toc-title">
                    <span>目录</span>
                    <span><i class="details-icon fas fa-angle-right"></i></span>
                </div>
                <div class="details-content toc-content" id="toc-content-static"><nav id="TableOfContents">
  <ul>
    <li><a href="#introduction">Introduction</a></li>
    <li><a href="#pipeline-summary">Pipeline summary</a></li>
    <li><a href="#pipeline-summary-1">Pipeline Summary</a></li>
    <li><a href="#quick-start">Quick Start</a></li>
    <li><a href="#documentation">Documentation</a></li>
    <li><a href="#credits">Credits</a></li>
    <li><a href="#contributions-and-support">Contributions and Support</a></li>
    <li><a href="#citations">Citations</a></li>
  </ul>
</nav></div>
            </div><div class="content" id="content"><h2 id="introduction">Introduction</h2>
<p><strong>nfcore/ontworkflow</strong> is a bioinformatics analysis pipeline for Nanopore DNA/RNA sequencing data that can be used to perform basecalling, demultiplexing, QC, mapping and downstream analysis.</p>
<p>The pipeline is built using <a href="https://www.nextflow.io" target="_blank" rel="noopener noreffer">Nextflow</a>, a workflow tool to run tasks across multiple compute infrastructures in a very portable manner. It uses Docker/Singularity containers making installation trivial and results highly reproducible. The <a href="https://www.nextflow.io/docs/latest/dsl2.html" target="_blank" rel="noopener noreffer">Nextflow DSL2</a> implementation of this pipeline uses one container per process which makes it much easier to maintain and update software dependencies. Where possible, these processes have been submitted to and installed from <a href="https://github.com/nf-core/modules" target="_blank" rel="noopener noreffer">nf-core/modules</a> in order to make them available to all nf-core pipelines, and to everyone within the Nextflow community!</p>
<p>On release, automated continuous integration tests run the pipeline on a full-sized dataset on the AWS cloud infrastructure. This ensures that the pipeline runs on AWS, has sensible resource allocation defaults set to run on real-world datasets, and permits the persistent storage of results to benchmark between pipeline releases and other analysis sources. The results obtained from the full-sized test can be viewed on the <a href="https://nf-co.re/nanoseq/results" target="_blank" rel="noopener noreffer">nf-core website</a>.</p>
<h2 id="pipeline-summary">Pipeline summary</h2>
<ol>
<li>Read QC (<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/" target="_blank" rel="noopener noreffer"><code>FastQC</code></a>)</li>
<li>Present QC for raw reads (<a href="http://multiqc.info/" target="_blank" rel="noopener noreffer"><code>MultiQC</code></a>)</li>
</ol>
<p>On release, automated continuous integration tests run the pipeline on a <a href="https://github.com/nf-core/test-datasets/tree/nanoseq#full-sized-test-data" target="_blank" rel="noopener noreffer">full-sized dataset</a> obtained from the <a href="https://github.com/GoekeLab/sg-nex-data" target="_blank" rel="noopener noreffer">Singapore Nanopore Expression Consortium</a> on the AWS cloud infrastructure. This ensures that the pipeline runs on AWS, has sensible resource allocation defaults set to run on real-world datasets, and permits the persistent storage of results to benchmark between pipeline releases and other analysis sources. The results obtained from the full-sized test can be viewed on the <a href="https://nf-co.re/nanoseq/results" target="_blank" rel="noopener noreffer">nf-core website</a>.</p>
<h2 id="pipeline-summary-1">Pipeline Summary</h2>
<ol>
<li>Basecalling and/or demultiplexing (<a href="https://nanoporetech.com/nanopore-sequencing-data-analysis" target="_blank" rel="noopener noreffer"><code>Guppy</code></a> or <a href="https://github.com/nanoporetech/qcat" target="_blank" rel="noopener noreffer"><code>qcat</code></a>; <em>optional</em>)</li>
<li>Sequencing QC (<a href="https://github.com/a-slide/pycoQC" target="_blank" rel="noopener noreffer"><code>pycoQC</code></a>, <a href="https://github.com/wdecoster/NanoPlot" target="_blank" rel="noopener noreffer"><code>NanoPlot</code></a>)</li>
<li>Raw read DNA cleaning (<a href="https://github.com/wdecoster/nanolyse" target="_blank" rel="noopener noreffer">NanoLyse</a>; <em>optional</em>)</li>
<li>Raw read QC (<a href="https://github.com/wdecoster/NanoPlot" target="_blank" rel="noopener noreffer"><code>NanoPlot</code></a>, <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/" target="_blank" rel="noopener noreffer"><code>FastQC</code></a>)</li>
<li>Alignment (<a href="https://github.com/lbcb-sci/graphmap2" target="_blank" rel="noopener noreffer"><code>GraphMap2</code></a> or <a href="https://github.com/lh3/minimap2" target="_blank" rel="noopener noreffer"><code>minimap2</code></a>)
<ul>
<li>Both aligners are capable of performing unspliced and spliced alignment. Sensible defaults will be applied automatically based on a combination of the input data and user-specified parameters</li>
<li>Each sample can be mapped to its own reference genome if multiplexed in this way</li>
<li>Convert SAM to co-ordinate sorted BAM and obtain mapping metrics (<a href="http://www.htslib.org/doc/samtools.html" target="_blank" rel="noopener noreffer"><code>SAMtools</code></a>)</li>
</ul>
</li>
<li>Create bigWig (<a href="https://github.com/arq5x/bedtools2/" target="_blank" rel="noopener noreffer"><code>BEDTools</code></a>, <a href="http://hgdownload.soe.ucsc.edu/admin/exe/" target="_blank" rel="noopener noreffer"><code>bedGraphToBigWig</code></a>) and bigBed (<a href="https://github.com/arq5x/bedtools2/" target="_blank" rel="noopener noreffer"><code>BEDTools</code></a>, <a href="http://hgdownload.soe.ucsc.edu/admin/exe/" target="_blank" rel="noopener noreffer"><code>bedToBigBed</code></a>) coverage tracks for visualisation</li>
<li>RNA-specific downstream analysis:
<ul>
<li>Transcript reconstruction and quantification (<a href="https://bioconductor.org/packages/release/bioc/html/bambu.html" target="_blank" rel="noopener noreffer"><code>bambu</code></a> or <a href="https://ccb.jhu.edu/software/stringtie/" target="_blank" rel="noopener noreffer"><code>StringTie2</code></a>)
<ul>
<li>bambu performs both transcript reconstruction and quantification.</li>
<li>When StringTie2 is chosen, each sample can be processed individually and combined. After which, <a href="http://bioinf.wehi.edu.au/featureCounts/" target="_blank" rel="noopener noreffer"><code>featureCounts</code></a> will be used for both gene and transcript quantification.</li>
</ul>
</li>
<li>Differential expression analysis (<a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html" target="_blank" rel="noopener noreffer"><code>DESeq2</code></a> or <a href="https://bioconductor.org/packages/release/bioc/html/DEXSeq.html" target="_blank" rel="noopener noreffer"><code>DEXSeq</code></a>)</li>
</ul>
</li>
<li>Present QC for raw read and alignment results (<a href="https://multiqc.info/docs/" target="_blank" rel="noopener noreffer"><code>MultiQC</code></a>)</li>
</ol>
<h2 id="quick-start">Quick Start</h2>
<ol>
<li>
<p>Install <a href="https://www.nextflow.io/docs/latest/getstarted.html#installation" target="_blank" rel="noopener noreffer"><code>Nextflow</code></a> (<code>&gt;=21.04.0</code>)</p>
</li>
<li>
<p>Install any of <a href="https://docs.docker.com/engine/installation/" target="_blank" rel="noopener noreffer"><code>Docker</code></a>, <a href="https://www.sylabs.io/guides/3.0/user-guide/" target="_blank" rel="noopener noreffer"><code>Singularity</code></a>, <a href="https://podman.io/" target="_blank" rel="noopener noreffer"><code>Podman</code></a>, <a href="https://nersc.gitlab.io/development/shifter/how-to-use/" target="_blank" rel="noopener noreffer"><code>Shifter</code></a> or <a href="https://hpc.github.io/charliecloud/" target="_blank" rel="noopener noreffer"><code>Charliecloud</code></a> for full pipeline reproducibility <em>(please only use <a href="https://conda.io/miniconda.html" target="_blank" rel="noopener noreffer"><code>Conda</code></a> as a last resort; see <a href="https://nf-co.re/usage/configuration#basic-configuration-profiles" target="_blank" rel="noopener noreffer">docs</a>)</em></p>
</li>
<li>
<p>Download the pipeline and test it on a minimal dataset with a single command:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">nextflow run nf-core/nanoseq -profile test,&lt;docker/singularity/podman/shifter/charliecloud/conda/institute&gt;
</code></pre></td></tr></table>
</div>
</div><blockquote>
<ul>
<li>Please check <a href="https://github.com/nf-core/configs#documentation" target="_blank" rel="noopener noreffer">nf-core/configs</a> to see if a custom config file to run nf-core pipelines already exists for your Institute. If so, you can simply use <code>-profile &lt;institute&gt;</code> in your command. This will enable either <code>docker</code> or <code>singularity</code> and set the appropriate execution settings for your local compute environment.</li>
<li>If you are using <code>singularity</code> then the pipeline will auto-detect this and attempt to download the Singularity images directly as opposed to performing a conversion from Docker images. If you are persistently observing issues downloading Singularity images directly due to timeout or network issues then please use the <code>--singularity_pull_docker_container</code> parameter to pull and convert the Docker image instead. Alternatively, it is highly recommended to use the <a href="https://nf-co.re/tools/#downloading-pipelines-for-offline-use" target="_blank" rel="noopener noreffer"><code>nf-core download</code></a> command to pre-download all of the required containers before running the pipeline and to set the <a href="https://www.nextflow.io/docs/latest/singularity.html?#singularity-docker-hub" target="_blank" rel="noopener noreffer"><code>NXF_SINGULARITY_CACHEDIR</code> or <code>singularity.cacheDir</code></a> Nextflow options to be able to store and re-use the images from a central location for future pipeline runs.</li>
<li>If you are using <code>conda</code>, it is highly recommended to use the <a href="https://www.nextflow.io/docs/latest/conda.html" target="_blank" rel="noopener noreffer"><code>NXF_CONDA_CACHEDIR</code> or <code>conda.cacheDir</code></a> settings to store the environments in a central location for future pipeline runs.</li>
</ul>
</blockquote>
</li>
<li>
<p>Start running your own analysis!</p>
</li>
</ol>
<h2 id="documentation">Documentation</h2>
<p>The nf-core/nanoseq pipeline comes with documentation about the pipeline <a href="https://nf-co.re/nanoseq/usage" target="_blank" rel="noopener noreffer">usage</a>, <a href="https://nf-co.re/nanoseq/parameters" target="_blank" rel="noopener noreffer">parameters</a> and <a href="https://nf-co.re/nanoseq/output" target="_blank" rel="noopener noreffer">output</a>.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-bash" data-lang="bash">nextflow run nf-core/nanoseq <span class="se">\
</span><span class="se"></span>    --input samplesheet.csv <span class="se">\
</span><span class="se"></span>    --protocol DNA <span class="se">\
</span><span class="se"></span>    --input_path ./fast5/ <span class="se">\
</span><span class="se"></span>    --flowcell FLO-MIN106 <span class="se">\
</span><span class="se"></span>    --kit SQK-LSK109 <span class="se">\
</span><span class="se"></span>    --barcode_kit SQK-PBK004 <span class="se">\
</span><span class="se"></span>    -profile &lt;docker/singularity/podman/institute&gt;
</code></pre></td></tr></table>
</div>
</div><p>See <a href="https://nf-co.re/nanoseq/usage" target="_blank" rel="noopener noreffer">usage docs</a> for all of the available options when running the pipeline.</p>
<p>An example input samplesheet for performing both basecalling and demultiplexing can be found <a href="assets/samplesheet.csv" rel="">here</a>.</p>
<h2 id="credits">Credits</h2>
<p>nf-core/nanoseq was originally written by <a href="https://github.com/csawye01" target="_blank" rel="noopener noreffer">Chelsea Sawyer</a> and <a href="https://github.com/drpatelh" target="_blank" rel="noopener noreffer">Harshil Patel</a> from <a href="https://www.crick.ac.uk/research/science-technology-platforms/bioinformatics-and-biostatistics/" target="_blank" rel="noopener noreffer">The Bioinformatics &amp; Biostatistics Group</a> for use at <a href="https://www.crick.ac.uk/" target="_blank" rel="noopener noreffer">The Francis Crick Institute</a>, London. Other primary contributors include <a href="https://github.com/lwratten" target="_blank" rel="noopener noreffer">Laura Wratten</a>, <a href="https://github.com/cying111" target="_blank" rel="noopener noreffer">Ying Chen</a>, <a href="https://github.com/yuukiiwa" target="_blank" rel="noopener noreffer">Yuk Kei Wan</a> and <a href="https://github.com/jonathangoeke" target="_blank" rel="noopener noreffer">Jonathan Goeke</a> from the <a href="https://www.a-star.edu.sg/gis" target="_blank" rel="noopener noreffer">Genome Institute of Singapore</a>, <a href="https://github.com/alneberg" target="_blank" rel="noopener noreffer">Johannes Alneberg</a> and <a href="https://github.com/FranBonath" target="_blank" rel="noopener noreffer">Franziska Bonath</a> from <a href="https://www.scilifelab.se/" target="_blank" rel="noopener noreffer">SciLifeLab</a>, Sweden.</p>
<p>Many thanks to others who have helped out along the way too, including (but not limited to): <a href="https://github.com/crickbabs" target="_blank" rel="noopener noreffer">@crickbabs</a>, <a href="https://github.com/AnnaSyme" target="_blank" rel="noopener noreffer">@AnnaSyme</a>.</p>
<h2 id="contributions-and-support">Contributions and Support</h2>
<p>If you would like to contribute to this pipeline, please see the <a href=".github/CONTRIBUTING.md" rel="">contributing guidelines</a>.</p>
<p>For further information or help, don&rsquo;t hesitate to get in touch on <a href="https://nfcore.slack.com/channels/nanoseq" target="_blank" rel="noopener noreffer">Slack</a> (you can join with <a href="https://nf-co.re/join/slack" target="_blank" rel="noopener noreffer">this invite</a>).</p>
<h2 id="citations">Citations</h2>
<p>An extensive list of references for the tools used by the pipeline can be found in the <a href="CITATIONS.md" rel=""><code>CITATIONS.md</code></a> file.</p>
<p>You can cite the <code>nf-core</code> publication as follows:</p>
<blockquote>
<p><strong>The nf-core framework for community-curated bioinformatics pipelines.</strong></p>
<p>Philip Ewels, Alexander Peltzer, Sven Fillinger, Harshil Patel, Johannes Alneberg, Andreas Wilm, Maxime Ulysse Garcia, Paolo Di Tommaso &amp; Sven Nahnsen.</p>
<p><em>Nat Biotechnol.</em> 2020 Feb 13. doi: <a href="https://dx.doi.org/10.1038/s41587-020-0439-x" target="_blank" rel="noopener noreffer">10.1038/s41587-020-0439-x</a>.</p>
</blockquote>
</div><div class="post-footer" id="post-footer">
    <div class="post-info">
        <div class="post-info-line">
            <div class="post-info-mod">
                <span>更新于 2021-12-12</span>
            </div>
            <div class="post-info-license"></div>
        </div>
        <div class="post-info-line">
            <div class="post-info-md"><span>
                            <a class="link-to-markdown" href="/posts/ont-workflow/index.md" target="_blank">阅读原始文档</a>
                        </span></div>
            <div class="post-info-share">
                <span><a href="javascript:void(0);" title="分享到 Twitter" data-sharer="twitter" data-url="http://cb-majorbiolab.work/posts/ont-workflow/" data-title="Nanopore demultiplexing, QC, mapping and downstream analysi pipeline" data-via="https://twitter.com/czheluo" data-hashtags="DNA,workflow,RNA"><i class="fab fa-twitter fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Facebook" data-sharer="facebook" data-url="http://cb-majorbiolab.work/posts/ont-workflow/" data-hashtag="DNA"><i class="fab fa-facebook-square fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Linkedin" data-sharer="linkedin" data-url="http://cb-majorbiolab.work/posts/ont-workflow/"><i class="fab fa-linkedin fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Hacker News" data-sharer="hackernews" data-url="http://cb-majorbiolab.work/posts/ont-workflow/" data-title="Nanopore demultiplexing, QC, mapping and downstream analysi pipeline"><i class="fab fa-hacker-news fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Line" data-sharer="line" data-url="http://cb-majorbiolab.work/posts/ont-workflow/" data-title="Nanopore demultiplexing, QC, mapping and downstream analysi pipeline"><i data-svg-src="/lib/simple-icons/icons/line.min.svg"></i></a><a href="javascript:void(0);" title="分享到 微博" data-sharer="weibo" data-url="http://cb-majorbiolab.work/posts/ont-workflow/" data-title="Nanopore demultiplexing, QC, mapping and downstream analysi pipeline" data-ralateuid="https://weibo.com/2161302013/profile"><i class="fab fa-weibo fa-fw"></i></a></span>
            </div>
        </div>
    </div>

    <div class="post-info-more">
        <section class="post-tags"><i class="fas fa-tags fa-fw"></i>&nbsp;<a href="/tags/dna/">DNA</a>,&nbsp;<a href="/tags/workflow/">workflow</a>,&nbsp;<a href="/tags/rna/">RNA</a></section>
        <section>
            <span><a href="javascript:void(0);" onclick="window.history.back();">返回</a></span>&nbsp;|&nbsp;<span><a href="/">主页</a></span>
        </section>
    </div>

    <div class="post-nav"><a href="/posts/circrna-pipeline/" class="prev" rel="prev" title="circRNA quantification, differential expression analysis and miRNA target prediction of RNA-Seq data"><i class="fas fa-angle-left fa-fw"></i>circRNA quantification, differential expression analysis and miRNA target prediction of RNA-Seq data</a></div>
</div>
<div id="comments"><div id="valine" class="comment"></div><noscript>
                Please enable JavaScript to view the comments powered by <a href="https://valine.js.org/">Valine</a>.
            </noscript><div id="fb-root" class="comment"></div>
            <div
                class="fb-comments"
                data-href="http://cb-majorbiolab.work/posts/ont-workflow/"
                data-width="100%"
                data-numposts="10"
            ></div><noscript>
                Please enable JavaScript to view the comments powered by <a href="https://developers.facebook.com/docs/plugins/comments/"></a>Facebook</a>.
            </noscript></div></article></div>
            </main><footer class="footer">
        <div class="footer-container"><div class="footer-line">MAJOR Computational Biology Research Lab</div>
                <div class="footer-line"><i class="far fa-copyright fa-fw"></i><span itemprop="copyrightYear">2020-2022, MENG LUO, All rights reserved.</span><span class="author" itemprop="copyrightHolder">&nbsp;<a href="/" target="_blank"></a></span></div>
                
        </div>
    </footer></div>

        <div id="fixed-buttons"><a href="#" id="back-to-top" class="fixed-button" title="回到顶部">
                <i class="fas fa-arrow-up fa-fw"></i>
            </a><a href="#" id="view-comments" class="fixed-button" title="查看评论">
                <i class="fas fa-comment fa-fw"></i>
            </a>
        </div><link rel="stylesheet" href="/lib/valine/valine.min.css"><link rel="stylesheet" href="/lib/lightgallery/lightgallery.min.css"><link rel="stylesheet" href="/lib/katex/katex.min.css"><link rel="stylesheet" href="/lib/katex/copy-tex.min.css"><script type="text/javascript" src="/lib/valine/Valine.min.js"></script><script type="text/javascript" src="https://connect.facebook.net/zh_CN/sdk.js#xfbml=1&amp;version=v5.0&amp;appId=&amp;autoLogAppEvents=1" defer></script><script type="text/javascript" src="/lib/smooth-scroll/smooth-scroll.min.js"></script><script type="text/javascript" src="/lib/autocomplete/autocomplete.min.js"></script><script type="text/javascript" src="/lib/algoliasearch/algoliasearch-lite.umd.min.js"></script><script type="text/javascript" src="/lib/lazysizes/lazysizes.min.js"></script><script type="text/javascript" src="/lib/lightgallery/lightgallery.min.js"></script><script type="text/javascript" src="/lib/lightgallery/lg-thumbnail.min.js"></script><script type="text/javascript" src="/lib/lightgallery/lg-zoom.min.js"></script><script type="text/javascript" src="/lib/clipboard/clipboard.min.js"></script><script type="text/javascript" src="/lib/sharer/sharer.min.js"></script><script type="text/javascript" src="/lib/katex/katex.min.js"></script><script type="text/javascript" src="/lib/katex/auto-render.min.js"></script><script type="text/javascript" src="/lib/katex/copy-tex.min.js"></script><script type="text/javascript" src="/lib/katex/mhchem.min.js"></script><script type="text/javascript">window.config={"code":{"copyTitle":"复制到剪贴板","maxShownLines":10},"comment":{"valine":{"appId":"Noyy9UKaXGwYBIUdvBTt7u76-gzGzoHsz","appKey":"15w6pTF8mFoxnm8muOFx5buM","avatar":"mp","el":"#valine","emojiCDN":"https://cdn.jsdelivr.net/npm/emoji-datasource-google@5.0.1/img/google/64/","emojiMaps":{"100":"1f4af.png","alien":"1f47d.png","anger":"1f4a2.png","angry":"1f620.png","anguished":"1f627.png","astonished":"1f632.png","black_heart":"1f5a4.png","blue_heart":"1f499.png","blush":"1f60a.png","bomb":"1f4a3.png","boom":"1f4a5.png","broken_heart":"1f494.png","brown_heart":"1f90e.png","clown_face":"1f921.png","cold_face":"1f976.png","cold_sweat":"1f630.png","confounded":"1f616.png","confused":"1f615.png","cry":"1f622.png","crying_cat_face":"1f63f.png","cupid":"1f498.png","dash":"1f4a8.png","disappointed":"1f61e.png","disappointed_relieved":"1f625.png","dizzy":"1f4ab.png","dizzy_face":"1f635.png","drooling_face":"1f924.png","exploding_head":"1f92f.png","expressionless":"1f611.png","face_vomiting":"1f92e.png","face_with_cowboy_hat":"1f920.png","face_with_hand_over_mouth":"1f92d.png","face_with_head_bandage":"1f915.png","face_with_monocle":"1f9d0.png","face_with_raised_eyebrow":"1f928.png","face_with_rolling_eyes":"1f644.png","face_with_symbols_on_mouth":"1f92c.png","face_with_thermometer":"1f912.png","fearful":"1f628.png","flushed":"1f633.png","frowning":"1f626.png","ghost":"1f47b.png","gift_heart":"1f49d.png","green_heart":"1f49a.png","grimacing":"1f62c.png","grin":"1f601.png","grinning":"1f600.png","hankey":"1f4a9.png","hear_no_evil":"1f649.png","heart":"2764-fe0f.png","heart_decoration":"1f49f.png","heart_eyes":"1f60d.png","heart_eyes_cat":"1f63b.png","heartbeat":"1f493.png","heartpulse":"1f497.png","heavy_heart_exclamation_mark_ornament":"2763-fe0f.png","hole":"1f573-fe0f.png","hot_face":"1f975.png","hugging_face":"1f917.png","hushed":"1f62f.png","imp":"1f47f.png","innocent":"1f607.png","japanese_goblin":"1f47a.png","japanese_ogre":"1f479.png","joy":"1f602.png","joy_cat":"1f639.png","kiss":"1f48b.png","kissing":"1f617.png","kissing_cat":"1f63d.png","kissing_closed_eyes":"1f61a.png","kissing_heart":"1f618.png","kissing_smiling_eyes":"1f619.png","laughing":"1f606.png","left_speech_bubble":"1f5e8-fe0f.png","love_letter":"1f48c.png","lying_face":"1f925.png","mask":"1f637.png","money_mouth_face":"1f911.png","nauseated_face":"1f922.png","nerd_face":"1f913.png","neutral_face":"1f610.png","no_mouth":"1f636.png","open_mouth":"1f62e.png","orange_heart":"1f9e1.png","partying_face":"1f973.png","pensive":"1f614.png","persevere":"1f623.png","pleading_face":"1f97a.png","pouting_cat":"1f63e.png","purple_heart":"1f49c.png","rage":"1f621.png","relaxed":"263a-fe0f.png","relieved":"1f60c.png","revolving_hearts":"1f49e.png","right_anger_bubble":"1f5ef-fe0f.png","robot_face":"1f916.png","rolling_on_the_floor_laughing":"1f923.png","scream":"1f631.png","scream_cat":"1f640.png","see_no_evil":"1f648.png","shushing_face":"1f92b.png","skull":"1f480.png","skull_and_crossbones":"2620-fe0f.png","sleeping":"1f634.png","sleepy":"1f62a.png","slightly_frowning_face":"1f641.png","slightly_smiling_face":"1f642.png","smile":"1f604.png","smile_cat":"1f638.png","smiley":"1f603.png","smiley_cat":"1f63a.png","smiling_face_with_3_hearts":"1f970.png","smiling_imp":"1f608.png","smirk":"1f60f.png","smirk_cat":"1f63c.png","sneezing_face":"1f927.png","sob":"1f62d.png","space_invader":"1f47e.png","sparkling_heart":"1f496.png","speak_no_evil":"1f64a.png","speech_balloon":"1f4ac.png","star-struck":"1f929.png","stuck_out_tongue":"1f61b.png","stuck_out_tongue_closed_eyes":"1f61d.png","stuck_out_tongue_winking_eye":"1f61c.png","sunglasses":"1f60e.png","sweat":"1f613.png","sweat_drops":"1f4a6.png","sweat_smile":"1f605.png","thinking_face":"1f914.png","thought_balloon":"1f4ad.png","tired_face":"1f62b.png","triumph":"1f624.png","two_hearts":"1f495.png","unamused":"1f612.png","upside_down_face":"1f643.png","weary":"1f629.png","white_frowning_face":"2639-fe0f.png","white_heart":"1f90d.png","wink":"1f609.png","woozy_face":"1f974.png","worried":"1f61f.png","yawning_face":"1f971.png","yellow_heart":"1f49b.png","yum":"1f60b.png","zany_face":"1f92a.png","zipper_mouth_face":"1f910.png","zzz":"1f4a4.png"},"enableQQ":false,"highlight":true,"lang":"zh-cn","pageSize":10,"placeholder":"你的评论 ...","recordIP":true,"visitor":true}},"lightGallery":{"actualSize":false,"exThumbImage":"data-thumbnail","hideBarsDelay":2000,"selector":".lightgallery","speed":400,"thumbContHeight":80,"thumbWidth":80,"thumbnail":true},"math":{"delimiters":[{"display":true,"left":"$$","right":"$$"},{"display":true,"left":"\\[","right":"\\]"},{"display":false,"left":"$","right":"$"},{"display":false,"left":"\\(","right":"\\)"}],"strict":false},"search":{"algoliaAppID":"VX4BXOJBC5","algoliaIndex":"majorbio.zh-cn","algoliaSearchKey":"53790f167edb548b217bfe91dd891ac0","highlightTag":"em","maxResultLength":10,"noResultsFound":"没有找到结果","snippetLength":50,"type":"algolia"}};</script><script type="text/javascript" src="/js/theme.min.js"></script></body>
</html>
