<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="robots" content="noodp" />
        <meta http-equiv="X-UA-Compatible" content="IE=edge, chrome=1">
        <title>circRNA quantification, differential expression analysis and miRNA target prediction of RNA-Seq data - MAJOR Computational Biology Research Lab</title><meta name="Description" content="MAJOR Computational Biology Research Lab"><meta property="og:title" content="circRNA quantification, differential expression analysis and miRNA target prediction of RNA-Seq data" />
<meta property="og:description" content="nf-core/circrnaworkflow: Usage It is recommended that first time users run nf-core/circrnaworkflow with the minimal test dataset either locally or on a HPC, referring to the output documentation before running a full analysis.
1  nextflow run nf-core/circrnaworkflow -profile test   Run the test dataset on a HPC:
1  nextflow run nf-core/circrnaworkflow -profile test,&lt;docker/singularity/podman/institute&gt;   Running the pipeline A typical command for running the pipeline is as follows:" />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://cb-majorbiolab.work/posts/circrna-pipeline/" />
<meta property="og:image" content="http://cb-majorbiolab.work/posts/circrna-pipeline/circRNA.png"/>
<meta property="article:published_time" content="2021-12-06T17:29:59+08:00" />
<meta property="article:modified_time" content="2021-12-06T16:29:59+08:00" />
<meta name="twitter:card" content="summary_large_image"/>
<meta name="twitter:image" content="http://cb-majorbiolab.work/posts/circrna-pipeline/circRNA.png"/>
<meta name="twitter:title" content="circRNA quantification, differential expression analysis and miRNA target prediction of RNA-Seq data"/>
<meta name="twitter:description" content="nf-core/circrnaworkflow: Usage It is recommended that first time users run nf-core/circrnaworkflow with the minimal test dataset either locally or on a HPC, referring to the output documentation before running a full analysis.
1  nextflow run nf-core/circrnaworkflow -profile test   Run the test dataset on a HPC:
1  nextflow run nf-core/circrnaworkflow -profile test,&lt;docker/singularity/podman/institute&gt;   Running the pipeline A typical command for running the pipeline is as follows:"/>
<meta name="application-name" content="MAJOR Computational Biology Research Lab">
<meta name="apple-mobile-web-app-title" content="MAJOR Computational Biology Research Lab"><meta name="theme-color" content="#ffffff"><meta name="msapplication-TileColor" content="#da532c"><link rel="icon" href="/images/MBIL.svg"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5"><link rel="manifest" href="/site.webmanifest"><link rel="canonical" href="http://cb-majorbiolab.work/posts/circrna-pipeline/" /><link rel="prev" href="http://cb-majorbiolab.work/posts/visually-combining-expression-data-with-functional-analysis/" /><link rel="next" href="http://cb-majorbiolab.work/posts/ont-workflow/" /><link rel="stylesheet" href="/lib/normalize/normalize.min.css"><link rel="stylesheet" href="/css/style.min.css"><link rel="stylesheet" href="/lib/fontawesome-free/all.min.css"><link rel="stylesheet" href="/lib/animate/animate.min.css"><script type="application/ld+json">
    {
        "@context": "http://schema.org",
        "@type": "BlogPosting",
        "headline": "circRNA quantification, differential expression analysis and miRNA target prediction of RNA-Seq data",
        "inLanguage": "en",
        "mainEntityOfPage": {
            "@type": "WebPage",
            "@id": "http:\/\/cb-majorbiolab.work\/posts\/circrna-pipeline\/"
        },"image": [{
                            "@type": "ImageObject",
                            "url": "http:\/\/cb-majorbiolab.work\/posts\/circrna-pipeline\/circRNA.png",
                            "width":  1235 ,
                            "height":  1080 
                        }],"genre": "posts","keywords": "DEGs, workflow, cricRNA","wordcount":  2340 ,
        "url": "http:\/\/cb-majorbiolab.work\/posts\/circrna-pipeline\/","datePublished": "2021-12-06T17:29:59+08:00","dateModified": "2021-12-06T16:29:59+08:00","publisher": {
            "@type": "Organization",
            "name": "美吉生物信息团队"},"author": {
                "@type": "Person",
                "name": "Meng Luo"
            },"description": ""
    }
    </script></head>
    <body header-desktop="" header-mobile=""><script type="text/javascript">(window.localStorage && localStorage.getItem('theme') ? localStorage.getItem('theme') === 'dark' : ('' === 'auto' ? window.matchMedia('(prefers-color-scheme: dark)').matches : '' === 'dark')) && document.body.setAttribute('theme', 'dark');</script>

        <div id="mask"></div><div class="wrapper"><header class="desktop" id="header-desktop">
    <div class="header-wrapper">
        <div class="header-title">
            <a href="/" title="MAJOR Computational Biology Research Lab"><img
        class="lazyload logo"
        src="/svg/loading.min.svg"
        data-src="/images/MBIL.svg"
        data-srcset="/images/MBIL.svg, /images/MBIL.svg 1.5x, /images/MBIL.svg 2x"
        data-sizes="auto"
        alt="/images/MBIL.svg"
        title="/images/MBIL.svg" /></a>
        </div>
        <div class="menu">
            <div class="menu-inner"><a class="menu-item" href="/posts/" title="Home"> Home </a><a class="menu-item" href="/workshop/" title=" "> Workshop/Talk </a><a class="menu-item" href="/publication/" title="Publications"> Publications </a><a class="menu-item" href="/team/" title="Team"> Team </a><a class="menu-item" href="/resource/" title="Resource"> Resource </a><a class="menu-item" href="/software/" title="Software"> Software </a><a class="menu-item" href="/ne/" title="News/Events"> News/Events </a><a class="menu-item" href="/joinus/" title="Join/Contact"> Join/Contact </a><a class="menu-item" href="https://github.com/MajorBio" title="GitHub" rel="noopener noreffer" target="_blank"><i class='fab fa-github fa-fw'></i>  </a><span class="menu-item delimiter"></span><span class="menu-item search" id="search-desktop">
                        <input type="text" placeholder="搜索文章标题或内容..." id="search-input-desktop">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-desktop" title="搜索">
                            <i class="fas fa-search fa-fw"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-desktop" title="清空">
                            <i class="fas fa-times-circle fa-fw"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-desktop">
                            <i class="fas fa-spinner fa-fw fa-spin"></i>
                        </span>
                    </span><a href="javascript:void(0);" class="menu-item theme-switch" title="切换主题">
                    <i class="fas fa-adjust fa-fw"></i>
                </a>
            </div>
        </div>
    </div>
</header><header class="mobile" id="header-mobile">
    <div class="header-container">
        <div class="header-wrapper">
            <div class="header-title">
                <a href="/" title="MAJOR Computational Biology Research Lab"><img
        class="lazyload logo"
        src="/svg/loading.min.svg"
        data-src="/images/MBIL.svg"
        data-srcset="/images/MBIL.svg, /images/MBIL.svg 1.5x, /images/MBIL.svg 2x"
        data-sizes="auto"
        alt="/images/MBIL.svg"
        title="/images/MBIL.svg" /></a>
            </div>
            <div class="menu-toggle" id="menu-toggle-mobile">
                <span></span><span></span><span></span>
            </div>
        </div>
        <div class="menu" id="menu-mobile"><div class="search-wrapper">
                    <div class="search mobile" id="search-mobile">
                        <input type="text" placeholder="搜索文章标题或内容..." id="search-input-mobile">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-mobile" title="搜索">
                            <i class="fas fa-search fa-fw"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-mobile" title="清空">
                            <i class="fas fa-times-circle fa-fw"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-mobile">
                            <i class="fas fa-spinner fa-fw fa-spin"></i>
                        </span>
                    </div>
                    <a href="javascript:void(0);" class="search-cancel" id="search-cancel-mobile">
                        取消
                    </a>
                </div><a class="menu-item" href="/posts/" title="Home">Home</a><a class="menu-item" href="/workshop/" title=" ">Workshop/Talk</a><a class="menu-item" href="/publication/" title="Publications">Publications</a><a class="menu-item" href="/team/" title="Team">Team</a><a class="menu-item" href="/resource/" title="Resource">Resource</a><a class="menu-item" href="/software/" title="Software">Software</a><a class="menu-item" href="/ne/" title="News/Events">News/Events</a><a class="menu-item" href="/joinus/" title="Join/Contact">Join/Contact</a><a class="menu-item" href="https://github.com/MajorBio" title="GitHub" rel="noopener noreffer" target="_blank"><i class='fab fa-github fa-fw'></i></a><a href="javascript:void(0);" class="menu-item theme-switch" title="切换主题">
                <i class="fas fa-adjust fa-fw"></i>
            </a></div>
    </div>
</header>
<div class="search-dropdown desktop">
    <div id="search-dropdown-desktop"></div>
</div>
<div class="search-dropdown mobile">
    <div id="search-dropdown-mobile"></div>
</div><main class="main">
                <div class="container"><div class="toc" id="toc-auto">
            <h2 class="toc-title">目录</h2>
            <div class="toc-content" id="toc-content-auto"></div>
        </div><article class="page single"><h1 class="single-title animated flipInX">circRNA quantification, differential expression analysis and miRNA target prediction of RNA-Seq data</h1><div class="post-meta">
            <div class="post-meta-line"><span class="post-author"><a href="https://github.com/czheluo" title="Author" target="_blank" rel="noopener noreffer author" class="author"><i class="fas fa-user-circle fa-fw"></i>Meng Luo</a></span>&nbsp;<span class="post-category">收录于 <a href="/categories/circrnaseq/"><i class="far fa-folder fa-fw"></i>circrnaseq</a>&nbsp;<a href="/categories/pipeline/"><i class="far fa-folder fa-fw"></i>Pipeline</a>&nbsp;<a href="/categories/workflow/"><i class="far fa-folder fa-fw"></i>workflow</a></span></div>
            <div class="post-meta-line"><i class="far fa-calendar-alt fa-fw"></i>&nbsp;<time datetime="2021-12-06">2021-12-06</time>&nbsp;<i class="fas fa-pencil-alt fa-fw"></i>&nbsp;约 2340 字&nbsp;
                <i class="far fa-clock fa-fw"></i>&nbsp;预计阅读 11 分钟&nbsp;<span id="/posts/circrna-pipeline/" class="leancloud_visitors" data-flag-title="circRNA quantification, differential expression analysis and miRNA target prediction of RNA-Seq data">
                        <i class="far fa-eye fa-fw"></i>&nbsp;<span class=leancloud-visitors-count></span>&nbsp;次阅读
                    </span>&nbsp;</div>
        </div><div class="featured-image"><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/posts/circrna-pipeline/circRNA.png"
        data-srcset="/posts/circrna-pipeline/circRNA.png, /posts/circrna-pipeline/circRNA.png 1.5x, /posts/circrna-pipeline/circRNA.png 2x"
        data-sizes="auto"
        alt="/posts/circrna-pipeline/circRNA.png"
        title="/posts/circrna-pipeline/circRNA.png" /></div><div class="details toc" id="toc-static"  kept="">
                <div class="details-summary toc-title">
                    <span>目录</span>
                    <span><i class="details-icon fas fa-angle-right"></i></span>
                </div>
                <div class="details-content toc-content" id="toc-content-static"><nav id="TableOfContents">
  <ul>
    <li><a href="#running-the-pipeline">Running the pipeline</a>
      <ul>
        <li><a href="#updating-the-pipeline">Updating the pipeline</a></li>
        <li><a href="#reproducibility">Reproducibility</a></li>
      </ul>
    </li>
    <li><a href="#input-specifications">Input specifications</a>
      <ul>
        <li><a href="#--input-path"><code>--input &quot;&lt;path&gt;&quot;</code></a>
          <ul>
            <li><a href="#fastq">fastq</a>
              <ul>
                <li><a href="#bam">bam</a></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><a href="#--input-samplescsv"><code>--input samples.csv</code></a></li>
        <li><a href="#--phenotype"><code>--phenotype</code></a></li>
      </ul>
    </li>
    <li><a href="#analysis-modules">Analysis modules</a>
      <ul>
        <li><a href="#circrnaworkflow-discovery">circrnaworkflow discovery</a>
          <ul>
            <li><a href="#tool-selection">Tool selection</a></li>
            <li><a href="#circrnaworkflow-filtering">circrnaworkflow filtering</a>
              <ul>
                <li><a href="#bsj-reads">BSJ reads</a></li>
                <li><a href="#multiple-tool-filtering">Multiple tool filtering</a></li>
              </ul>
            </li>
          </ul>
        </li>
        <li><a href="#mirna-prediction">miRNA prediction</a></li>
        <li><a href="#differential-circrnaworkflow-analysis">Differential circrnaworkflow analysis</a></li>
      </ul>
    </li>
    <li><a href="#core-nextflow-arguments">Core Nextflow arguments</a>
      <ul>
        <li><a href="#-profile"><code>-profile</code></a></li>
        <li><a href="#-resume"><code>-resume</code></a></li>
        <li><a href="#-c"><code>-c</code></a>
          <ul>
            <li><a href="#custom-resource-requests">Custom resource requests</a></li>
          </ul>
        </li>
        <li><a href="#running-in-the-background">Running in the background</a>
          <ul>
            <li><a href="#nextflow-memory-requirements">Nextflow memory requirements</a></li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</nav></div>
            </div><div class="content" id="content"><h1 id="nf-corecircrnaworkflow-usage">nf-core/circrnaworkflow: Usage</h1>
<p>It is recommended that first time users run <code>nf-core/circrnaworkflow</code> with the minimal test dataset either locally or on a HPC, referring to the <a href="https://nf-co.re/circrnaworkflow/dev/output" target="_blank" rel="noopener noreffer">output documentation</a> before running a full analysis.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">nextflow run nf-core/circrnaworkflow -profile test
</code></pre></td></tr></table>
</div>
</div><p>Run the test dataset on a HPC:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">nextflow run nf-core/circrnaworkflow -profile test,&lt;docker/singularity/podman/institute&gt;
</code></pre></td></tr></table>
</div>
</div><h2 id="running-the-pipeline">Running the pipeline</h2>
<p>A typical command for running the pipeline is as follows:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">nextflow run nf-core/circrnaworkflow \
    -profile &lt;docker/singularity/podman/institute&gt; \
    --genome &#39;GRCh37&#39; \
    --input &#39;samples.csv&#39; \
    --input_type &#39;fastq&#39;
</code></pre></td></tr></table>
</div>
</div><p>By default, <code>nf-core/circrnaworkflow</code> runs the circrnaworkflow discovery analysis module using <code>CIRCexplorer2</code>. The above command will perform circrnaworkflow quantification using these tools on ENSEMBL GRCh37 reference annotation files as defined in the iGenomes config.</p>
<h3 id="updating-the-pipeline">Updating the pipeline</h3>
<p>To make sure that you&rsquo;re running the latest version of the pipeline, make sure that you regularly update the cached version of the pipeline:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">nextflow pull nf-core/circrnaworkflow
</code></pre></td></tr></table>
</div>
</div><p>When you run the above command, Nextflow automatically pulls the pipeline code from GitHub and stores it as a cached version. When running the pipeline after this, it will always use the cached version if available - even if the pipeline has been updated since.</p>
<h3 id="reproducibility">Reproducibility</h3>
<p>It&rsquo;s a good idea to specify a pipeline version when running the pipeline on your data. This ensures that a specific version of the pipeline code and software are used when you run your pipeline. If you keep using the same tag, you&rsquo;ll be running the same version of the pipeline, even if there have been changes to the code since.</p>
<p>First, go to the <a href="https://github.com/nf-core/circrnaworkflow/releases" target="_blank" rel="noopener noreffer">nf-core/circrnaworkflow releases page</a> and find the latest version number - numeric only (eg. <code>1.3.1</code>). Then specify this when running the pipeline with <code>-r</code> (one hyphen) - eg. <code>-r 1.3.1</code>.</p>
<p>This version number will be logged in reports when you run the pipeline, so that you&rsquo;ll know what you used when you look back in the future.</p>
<h2 id="input-specifications">Input specifications</h2>
<p>Input data can be passed to <code>nf-core/circrnaworkflow</code> in two possible ways using the <code>--input</code> parameter.</p>
<h3 id="--input-path"><code>--input &quot;&lt;path&gt;&quot;</code></h3>
<p>The simplest way to pass input data to <code>nf-core/circrnaworkflow</code> is by providing the path to the input data with a suitable wildcard glob pattern:</p>
<h4 id="fastq">fastq</h4>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">--input &#34;/data/*_r{1,2}.fastq.gz&#34;
</code></pre></td></tr></table>
</div>
</div><h5 id="bam">bam</h5>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">--input &#34;/data/*.bam&#34;
</code></pre></td></tr></table>
</div>
</div><blockquote>
<p>Beware that providing a path to input data will result in samples being named according to the common tuple key based on the glob pattern supplied. Take this into consideration when designing your phenotype file for differential expression analysis.</p>
</blockquote>
<h3 id="--input-samplescsv"><code>--input samples.csv</code></h3>
<p>Alternatively, the user may wish to provide a CSV file containing the absolute paths to input fastq/bam files.</p>
<p>The headers of the CSV file must be: <code>Sample_ID,Read1,Read2,Bam</code>.</p>
<blockquote>
<p>This approach is recommended for most real life situations, where in-house sequencing facilities file naming convention requires the user to manually match file names to metadata. The below input files use <code>TCGA</code> identifiers as proof of concept.</p>
</blockquote>
<p>Valid examples for fastq/bam input data in a CSV file is given below:</p>
<table>
<thead>
<tr>
<th>Sample_ID</th>
<th style="text-align:left">Read1</th>
<th>Read2</th>
<th>Bam</th>
</tr>
</thead>
<tbody>
<tr>
<td>TCGA-EJ-7783-11A</td>
<td style="text-align:left">/data/f4c1b2b1-ba1f-4355-a1ac-3e952cf351a5_gdc_realn_rehead_R1.fastq.gz</td>
<td>/data/f4c1b2b1-ba1f-4355-a1ac-3e952cf351a5_gdc_realn_rehead_R2.fastq.gz</td>
<td>NA</td>
</tr>
<tr>
<td>TCGA-G9-6365-11A</td>
<td style="text-align:left">/data/8a36555b-9e27-40ee-a8df-4b15d6580a02_gdc_realn_rehead_R1.fastq.gz</td>
<td>/data/8a36555b-9e27-40ee-a8df-4b15d6580a02_gdc_realn_rehead_R2.fastq.gz</td>
<td>NA</td>
</tr>
<tr>
<td>TCGA-EJ-7782-11A</td>
<td style="text-align:left">/data/8b3d4a3d-2bfa-48f8-b31f-901f49a5bf6b_gdc_realn_rehead_R1.fastq.gz</td>
<td>/data/8b3d4a3d-2bfa-48f8-b31f-901f49a5bf6b_gdc_realn_rehead_R2.fastq.gz</td>
<td>NA</td>
</tr>
<tr>
<td>TCGA-CH-5772-01A</td>
<td style="text-align:left">/data/b6546f66-3c13-4390-9643-d1fb3d660a2f_gdc_realn_rehead_R1.fastq.gz</td>
<td>/data/b6546f66-3c13-4390-9643-d1fb3d660a2f_gdc_realn_rehead_R2.fastq.gz</td>
<td>NA</td>
</tr>
<tr>
<td>TCGA-EJ-5518-01A</td>
<td style="text-align:left">/data/afbbc370-5970-43d3-b9f8-f40f8e649bb6_gdc_realn_rehead_R1.fastq.gz</td>
<td>/data/afbbc370-5970-43d3-b9f8-f40f8e649bb6_gdc_realn_rehead_R2.fastq.gz</td>
<td>NA</td>
</tr>
<tr>
<td>TCGA-KK-A8I4-01A</td>
<td style="text-align:left">/data/81254692-ee1e-4985-bd0a-4929eed4c620_gdc_realn_rehead_R1.fastq.gz</td>
<td>/data/81254692-ee1e-4985-bd0a-4929eed4c620_gdc_realn_rehead_R2.fastq.gz</td>
<td>NA</td>
</tr>
</tbody>
</table>
<hr>
<table>
<thead>
<tr>
<th style="text-align:left">Sample_ID</th>
<th>Read1</th>
<th>Read2</th>
<th style="text-align:left">Bam</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left">TCGA-EJ-7783-11A</td>
<td>NA</td>
<td>NA</td>
<td style="text-align:left">/data/f4c1b2b1-ba1f-4355-a1ac-3e952cf351a5_gdc_realn_rehead.bam</td>
</tr>
<tr>
<td style="text-align:left">TCGA-G9-6365-11A</td>
<td>NA</td>
<td>NA</td>
<td style="text-align:left">/data/8a36555b-9e27-40ee-a8df-4b15d6580a02_gdc_realn_rehead.bam</td>
</tr>
<tr>
<td style="text-align:left">TCGA-EJ-7782-11A</td>
<td>NA</td>
<td>NA</td>
<td style="text-align:left">/data/8b3d4a3d-2bfa-48f8-b31f-901f49a5bf6b_gdc_realn_rehead.bam</td>
</tr>
<tr>
<td style="text-align:left">TCGA-CH-5772-01A</td>
<td>NA</td>
<td>NA</td>
<td style="text-align:left">/data/b6546f66-3c13-4390-9643-d1fb3d660a2f_gdc_realn_rehead.bam</td>
</tr>
<tr>
<td style="text-align:left">TCGA-EJ-5518-01A</td>
<td>NA</td>
<td>NA</td>
<td style="text-align:left">/data/afbbc370-5970-43d3-b9f8-f40f8e649bb6_gdc_realn_rehead.bam</td>
</tr>
<tr>
<td style="text-align:left">TCGA-KK-A8I4-01A</td>
<td>NA</td>
<td>NA</td>
<td style="text-align:left">/data/81254692-ee1e-4985-bd0a-4929eed4c620_gdc_realn_rehead.bam</td>
</tr>
</tbody>
</table>
<blockquote>
<p>Do not leave any cell empty in the CSV file.</p>
</blockquote>
<h3 id="--phenotype"><code>--phenotype</code></h3>
<p>When running the differential expression analysis module, an input <code>phenotype.csv</code> file is required to specify levels for <code>DESeq2</code>. At a minimum, the user must supply one column of levels for <code>DESeq2</code> which <strong>must be called condition</strong>. This should be the primary contrast of interest in your experiment (e.g case vs. control). If additional columns are supplied to the phenotype file, they will be controlled for in the linear mixed model. A brief proof of concept is given below in R notation:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-R" data-lang="R"><span class="nf">colnames</span><span class="p">(</span><span class="n">phenotype</span><span class="p">)</span>
 <span class="n">[1]</span> <span class="s">&#39;Sample_ID&#39;</span> <span class="s">&#39;condition&#39;</span>

<span class="nf">print</span><span class="p">(</span><span class="n">dds</span><span class="o">$</span><span class="n">design</span><span class="p">)</span>
 <span class="n">[1]</span> <span class="s">&#39; ~ condition&#39;</span>
</code></pre></td></tr></table>
</div>
</div><div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-R" data-lang="R"><span class="nf">colnames</span><span class="p">(</span><span class="n">phenotype</span><span class="p">)</span>
 <span class="n">[1]</span> <span class="s">&#39;Sample_ID&#39;</span> <span class="s">&#39;condition&#39;</span> <span class="s">&#39;replicates&#39;</span> <span class="s">&#39;location&#39;</span>

<span class="nf">print</span><span class="p">(</span><span class="n">dds</span><span class="o">$</span><span class="n">design</span><span class="p">)</span>
 <span class="n">[1]</span> <span class="s">&#39; ~ location + replicates + condition&#39;</span>
</code></pre></td></tr></table>
</div>
</div><p>It is recommended to use an input CSV file in conjunction with your phenotype file as the <code>Sample_ID</code> column <strong>must match</strong> the first column of the <code>phenotype.csv</code> file.</p>
<p>A valid example of a <code>phenotype.csv</code> file (matching the TCGA example input CSV file above) is given below:</p>
<table>
<thead>
<tr>
<th>Sample_ID</th>
<th>condition</th>
</tr>
</thead>
<tbody>
<tr>
<td>TCGA-EJ-7783-11A</td>
<td>control</td>
</tr>
<tr>
<td>TCGA-G9-6365-11A</td>
<td>control</td>
</tr>
<tr>
<td>TCGA-EJ-7782-11A</td>
<td>control</td>
</tr>
<tr>
<td>TCGA-CH-5772-01A</td>
<td>tumor</td>
</tr>
<tr>
<td>TCGA-EJ-5518-01A</td>
<td>tumor</td>
</tr>
<tr>
<td>TCGA-KK-A8I4-01A</td>
<td>tumor</td>
</tr>
</tbody>
</table>
<h2 id="analysis-modules">Analysis modules</h2>
<p><code>nf-core/circrnaworkflow</code> provides 3 analysis modules to the user:</p>
<ol>
<li>circrnaworkflow quantification &amp; annotation.</li>
<li>miRNA target prediction.</li>
<li>Differential circrnaworkflow expression analysis.</li>
</ol>
<h3 id="circrnaworkflow-discovery">circrnaworkflow discovery</h3>
<p>The core module of <code>nf-core/circrnaworkflow</code>, the user can utilise the most popular circrnaworkflow quantification tools to fully characterise the circrnaworkflow profile in samples. Currently, supported tools include <code>CIRCexplorer2</code>, <code>circrnaworkflow finder</code>, <code>CIRIquant</code>, <code>DCC</code>, <code>find circ</code> , <code>MapSplice</code> &amp; <code>Segemehl</code> however, the authors of <code>nf-core/circrnaworkflow</code> welcome contributions from authors of novel quantification tools to keep the workflow current.</p>
<p>By default, <code>nf-core/circrnaworkflow</code> runs the circrnaworkflow discovery analysis module.</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">nextflow run nf-core/circrnaworkflow \
    -profile &lt;docker/singularity/podman/institute&gt; \
    --genome &#39;GRCh37&#39; \
    --input &#39;samples.csv&#39; \
    --input_type &#39;fastq&#39; \
    --module &#39;circrnaworkflow_discovery&#39;
</code></pre></td></tr></table>
</div>
</div><p>To view the outputs of the module, please see the output <a href="https://nf-co.re/circrnaworkflow/dev/output#circrnaworkflow-quantification" target="_blank" rel="noopener noreffer">documentation</a>.</p>
<blockquote>
<p>Please note that this module must be included for every run of the workflow</p>
</blockquote>
<h4 id="tool-selection">Tool selection</h4>
<p>The user may use one, all or any combination of circrnaworkflow quantification tools listed above in the analysis. To select which tools to use for the analysis, specify the <code>--tool</code> parameter in the configuration profile or pass it via the command line when running the workflow:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">nextflow run nf-core/circrnaworkflow \
    -profile &lt;docker/singularity/podman/institute&gt; \
    --genome &#39;GRCh37&#39; \
    --input &#39;samples.csv&#39; \
    --input_type &#39;fastq&#39; \
    --module &#39;circrnaworkflow_discovery&#39; \
    --tool &#39;ciriquant, dcc, find_circ&#39;
</code></pre></td></tr></table>
</div>
</div><blockquote>
<p>When providing multiple tools, separate each entry with a comma.</p>
</blockquote>
<h4 id="circrnaworkflow-filtering">circrnaworkflow filtering</h4>
<p><code>nf-core/circrnaworkflow</code> offers robust filtering of each called circrnaworkflow to reduce the number of spurious calls within the dataset.</p>
<h5 id="bsj-reads">BSJ reads</h5>
<p>The user can specify the minimum number of reads spanning the back-splice junction site required for a circrnaworkflow to be considered for further analysis. circrnaworkflows with counts below this value will be filtered to remove from the results.</p>
<p>To apply this filtering method, specify the <code>--bsj_reads</code> parameter in the configuration profile or pass it via the command line when running the workflow:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">nextflow run nf-core/circrnaworkflow \
    -profile &lt;docker/singularity/podman/institute&gt; \
    --genome &#39;GRCh37&#39; \
    --input &#39;samples.csv&#39; \
    --input_type &#39;fastq&#39; \
    --phenotype &#39;phenotype.csv&#39; \
    --module &#39;circrnaworkflow_discovery&#39; \
    --tool &#39;ciriquant, dcc, find_circ&#39; \
    --bsj_reads 2
</code></pre></td></tr></table>
</div>
</div><p>Disable the filter by setting the value to 0.</p>
<h5 id="multiple-tool-filtering">Multiple tool filtering</h5>
<p>When more than one tool has been provided using the <code>--tool</code> parameter, the user can specify the minimum number of tools circrnaworkflows must be called by using <code>--tool_filter</code>. Setting this parameter to 0 or 1 will result in the union being output, i.e no filtering is applied. Setting this parameter to 2 will output circrnaworkflows that have been called by at least 2 quantification tools and so on.</p>
<blockquote>
<p>The integer provided to the parameter must be less than or equal to the number of quantification tools provided to <code>--tool</code>.</p>
</blockquote>
<p>To apply this filtering method, specify the <code>--tool_filter</code> parameter in the configuration profile or pass it via the command line when running the workflow:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span><span class="lnt">8
</span><span class="lnt">9
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">nextflow run nf-core/circrnaworkflow \
    -profile &lt;docker/singularity/podman/institute&gt; \
    --genome &#39;GRCh37&#39; \
    --input &#39;samples.csv&#39; \
    --input_type &#39;fastq&#39; \
    --module &#39;circrnaworkflow_discovery&#39; \
    --tool &#39;ciriquant, dcc, find_circ&#39; \
    --bsj_reads 2 \
    --tool_filter 2
</code></pre></td></tr></table>
</div>
</div><blockquote>
<p>This filtering method is reflected in the circrnaworkflow count matrix. Per tool circrnaworkflow annotations are subject to back-splice read filtering only.</p>
</blockquote>
<h3 id="mirna-prediction">miRNA prediction</h3>
<p>The second module of <code>nf-core/circrnaworkflow</code>, <code>mirna_prediction</code> analyses the mature spliced sequences of circrnaworkflows to test for the presence of miRNA response elements using both <code>miRanda</code> and <code>TargetScan</code>. Results from both tools are consolidated and filtering methods are applied to produce robust miRNA target predictions of circrnaworkflows in the dataset.</p>
<p>To invoke the module, specify the <code>--module</code> parameter via the configuration profile or pass it via the command line when running the workflow:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">nextflow run nf-core/circrnaworkflow \
    -profile &lt;docker/singularity/podman/institute&gt; \
    --genome &#39;GRCh37&#39; \
    --input &#39;samples.csv&#39; \
    --input_type &#39;fastq&#39; \
    --module &#39;circrnaworkflow_discovery, mirna_prediction&#39;
</code></pre></td></tr></table>
</div>
</div><p>To view the outputs of the module, please see the output <a href="https://nf-co.re/circrnaworkflow/dev/output#mirna-prediction" target="_blank" rel="noopener noreffer">documentation</a>.</p>
<h3 id="differential-circrnaworkflow-analysis">Differential circrnaworkflow analysis</h3>
<p>The third and final module of <code>nf-core/circrnaworkflow</code> performs differential expression analysis of circrnaworkflows, returning <code>DESeq2</code> result outputs, plots and diagnostic plots for the user. In order to run this module, it is essential that your <code>phenotype.csv</code> file is in the correct format - please refer to the input <a href="https://nf-co.re/circrnaworkflow/dev/usage#differential-expression-analysis" target="_blank" rel="noopener noreffer">specifications</a>.</p>
<p>To invoke the module, specify the <code>--module</code> parameter via the configuration profile or pass it via the command line when running the workflow:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span><span class="lnt">6
</span><span class="lnt">7
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">nextflow run nf-core/circrnaworkflow \
    -profile &lt;docker/singularity/podman/institute&gt; \
    --genome &#39;GRCh37&#39; \
    --input &#39;samples.csv&#39; \
    --input_type &#39;fastq&#39; \
    --phenotype &#39;phenotype.csv&#39; \
    --module &#39;circrnaworkflow_discovery, differential_expression&#39;
</code></pre></td></tr></table>
</div>
</div><p>To view the outputs of the module, please see the output <a href="https://nf-co.re/circrnaworkflow/dev/output#differential-expression-analysis" target="_blank" rel="noopener noreffer">documentation</a>.</p>
<h2 id="core-nextflow-arguments">Core Nextflow arguments</h2>
<blockquote>
<p><strong>NB:</strong> These options are part of Nextflow and use a <em>single</em> hyphen (pipeline parameters use a double-hyphen).</p>
</blockquote>
<h3 id="-profile"><code>-profile</code></h3>
<p>Use this parameter to choose a configuration profile. Profiles can give configuration presets for different compute environments.</p>
<p>Several generic profiles are bundled with the pipeline which instruct the pipeline to use software packaged using different methods (Docker, Singularity, Podman, Shifter, Charliecloud, Conda) - see below.</p>
<blockquote>
<p>We highly recommend the use of Docker or Singularity containers for full pipeline reproducibility, however when this is not possible, Conda is also supported.</p>
</blockquote>
<p>The pipeline also dynamically loads configurations from <a href="https://github.com/nf-core/configs" target="_blank" rel="noopener noreffer">https://github.com/nf-core/configs</a> when it runs, making multiple config profiles for various institutional clusters available at run time. For more information and to see if your system is available in these configs please see the <a href="https://github.com/nf-core/configs#documentation" target="_blank" rel="noopener noreffer">nf-core/configs documentation</a>.</p>
<p>Note that multiple profiles can be loaded, for example: <code>-profile test,docker</code> - the order of arguments is important!
They are loaded in sequence, so later profiles can overwrite earlier profiles.</p>
<p>If <code>-profile</code> is not specified, the pipeline will run locally and expect all software to be installed and available on the <code>PATH</code>. This is <em>not</em> recommended.</p>
<ul>
<li><code>docker</code>
<ul>
<li>A generic configuration profile to be used with <a href="https://docker.com/" target="_blank" rel="noopener noreffer">Docker</a></li>
<li>Pulls software from Docker Hub: <a href="https://hub.docker.com/r/nfcore/circrnaworkflow/" target="_blank" rel="noopener noreffer"><code>nfcore/circrnaworkflow</code></a></li>
</ul>
</li>
<li><code>singularity</code>
<ul>
<li>A generic configuration profile to be used with <a href="https://sylabs.io/docs/" target="_blank" rel="noopener noreffer">Singularity</a></li>
<li>Pulls software from Docker Hub: <a href="https://hub.docker.com/r/nfcore/circrnaworkflow/" target="_blank" rel="noopener noreffer"><code>nfcore/circrnaworkflow</code></a></li>
</ul>
</li>
<li><code>podman</code>
<ul>
<li>A generic configuration profile to be used with <a href="https://podman.io/" target="_blank" rel="noopener noreffer">Podman</a></li>
<li>Pulls software from Docker Hub: <a href="https://hub.docker.com/r/nfcore/circrnaworkflow/" target="_blank" rel="noopener noreffer"><code>nfcore/circrnaworkflow</code></a></li>
</ul>
</li>
<li><code>shifter</code>
<ul>
<li>A generic configuration profile to be used with <a href="https://nersc.gitlab.io/development/shifter/how-to-use/" target="_blank" rel="noopener noreffer">Shifter</a></li>
<li>Pulls software from Docker Hub: <a href="https://hub.docker.com/r/nfcore/circrnaworkflow/" target="_blank" rel="noopener noreffer"><code>nfcore/circrnaworkflow</code></a></li>
</ul>
</li>
<li><code>charliecloud</code>
<ul>
<li>A generic configuration profile to be used with <a href="https://hpc.github.io/charliecloud/" target="_blank" rel="noopener noreffer">Charliecloud</a></li>
<li>Pulls software from Docker Hub: <a href="https://hub.docker.com/r/nfcore/circrnaworkflow/" target="_blank" rel="noopener noreffer"><code>nfcore/circrnaworkflow</code></a></li>
</ul>
</li>
<li><code>conda</code>
<ul>
<li>Please only use Conda as a last resort i.e. when it&rsquo;s not possible to run the pipeline with Docker, Singularity, Podman, Shifter or Charliecloud.</li>
<li>A generic configuration profile to be used with <a href="https://conda.io/docs/" target="_blank" rel="noopener noreffer">Conda</a></li>
<li>Pulls most software from <a href="https://bioconda.github.io/" target="_blank" rel="noopener noreffer">Bioconda</a></li>
</ul>
</li>
<li><code>test</code>
<ul>
<li>A profile with a complete configuration for automated testing</li>
<li>Includes links to test data so needs no other parameters</li>
</ul>
</li>
</ul>
<h3 id="-resume"><code>-resume</code></h3>
<p>Specify this when restarting a pipeline. Nextflow will used cached results from any pipeline steps where the inputs are the same, continuing from where it got to previously.</p>
<p>You can also supply a run name to resume a specific run: <code>-resume [run-name]</code>. Use the <code>nextflow log</code> command to show previous run names.</p>
<h3 id="-c"><code>-c</code></h3>
<p>Specify the path to a specific config file (this is a core Nextflow command). See the <a href="https://nf-co.re/usage/configuration" target="_blank" rel="noopener noreffer">nf-core website documentation</a> for more information.</p>
<h4 id="custom-resource-requests">Custom resource requests</h4>
<p>Each step in the pipeline has a default set of requirements for number of CPUs, memory and time. For most of the steps in the pipeline, if the job exits with an error code of <code>143</code> (exceeded requested resources) it will automatically resubmit with higher requests (2 x original, then 3 x original). If it still fails after three times then the pipeline is stopped.</p>
<p>Whilst these default requirements will hopefully work for most people with most data, you may find that you want to customise the compute resources that the pipeline requests. You can do this by creating a custom config file. For example, to give the workflow process <code>star</code> 32GB of memory, you could use the following config:</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span><span class="lnt">2
</span><span class="lnt">3
</span><span class="lnt">4
</span><span class="lnt">5
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">process {
  withName: star {
    memory = 32.GB
  }
}
</code></pre></td></tr></table>
</div>
</div><p>To find the exact name of a process you wish to modify the compute resources, check the live-status of a nextflow run displayed on your terminal or check the nextflow error for a line like so: <code>Error executing process &gt; 'bwa'</code>. In this case the name to specify in the custom config file is <code>bwa</code>.</p>
<p>See the main <a href="https://www.nextflow.io/docs/latest/config.html" target="_blank" rel="noopener noreffer">Nextflow documentation</a> for more information.</p>
<p>If you are likely to be running <code>nf-core</code> pipelines regularly it may be a good idea to request that your custom config file is uploaded to the <code>nf-core/configs</code> git repository. Before you do this please can you test that the config file works with your pipeline of choice using the <code>-c</code> parameter (see definition above). You can then create a pull request to the <code>nf-core/configs</code> repository with the addition of your config file, associated documentation file (see examples in <a href="https://github.com/nf-core/configs/tree/master/docs" target="_blank" rel="noopener noreffer"><code>nf-core/configs/docs</code></a>), and amending <a href="https://github.com/nf-core/configs/blob/master/nfcore_custom.config" target="_blank" rel="noopener noreffer"><code>nfcore_custom.config</code></a> to include your custom profile.</p>
<p>If you have any questions or issues please send us a message on <a href="https://nf-co.re/join/slack" target="_blank" rel="noopener noreffer">Slack</a> on the <a href="https://nfcore.slack.com/channels/configs" target="_blank" rel="noopener noreffer"><code>#configs</code> channel</a>.</p>
<h3 id="running-in-the-background">Running in the background</h3>
<p>Nextflow handles job submissions and supervises the running jobs. The Nextflow process must run until the pipeline is finished.</p>
<p>The Nextflow <code>-bg</code> flag launches Nextflow in the background, detached from your terminal so that the workflow does not stop if you log out of your session. The logs are saved to a file.</p>
<p>Alternatively, you can use <code>screen</code> / <code>tmux</code> or similar tool to create a detached session which you can log back into at a later time.
Some HPC setups also allow you to run nextflow within a cluster job submitted your job scheduler (from where it submits more jobs).</p>
<h4 id="nextflow-memory-requirements">Nextflow memory requirements</h4>
<p>In some cases, the Nextflow Java virtual machines can start to request a large amount of memory.
We recommend adding the following line to your environment to limit this (typically in <code>~/.bashrc</code> or <code>~./bash_profile</code>):</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">1
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-fallback" data-lang="fallback">NXF_OPTS=&#39;-Xms1g -Xmx4g&#39;
</code></pre></td></tr></table>
</div>
</div></div><div class="post-footer" id="post-footer">
    <div class="post-info">
        <div class="post-info-line">
            <div class="post-info-mod">
                <span>更新于 2021-12-06</span>
            </div>
            <div class="post-info-license"></div>
        </div>
        <div class="post-info-line">
            <div class="post-info-md"><span>
                            <a class="link-to-markdown" href="/posts/circrna-pipeline/index.md" target="_blank">阅读原始文档</a>
                        </span></div>
            <div class="post-info-share">
                <span><a href="javascript:void(0);" title="分享到 Twitter" data-sharer="twitter" data-url="http://cb-majorbiolab.work/posts/circrna-pipeline/" data-title="circRNA quantification, differential expression analysis and miRNA target prediction of RNA-Seq data" data-via="https://twitter.com/czheluo" data-hashtags="DEGs,workflow,cricRNA"><i class="fab fa-twitter fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Facebook" data-sharer="facebook" data-url="http://cb-majorbiolab.work/posts/circrna-pipeline/" data-hashtag="DEGs"><i class="fab fa-facebook-square fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Linkedin" data-sharer="linkedin" data-url="http://cb-majorbiolab.work/posts/circrna-pipeline/"><i class="fab fa-linkedin fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Hacker News" data-sharer="hackernews" data-url="http://cb-majorbiolab.work/posts/circrna-pipeline/" data-title="circRNA quantification, differential expression analysis and miRNA target prediction of RNA-Seq data"><i class="fab fa-hacker-news fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Line" data-sharer="line" data-url="http://cb-majorbiolab.work/posts/circrna-pipeline/" data-title="circRNA quantification, differential expression analysis and miRNA target prediction of RNA-Seq data"><i data-svg-src="/lib/simple-icons/icons/line.min.svg"></i></a><a href="javascript:void(0);" title="分享到 微博" data-sharer="weibo" data-url="http://cb-majorbiolab.work/posts/circrna-pipeline/" data-title="circRNA quantification, differential expression analysis and miRNA target prediction of RNA-Seq data" data-ralateuid="https://weibo.com/2161302013/profile"><i class="fab fa-weibo fa-fw"></i></a></span>
            </div>
        </div>
    </div>

    <div class="post-info-more">
        <section class="post-tags"><i class="fas fa-tags fa-fw"></i>&nbsp;<a href="/tags/degs/">DEGs</a>,&nbsp;<a href="/tags/workflow/">workflow</a>,&nbsp;<a href="/tags/cricrna/">cricRNA</a></section>
        <section>
            <span><a href="javascript:void(0);" onclick="window.history.back();">返回</a></span>&nbsp;|&nbsp;<span><a href="/">主页</a></span>
        </section>
    </div>

    <div class="post-nav"><a href="/posts/visually-combining-expression-data-with-functional-analysis/" class="prev" rel="prev" title="visually combining expression data with functional analysis"><i class="fas fa-angle-left fa-fw"></i>visually combining expression data with functional analysis</a>
            <a href="/posts/ont-workflow/" class="next" rel="next" title="Nanopore demultiplexing, QC, mapping and downstream analysi pipeline">Nanopore demultiplexing, QC, mapping and downstream analysi pipeline<i class="fas fa-angle-right fa-fw"></i></a></div>
</div>
<div id="comments"><div id="valine" class="comment"></div><noscript>
                Please enable JavaScript to view the comments powered by <a href="https://valine.js.org/">Valine</a>.
            </noscript><div id="fb-root" class="comment"></div>
            <div
                class="fb-comments"
                data-href="http://cb-majorbiolab.work/posts/circrna-pipeline/"
                data-width="100%"
                data-numposts="10"
            ></div><noscript>
                Please enable JavaScript to view the comments powered by <a href="https://developers.facebook.com/docs/plugins/comments/"></a>Facebook</a>.
            </noscript></div></article></div>
            </main><footer class="footer">
        <div class="footer-container"><div class="footer-line">MAJOR Computational Biology Research Lab</div>
                <div class="footer-line"><i class="far fa-copyright fa-fw"></i><span itemprop="copyrightYear">2020-2022, MENG LUO, All rights reserved.</span><span class="author" itemprop="copyrightHolder">&nbsp;<a href="/" target="_blank"></a></span></div>
                
        </div>
    </footer></div>

        <div id="fixed-buttons"><a href="#" id="back-to-top" class="fixed-button" title="回到顶部">
                <i class="fas fa-arrow-up fa-fw"></i>
            </a><a href="#" id="view-comments" class="fixed-button" title="查看评论">
                <i class="fas fa-comment fa-fw"></i>
            </a>
        </div><link rel="stylesheet" href="/lib/valine/valine.min.css"><link rel="stylesheet" href="/lib/lightgallery/lightgallery.min.css"><link rel="stylesheet" href="/lib/katex/katex.min.css"><link rel="stylesheet" href="/lib/katex/copy-tex.min.css"><script type="text/javascript" src="/lib/valine/Valine.min.js"></script><script type="text/javascript" src="https://connect.facebook.net/zh_CN/sdk.js#xfbml=1&amp;version=v5.0&amp;appId=&amp;autoLogAppEvents=1" defer></script><script type="text/javascript" src="/lib/smooth-scroll/smooth-scroll.min.js"></script><script type="text/javascript" src="/lib/autocomplete/autocomplete.min.js"></script><script type="text/javascript" src="/lib/algoliasearch/algoliasearch-lite.umd.min.js"></script><script type="text/javascript" src="/lib/lazysizes/lazysizes.min.js"></script><script type="text/javascript" src="/lib/lightgallery/lightgallery.min.js"></script><script type="text/javascript" src="/lib/lightgallery/lg-thumbnail.min.js"></script><script type="text/javascript" src="/lib/lightgallery/lg-zoom.min.js"></script><script type="text/javascript" src="/lib/clipboard/clipboard.min.js"></script><script type="text/javascript" src="/lib/sharer/sharer.min.js"></script><script type="text/javascript" src="/lib/katex/katex.min.js"></script><script type="text/javascript" src="/lib/katex/auto-render.min.js"></script><script type="text/javascript" src="/lib/katex/copy-tex.min.js"></script><script type="text/javascript" src="/lib/katex/mhchem.min.js"></script><script type="text/javascript">window.config={"code":{"copyTitle":"复制到剪贴板","maxShownLines":10},"comment":{"valine":{"appId":"Noyy9UKaXGwYBIUdvBTt7u76-gzGzoHsz","appKey":"15w6pTF8mFoxnm8muOFx5buM","avatar":"mp","el":"#valine","emojiCDN":"https://cdn.jsdelivr.net/npm/emoji-datasource-google@5.0.1/img/google/64/","emojiMaps":{"100":"1f4af.png","alien":"1f47d.png","anger":"1f4a2.png","angry":"1f620.png","anguished":"1f627.png","astonished":"1f632.png","black_heart":"1f5a4.png","blue_heart":"1f499.png","blush":"1f60a.png","bomb":"1f4a3.png","boom":"1f4a5.png","broken_heart":"1f494.png","brown_heart":"1f90e.png","clown_face":"1f921.png","cold_face":"1f976.png","cold_sweat":"1f630.png","confounded":"1f616.png","confused":"1f615.png","cry":"1f622.png","crying_cat_face":"1f63f.png","cupid":"1f498.png","dash":"1f4a8.png","disappointed":"1f61e.png","disappointed_relieved":"1f625.png","dizzy":"1f4ab.png","dizzy_face":"1f635.png","drooling_face":"1f924.png","exploding_head":"1f92f.png","expressionless":"1f611.png","face_vomiting":"1f92e.png","face_with_cowboy_hat":"1f920.png","face_with_hand_over_mouth":"1f92d.png","face_with_head_bandage":"1f915.png","face_with_monocle":"1f9d0.png","face_with_raised_eyebrow":"1f928.png","face_with_rolling_eyes":"1f644.png","face_with_symbols_on_mouth":"1f92c.png","face_with_thermometer":"1f912.png","fearful":"1f628.png","flushed":"1f633.png","frowning":"1f626.png","ghost":"1f47b.png","gift_heart":"1f49d.png","green_heart":"1f49a.png","grimacing":"1f62c.png","grin":"1f601.png","grinning":"1f600.png","hankey":"1f4a9.png","hear_no_evil":"1f649.png","heart":"2764-fe0f.png","heart_decoration":"1f49f.png","heart_eyes":"1f60d.png","heart_eyes_cat":"1f63b.png","heartbeat":"1f493.png","heartpulse":"1f497.png","heavy_heart_exclamation_mark_ornament":"2763-fe0f.png","hole":"1f573-fe0f.png","hot_face":"1f975.png","hugging_face":"1f917.png","hushed":"1f62f.png","imp":"1f47f.png","innocent":"1f607.png","japanese_goblin":"1f47a.png","japanese_ogre":"1f479.png","joy":"1f602.png","joy_cat":"1f639.png","kiss":"1f48b.png","kissing":"1f617.png","kissing_cat":"1f63d.png","kissing_closed_eyes":"1f61a.png","kissing_heart":"1f618.png","kissing_smiling_eyes":"1f619.png","laughing":"1f606.png","left_speech_bubble":"1f5e8-fe0f.png","love_letter":"1f48c.png","lying_face":"1f925.png","mask":"1f637.png","money_mouth_face":"1f911.png","nauseated_face":"1f922.png","nerd_face":"1f913.png","neutral_face":"1f610.png","no_mouth":"1f636.png","open_mouth":"1f62e.png","orange_heart":"1f9e1.png","partying_face":"1f973.png","pensive":"1f614.png","persevere":"1f623.png","pleading_face":"1f97a.png","pouting_cat":"1f63e.png","purple_heart":"1f49c.png","rage":"1f621.png","relaxed":"263a-fe0f.png","relieved":"1f60c.png","revolving_hearts":"1f49e.png","right_anger_bubble":"1f5ef-fe0f.png","robot_face":"1f916.png","rolling_on_the_floor_laughing":"1f923.png","scream":"1f631.png","scream_cat":"1f640.png","see_no_evil":"1f648.png","shushing_face":"1f92b.png","skull":"1f480.png","skull_and_crossbones":"2620-fe0f.png","sleeping":"1f634.png","sleepy":"1f62a.png","slightly_frowning_face":"1f641.png","slightly_smiling_face":"1f642.png","smile":"1f604.png","smile_cat":"1f638.png","smiley":"1f603.png","smiley_cat":"1f63a.png","smiling_face_with_3_hearts":"1f970.png","smiling_imp":"1f608.png","smirk":"1f60f.png","smirk_cat":"1f63c.png","sneezing_face":"1f927.png","sob":"1f62d.png","space_invader":"1f47e.png","sparkling_heart":"1f496.png","speak_no_evil":"1f64a.png","speech_balloon":"1f4ac.png","star-struck":"1f929.png","stuck_out_tongue":"1f61b.png","stuck_out_tongue_closed_eyes":"1f61d.png","stuck_out_tongue_winking_eye":"1f61c.png","sunglasses":"1f60e.png","sweat":"1f613.png","sweat_drops":"1f4a6.png","sweat_smile":"1f605.png","thinking_face":"1f914.png","thought_balloon":"1f4ad.png","tired_face":"1f62b.png","triumph":"1f624.png","two_hearts":"1f495.png","unamused":"1f612.png","upside_down_face":"1f643.png","weary":"1f629.png","white_frowning_face":"2639-fe0f.png","white_heart":"1f90d.png","wink":"1f609.png","woozy_face":"1f974.png","worried":"1f61f.png","yawning_face":"1f971.png","yellow_heart":"1f49b.png","yum":"1f60b.png","zany_face":"1f92a.png","zipper_mouth_face":"1f910.png","zzz":"1f4a4.png"},"enableQQ":false,"highlight":true,"lang":"zh-cn","pageSize":10,"placeholder":"你的评论 ...","recordIP":true,"visitor":true}},"lightGallery":{"actualSize":false,"exThumbImage":"data-thumbnail","hideBarsDelay":2000,"selector":".lightgallery","speed":400,"thumbContHeight":80,"thumbWidth":80,"thumbnail":true},"math":{"delimiters":[{"display":true,"left":"$$","right":"$$"},{"display":true,"left":"\\[","right":"\\]"},{"display":false,"left":"$","right":"$"},{"display":false,"left":"\\(","right":"\\)"}],"strict":false},"search":{"algoliaAppID":"VX4BXOJBC5","algoliaIndex":"majorbio.zh-cn","algoliaSearchKey":"53790f167edb548b217bfe91dd891ac0","highlightTag":"em","maxResultLength":10,"noResultsFound":"没有找到结果","snippetLength":50,"type":"algolia"}};</script><script type="text/javascript" src="/js/theme.min.js"></script></body>
</html>
