<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="robots" content="noodp" />
        <meta http-equiv="X-UA-Compatible" content="IE=edge, chrome=1">
        <title>RNAseq定量软件比较 - MAJOR Computational Biology Research Lab</title><meta name="Description" content="RNAseq定量软件的简介"><meta property="og:title" content="RNAseq定量软件比较" />
<meta property="og:description" content="RNAseq定量软件的简介" />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://cb-majorbiolab.work/posts/rnaseq-quantification/" />
<meta property="og:image" content="http://cb-majorbiolab.work/posts/rnaseq-quantification/image-20211020155526985.png"/>
<meta property="article:published_time" content="2021-10-16T14:30:59+08:00" />
<meta property="article:modified_time" content="2021-10-16T16:29:59+08:00" />
<meta name="twitter:card" content="summary_large_image"/>
<meta name="twitter:image" content="http://cb-majorbiolab.work/posts/rnaseq-quantification/image-20211020155526985.png"/>
<meta name="twitter:title" content="RNAseq定量软件比较"/>
<meta name="twitter:description" content="RNAseq定量软件的简介"/>
<meta name="application-name" content="MAJOR Computational Biology Research Lab">
<meta name="apple-mobile-web-app-title" content="MAJOR Computational Biology Research Lab"><meta name="theme-color" content="#ffffff"><meta name="msapplication-TileColor" content="#da532c"><link rel="icon" href="/images/MBIL.svg"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5"><link rel="manifest" href="/site.webmanifest"><link rel="canonical" href="http://cb-majorbiolab.work/posts/rnaseq-quantification/" /><link rel="prev" href="http://cb-majorbiolab.work/posts/differential-analysis-of-circrna-count-data-circrna/" /><link rel="next" href="http://cb-majorbiolab.work/posts/dimensionality-reduction-methods-for-rna-seq-analysis/" /><link rel="stylesheet" href="/lib/normalize/normalize.min.css"><link rel="stylesheet" href="/css/style.min.css"><link rel="stylesheet" href="/lib/fontawesome-free/all.min.css"><link rel="stylesheet" href="/lib/animate/animate.min.css"><script type="application/ld+json">
    {
        "@context": "http://schema.org",
        "@type": "BlogPosting",
        "headline": "RNAseq定量软件比较",
        "inLanguage": "en",
        "mainEntityOfPage": {
            "@type": "WebPage",
            "@id": "http:\/\/cb-majorbiolab.work\/posts\/rnaseq-quantification\/"
        },"image": [{
                            "@type": "ImageObject",
                            "url": "http:\/\/cb-majorbiolab.work\/posts\/rnaseq-quantification\/image-20211020155526985.png",
                            "width":  494 ,
                            "height":  396 
                        }],"genre": "posts","keywords": "RNAseq, DESeq2, 差异分析, count, 表达量","wordcount":  14611 ,
        "url": "http:\/\/cb-majorbiolab.work\/posts\/rnaseq-quantification\/","datePublished": "2021-10-16T14:30:59+08:00","dateModified": "2021-10-16T16:29:59+08:00","publisher": {
            "@type": "Organization",
            "name": "美吉生物信息团队"},"author": {
                "@type": "Person",
                "name": "Wanyun Yang"
            },"description": "RNAseq定量软件的简介"
    }
    </script></head>
    <body header-desktop="" header-mobile=""><script type="text/javascript">(window.localStorage && localStorage.getItem('theme') ? localStorage.getItem('theme') === 'dark' : ('' === 'auto' ? window.matchMedia('(prefers-color-scheme: dark)').matches : '' === 'dark')) && document.body.setAttribute('theme', 'dark');</script>

        <div id="mask"></div><div class="wrapper"><header class="desktop" id="header-desktop">
    <div class="header-wrapper">
        <div class="header-title">
            <a href="/" title="MAJOR Computational Biology Research Lab"><img
        class="lazyload logo"
        src="/svg/loading.min.svg"
        data-src="/images/MBIL.svg"
        data-srcset="/images/MBIL.svg, /images/MBIL.svg 1.5x, /images/MBIL.svg 2x"
        data-sizes="auto"
        alt="/images/MBIL.svg"
        title="/images/MBIL.svg" /></a>
        </div>
        <div class="menu">
            <div class="menu-inner"><a class="menu-item" href="/posts/" title="Home"> Home </a><a class="menu-item" href="/workshop/" title=" "> Workshop/Talk </a><a class="menu-item" href="/publication/" title="Publications"> Publications </a><a class="menu-item" href="/team/" title="Team"> Team </a><a class="menu-item" href="/resource/" title="Resource"> Resource </a><a class="menu-item" href="/software/" title="Software"> Software </a><a class="menu-item" href="/ne/" title="News/Events"> News/Events </a><a class="menu-item" href="/joinus/" title="Join/Contact"> Join/Contact </a><a class="menu-item" href="https://github.com/MajorBio" title="GitHub" rel="noopener noreffer" target="_blank"><i class='fab fa-github fa-fw'></i>  </a><span class="menu-item delimiter"></span><span class="menu-item search" id="search-desktop">
                        <input type="text" placeholder="搜索文章标题或内容..." id="search-input-desktop">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-desktop" title="搜索">
                            <i class="fas fa-search fa-fw"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-desktop" title="清空">
                            <i class="fas fa-times-circle fa-fw"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-desktop">
                            <i class="fas fa-spinner fa-fw fa-spin"></i>
                        </span>
                    </span><a href="javascript:void(0);" class="menu-item theme-switch" title="切换主题">
                    <i class="fas fa-adjust fa-fw"></i>
                </a>
            </div>
        </div>
    </div>
</header><header class="mobile" id="header-mobile">
    <div class="header-container">
        <div class="header-wrapper">
            <div class="header-title">
                <a href="/" title="MAJOR Computational Biology Research Lab"><img
        class="lazyload logo"
        src="/svg/loading.min.svg"
        data-src="/images/MBIL.svg"
        data-srcset="/images/MBIL.svg, /images/MBIL.svg 1.5x, /images/MBIL.svg 2x"
        data-sizes="auto"
        alt="/images/MBIL.svg"
        title="/images/MBIL.svg" /></a>
            </div>
            <div class="menu-toggle" id="menu-toggle-mobile">
                <span></span><span></span><span></span>
            </div>
        </div>
        <div class="menu" id="menu-mobile"><div class="search-wrapper">
                    <div class="search mobile" id="search-mobile">
                        <input type="text" placeholder="搜索文章标题或内容..." id="search-input-mobile">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-mobile" title="搜索">
                            <i class="fas fa-search fa-fw"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-mobile" title="清空">
                            <i class="fas fa-times-circle fa-fw"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-mobile">
                            <i class="fas fa-spinner fa-fw fa-spin"></i>
                        </span>
                    </div>
                    <a href="javascript:void(0);" class="search-cancel" id="search-cancel-mobile">
                        取消
                    </a>
                </div><a class="menu-item" href="/posts/" title="Home">Home</a><a class="menu-item" href="/workshop/" title=" ">Workshop/Talk</a><a class="menu-item" href="/publication/" title="Publications">Publications</a><a class="menu-item" href="/team/" title="Team">Team</a><a class="menu-item" href="/resource/" title="Resource">Resource</a><a class="menu-item" href="/software/" title="Software">Software</a><a class="menu-item" href="/ne/" title="News/Events">News/Events</a><a class="menu-item" href="/joinus/" title="Join/Contact">Join/Contact</a><a class="menu-item" href="https://github.com/MajorBio" title="GitHub" rel="noopener noreffer" target="_blank"><i class='fab fa-github fa-fw'></i></a><a href="javascript:void(0);" class="menu-item theme-switch" title="切换主题">
                <i class="fas fa-adjust fa-fw"></i>
            </a></div>
    </div>
</header>
<div class="search-dropdown desktop">
    <div id="search-dropdown-desktop"></div>
</div>
<div class="search-dropdown mobile">
    <div id="search-dropdown-mobile"></div>
</div><main class="main">
                <div class="container"><div class="toc" id="toc-auto">
            <h2 class="toc-title">目录</h2>
            <div class="toc-content" id="toc-content-auto"></div>
        </div><article class="page single"><h1 class="single-title animated flipInX">RNAseq定量软件比较</h1><div class="post-meta">
            <div class="post-meta-line"><span class="post-author"><a href="https://github.com/yangwanyun" title="Author" target="_blank" rel="noopener noreffer author" class="author"><i class="fas fa-user-circle fa-fw"></i>Wanyun Yang</a></span>&nbsp;<span class="post-category">收录于 <a href="/categories/rnaseq/"><i class="far fa-folder fa-fw"></i>RNAseq</a></span></div>
            <div class="post-meta-line"><i class="far fa-calendar-alt fa-fw"></i>&nbsp;<time datetime="2021-10-16">2021-10-16</time>&nbsp;<i class="fas fa-pencil-alt fa-fw"></i>&nbsp;约 14611 字&nbsp;
                <i class="far fa-clock fa-fw"></i>&nbsp;预计阅读 30 分钟&nbsp;<span id="/posts/rnaseq-quantification/" class="leancloud_visitors" data-flag-title="RNAseq定量软件比较">
                        <i class="far fa-eye fa-fw"></i>&nbsp;<span class=leancloud-visitors-count></span>&nbsp;次阅读
                    </span>&nbsp;</div>
        </div><div class="featured-image"><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/posts/rnaseq-quantification/image-20211020155526985.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020155526985.png, /posts/rnaseq-quantification/image-20211020155526985.png 1.5x, /posts/rnaseq-quantification/image-20211020155526985.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020155526985.png"
        title="RNAseq定量软件的简介" /></div><div class="details toc" id="toc-static"  kept="">
                <div class="details-summary toc-title">
                    <span>目录</span>
                    <span><i class="details-icon fas fa-angle-right"></i></span>
                </div>
                <div class="details-content toc-content" id="toc-content-static"><nav id="TableOfContents">
  <ul>
    <li><a href="#rnaseq定量软件的简介">RNAseq定量软件的简介</a></li>
    <li><a href="#kallistosalmonrsemstringtiefeaturecounthtseq软件的定量原理简介">kallisto,salmon,rsem,stringtie,featurecount,htseq软件的定量原理简介</a>
      <ul>
        <li><a href="#kallisto定量软件简介">kallisto定量软件简介</a>
          <ul>
            <li><a href="#kallisto官网">Kallisto官网</a></li>
            <li><a href="#kallisto-简介">kallisto 简介</a></li>
            <li><a href="#kallisto-比对原理">kallisto 比对原理</a></li>
            <li><a href="#kallisto-比对和定量结果分析">kallisto 比对和定量结果分析</a></li>
            <li><a href="#kallisto-的使用方法">kallisto 的使用方法</a></li>
            <li><a href="#kallisto-的输出文件">kallisto 的输出文件</a></li>
            <li><a href="#kallisto-的总结">kallisto 的总结</a></li>
          </ul>
        </li>
        <li><a href="#salmon定量软件简介">salmon定量软件简介</a>
          <ul>
            <li><a href="#salmon官网">salmon官网</a></li>
            <li><a href="#salmon简介">salmon简介</a></li>
            <li><a href="#salmon原理">salmon原理</a></li>
            <li><a href="#salmon的定量优势">Salmon的定量优势</a></li>
            <li><a href="#salmon的使用方法">Salmon的使用方法</a></li>
            <li><a href="#salmon输出结果">salmon输出结果</a></li>
            <li><a href="#salmon-的总结">salmon 的总结</a></li>
          </ul>
        </li>
        <li><a href="#rsemrnaseq-by-expectation-maximization定量软件简介">RSEM(RNASeq by Expectation Maximization)定量软件简介</a>
          <ul>
            <li><a href="#rsem官网">RSEM官网</a></li>
            <li><a href="#rsem简介">RSEM简介</a></li>
            <li><a href="#rsem的比对定量优势">RSEM的比对定量优势</a></li>
            <li><a href="#rsem原理">RSEM原理</a></li>
            <li><a href="#rsem结果评估">RSEM结果评估</a></li>
            <li><a href="#rsem的使用方法">RSEM的使用方法</a></li>
            <li><a href="#rsem输出结果">RSEM输出结果</a></li>
            <li><a href="#基因标准化常用方法">基因标准化常用方法</a></li>
            <li><a href="#rsem总结">RSEM总结</a></li>
          </ul>
        </li>
        <li><a href="#stringtie定量软件简介">StringTie定量软件简介</a>
          <ul>
            <li><a href="#stringtie官网">stringtie官网</a></li>
            <li><a href="#stringtie简介">stringtie简介</a></li>
            <li><a href="#stringtie原理和优势">StringTie原理和优势</a></li>
            <li><a href="#stingtie定量方法">StingTie定量方法</a></li>
            <li><a href="#stingtie输出文件">StingTie输出文件</a></li>
            <li><a href="#stingtie总结">StingTie总结</a></li>
          </ul>
        </li>
        <li><a href="#htseq-high-throughput-sequencing定量软件简介">HTSeq (high-throughput sequencing)定量软件简介</a>
          <ul>
            <li><a href="#htseq官网">HTSeq官网</a></li>
            <li><a href="#htseq简介">HTSeq简介</a></li>
            <li><a href="#htseq原理">HTSeq原理</a></li>
            <li><a href="#htseq定量方法">HTSeq定量方法</a></li>
            <li><a href="#htseq输出结果">HTSeq输出结果</a></li>
            <li><a href="#htseq总结">HTSeq总结</a></li>
          </ul>
        </li>
        <li><a href="#featurecountssubread-package定量软件简介">featurecounts（Subread package）定量软件简介</a>
          <ul>
            <li><a href="#featurecounts官网">featurecounts官网</a></li>
            <li><a href="#featurecounts简介">featurecounts简介</a></li>
            <li><a href="#featurecounts原理">featurecounts原理</a></li>
            <li><a href="#featurecounts结果评估">featurecounts结果评估</a></li>
            <li><a href="#featurecounts使用方法">featurecounts使用方法</a></li>
            <li><a href="#featurecounts输出结果">featurecounts输出结果</a></li>
            <li><a href="#featurecounts总结">featurecounts总结</a></li>
          </ul>
        </li>
      </ul>
    </li>
    <li><a href="#kallistosalmonrsemstringtiefeaturecounthtseq定量结果的差异分析比较">kallisto,salmon,rsem,stringtie,featurecount,htseq定量结果的差异分析比较</a>
      <ul>
        <li><a href="#测试数据不同软件定量结果分析">测试数据不同软件定量结果分析</a>
          <ul>
            <li><a href="#tpm值的相关性分析">tpm值的相关性分析</a></li>
            <li><a href="#tpm值的pca">tpm值的pca</a></li>
            <li><a href="#相关性和pca总结">相关性和pca总结：</a></li>
            <li><a href="#tpm值的大小情况">tpm值的大小情况</a></li>
            <li><a href="#tpm值大小总结">tpm值大小总结：</a></li>
          </ul>
        </li>
        <li><a href="#测试数据不同定量结果的差异分析">测试数据不同定量结果的差异分析</a>
          <ul>
            <li><a href="#差异gene的venn图">差异gene的Venn图</a></li>
            <li><a href="#差异基因聚类分析">差异基因聚类分析</a></li>
            <li><a href="#差异分析总结">差异分析总结</a></li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</nav></div>
            </div><div class="content" id="content"><h1 id="一些rnaseq定量软件的简介">一些RNAseq定量软件的简介</h1>
<p>RNAseq定量软件的简介</p>
<p>kallisto,salmon,rsem,stringtie,featurecount,htseq软件的定量原理简介</p>
<p>kallisto,salmon,rsem,stringtie,featurecount,htseq定量结果的差异分析比较</p>
<h2 id="rnaseq定量软件的简介">RNAseq定量软件的简介</h2>
<p><strong>基因水平</strong>(gene-level)：常见的软件包括HTSeq-count，featureCounts，BEDTools, Qualimap, Rsubread, GenomicRanges等(Anders, S.,2012)</p>
<p>这些软件在处理测序中得到的multimapping reads时方式不同，例如HTSeq-count就直接当它们不存在，而Qualimpa则是一人一份，平均分配,featureCounts是参数决定要不要multimapping reads</p>
<p><strong>转录本水平</strong>(transcript-level)：常用工具为RSEM，Cufflinks，StringTie，eXpress，kallisto，silfish，salmon等(Tuimala, J. , 2014.)</p>
<p>这些软件要处理的难题是转录本亚型（isoforms）之间通常是有重叠的，当二代测序读长低于转录本长度时，如何进行区分？这些工具大多采用的都是expectation maximization（EM）算法
上述软件中RSEM，Cufflinks，StringTie，eXpress都是alignment-based，目前许多alignment-free软件，如kallisto, silfish, salmon，能够省去比对这一步，直接得到read count，在运行效率上更高。不过最近一篇文献(Sahraeian, S. M. E,2017)指出这类方法在估计丰度时存在样本特异性和读长偏差</p>
<p>[1] Sahraeian, S. M. E., Mohiyuddin, M., Sebra, R., Tilgner, H., Afshar, P. T., Au, K. F., … Lam, H. Y. K. (2017). <em>Gaining comprehensive biological insight into the transcriptome by performing a broad-spectrum RNA-seq analysis. Nature Communications, 8(1).</em> doi:10.1038/s41467-017-00050-4 
[2] Anders, S., Reyes, A., &amp; Huber, W. (2012). <em>Detecting differential usage of exons from RNA-Seq data. Nature Precedings.</em> doi:10.1038/npre.2012.6837.2 
[3] Tuimala, J. . (2014). Rna-seq data analysis: a practical approach. <em>Crc Press</em>.</p>
<h2 id="kallistosalmonrsemstringtiefeaturecounthtseq软件的定量原理简介">kallisto,salmon,rsem,stringtie,featurecount,htseq软件的定量原理简介</h2>
<h3 id="kallisto定量软件简介">kallisto定量软件简介</h3>
<h4 id="kallisto官网">Kallisto官网</h4>
<p><a href="https://pachterlab.github.io/kallisto/starting">https://pachterlab.github.io/kallisto/starting</a></p>
<h4 id="kallisto-简介">kallisto 简介</h4>
<p>kallisto 是2016年Nature Biotechnology 杂志发布的一款<font color='red'>无须比对</font>的转录本定量工具（Nicolas L Bray，2016）
采用了名为pseudo-alignment的算法，pseudo-alignment 算法不关系reads具体的比对位置，而是通过reads的kmer特征来判断其属于哪一条转录本
kallisto 软件的优势：
<font color='red'>极大地降低了运行时间</font>
算法上更新——绕过传统方法中序列比对（alignment）的步骤直接做量化分析，软件重于确定一个 read 属于哪一个基因，而不关心这个 read 在基因上的具体位置</p>
<p>[1] Nicolas L Bray, Harold Pimentel, Páll Melsted and Lior Pachter, <a href="http://www.nature.com/nbt/journal/v34/n5/full/nbt.3519.html" target="_blank" rel="noopener noreffer">Near-optimal probabilistic RNA-seq quantification</a>, Nature Biotechnology <strong>34</strong>, 525–527 (2016), doi:10.1038/nbt.3519</p>
<h4 id="kallisto-比对原理">kallisto 比对原理</h4>
<p>第一步：建立索引
k-mers默认是31，k-mers越小会越敏感。图上的每一个node都代表一个k-mer。</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020133348523.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020133348523.png, image-20211020133348523.png 1.5x, /posts/rnaseq-quantification/image-20211020133348523.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020133348523.png"
        title="image-20211020133348523" /></p>
<p>如上图有三条overlapping的 transcripts，红、蓝、绿各代表一条。
k-compatibility class：transcripts包含node的k-mers ，那么称transcripts为该node的一个k-comptibility class。如图上最左边的node有三个k-compatibility class，因为三条transcripts都包含该node代表的k-mer。三个最上面的node只有两个k-compatibility class，因为只有蓝和红两条transcripts包含node 代表的k-mer。
contig:连续的有相同k-compatibility class的node组成。如上图中最左边的三个node构成一个contig，最上面的node也构成一个contig。
index会建立一个hash table&lt;KmerEntry,KmerHash&gt; KmerEntry里包含k-mer映射到contig的信息，KmerHash kmer的hash值。</p>
<p>第二步：比对</p>
<p>上图黑色的代表一条read。比对的时候read也会被切成一个个的k-mer。黑色的node代表比对上的。</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020133510645.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020133510645.png, image-20211020133510645.png 1.5x, /posts/rnaseq-quantification/image-20211020133510645.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020133510645.png"
        title="image-20211020133510645" /></p>
<p>然后找出每个比对上的node的k-compatibility class，然后取这些k-compatibility class的交集，就得到read的k-compatibility class，如果是paired-end，那就沿着一条参考序列取k-compatibility class。</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020133536309.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020133536309.png, image-20211020133536309.png 1.5x, /posts/rnaseq-quantification/image-20211020133536309.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020133536309.png"
        title="image-20211020133536309" /></p>
<p>冗余信息。最左段的3个node有相同的k-compatibility class，文章中又把这些相同的node称为k-equivalence class。在比对的过程中我们可以<font color='red'>跳过这些k-equivalence class</font>。这些信息已经存储在contig里。同一个contig里的node都有相似的k-compatibility class。为了确保跳过的是k-equivalence class,kallisto会检查跳过的最后一个node是否符合预期。正因为省掉这些冗余的node，比对速度快。如上图只需要取三个node的k-compatibility class的交集就能确定read的k-compatibility。</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020133612277.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020133612277.png, image-20211020133612277.png 1.5x, /posts/rnaseq-quantification/image-20211020133612277.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020133612277.png"
        title="image-20211020133612277" /></p>
<p>确定好要比对的node后使用expectation_maximization (EM) 的算法来定量转录本</p>
<h4 id="kallisto-比对和定量结果分析">kallisto 比对和定量结果分析</h4>
<p>kallisto比对分析的准确度：
Kallisto与 Bowtie2比对分析中，66.22%reads比对到了相同转录本上
对于比对到不同转录本上的reads（ambiguous reads），Bowtie2的平均比对得分 8.94，Kallisto是4.86
由于Kallisto读取的碱基少总比对率 90.8%，而Bowtie2的比对率为86.5%
Kallisto的比对运行时间短</p>
<p>kallisto定量分析的优势：
Kallisto采用EM算法，Cufflinks没有使用EM算法导致其multi-map遍布整个基因组
Kallisto的伪对齐保留了reads的k-mers信息，Sailfish将reads切割成k-mers，导致信息丢失
Kallisto的定量运行时间短</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020134030223.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020134030223.png, image-20211020134030223.png 1.5x, /posts/rnaseq-quantification/image-20211020134030223.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020134030223.png"
        title="image-20211020134030223" /></p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020134041712.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020134041712.png, image-20211020134041712.png 1.5x, /posts/rnaseq-quantification/image-20211020134041712.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020134041712.png"
        title="image-20211020134041712" /></p>
<p>median relative difference为不同定量软件的readcount值与GEUVADIS数据中心提供的20个模型的所有转录本的（RSEM）readcount之间的绝对差异值除以两者的平均值的均值</p>
<h4 id="kallisto-的使用方法">kallisto 的使用方法</h4>
<ol>
<li>
<p>对转录本序列建立索引</p>
<p>kallisto index -k 31 -i hg19.idx  hg19.refMrna.fa</p>
<p>-k参数指定kmer的长度
-i参数指定输出的索引的名字</p>
</li>
<li>
<p>定量</p>
<p>kallisto quant<br>
-i hg19.idx <br>
-o out_dir <br>
-t  20 \ 
R1.fastq.gz R2.fastq.gz</p>
<p>-i参数指定转录本的索引文件
-o参数指定输出结果的目录
-t参数指定线程数
对于单端数据而言，必须指定fragment长度的均值和方差，分别对应-l和-s参数
添加参数&ndash;genomebam &ndash;gtf ${gtf_file}定量完成后会在结果目录多生成一个bam文件，可以用igv查看</p>
</li>
</ol>
<h4 id="kallisto-的输出文件">kallisto 的输出文件</h4>
<p>在输出目录，会生成以下3个文件</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020134323562.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020134323562.png, image-20211020134323562.png 1.5x, /posts/rnaseq-quantification/image-20211020134323562.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020134323562.png"
        title="image-20211020134323562" /></p>
<p>run_info.json 文件为JSON格式，保存了运行的命令和参数。
前缀为abundance 的文件，保存了转录本的定量信息
后缀为h5为HDF5格式的文件，当转录本数量较多时，相比纯文本，这种格式的文件大小会小很多
对于HDF5的文件，可以采用如下命令转换为tsv格式的文件
kallisto h5dump  -o out_dir  abundance.h5
abundance.tsv中包含转录本名称，长度,count值和tpm,格式如下：</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020134347891.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020134347891.png, image-20211020134347891.png 1.5x, /posts/rnaseq-quantification/image-20211020134347891.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020134347891.png"
        title="image-20211020134347891" /></p>
<h4 id="kallisto-的总结">kallisto 的总结</h4>
<p>kallisto软件本身的特点就是对设备要求不高且速度快，但是其定量结果的准确性比较有争议</p>
<p>kallisto进行定量时，有专门的软件sleuth进行差异表达分析（Pimentel, H.2017）</p>
<p>[1] Pimentel, H., Bray, N. L., Puente, S., Melsted, P., &amp; Pachter, L. (2017). <em>Differential analysis of RNA-seq incorporating quantification uncertainty. Nature Methods, 14(7), 687–690.</em> doi:10.1038/nmeth.4324</p>
<h3 id="salmon定量软件简介">salmon定量软件简介</h3>
<h4 id="salmon官网">salmon官网</h4>
<p><a href="https://combine-lab.github.io/salmon/">https://combine-lab.github.io/salmon/</a></p>
<h4 id="salmon简介">salmon简介</h4>
<p>Salmon 是2017年Nature Methods发表的一篇<font color='red'>无需比对</font>的定量软件（Patro, R 2017）
将新的<font color='red'>双相并行推理算法</font>和功能丰富的<font color='red'>偏差模型</font>与超快速读长映射过程结合在一起
它与Kallisto(熊神星)和Sailfish(旗鱼)类似，不基于比对而直接对基因进行定量的工具，适用于转录组、宏基因组等的分析
salmon优势是：
<font color='red'>定量时考虑到不同样品中基因长度的改变</font>（比如不同isoform的使用）
速度快、需要的计算资源和存储资源小
敏感性高，不会丢弃匹配到多个基因同源区域的reads
可以直接校正GC-bias
自动判断文库类型</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020143247736.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020143247736.png, image-20211020143247736.png 1.5x, /posts/rnaseq-quantification/image-20211020143247736.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020143247736.png"
        title="image-20211020143247736" /></p>
<p>[1] Patro, R., Duggal, G., Love, M. I., Irizarry, R. A., &amp; Kingsford, C. (2017). Salmon provides fast and bias-aware quantification of transcript expression. Nature Methods.</p>
<h4 id="salmon原理">salmon原理</h4>
<p>salmon quasi-mapping的比对原理是基于k-mers+Suffix Array</p>
<p>第一步：建立索引</p>
<p>如果有多个transcript，那将多个transcripts连起来用$分割
基于k-mer建立hash table</p>
<p>第二步：比对</p>
<p>Suffix Array(T) ：建立Transcriptome的后缀数组
SA interval ：就是该kmer所对应的后缀数组区间。
计算MMP（Maximum Mappable Prefix）最大映射前缀。就是图中蓝色虚线圈出来的部分。
通过LCP(longest common prefix)最长公共前缀。如图中红色圈出来的部分
可能会出现LCP(T[SA[b’] ] ,T[SA[e’−1] ] )) |&gt;|MMPi|的情况，就是LCP比MMP长。也就是说，在几条transcript里，都是ATTGACTA，而read是ATTGAGTA。则认为这不是虚假的匹配，而是测序过程中出现了错误。
NIP(next informative position)下一个信息位置，与kallisto相似跳过相同k-mer
最后会生成一个查询的集合，根据kallisto获得的灵感，对每一个命中的区间取交集。但这里与kallisto不同的是，这里取交集的时候分正向反向，kallisto是不区分的</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020135357090.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020135357090.png, image-20211020135357090.png 1.5x, /posts/rnaseq-quantification/image-20211020135357090.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020135357090.png"
        title="image-20211020135357090" /></p>
<h4 id="salmon的定量优势">Salmon的定量优势</h4>
<p>准确度指标中所述的平均绝对相对差异（mean absolute relative differences，MARD）Salmon和Kallisto产生的MARD相似，尽管Salmon的MARD分布比Kallisto的小得多（Mann-Whitney U检验，p = 0.00017）。两种方法都优于eXpress（Mann-Whitney U检验，p = 3.39781×10-8）。
在典型的FDR值下，使用Salmon的估计找到真正的差异转录本的敏感性比使用Kallisto的估计要高53％–450％，比使用eXpress的Polyester模拟数据的估计要高210％-250％
Salmon的结果可以由R包转换后直接用edgeR/DESeq2等进行counts值的下游分析</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020135459185.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020135459185.png, image-20211020135459185.png 1.5x, /posts/rnaseq-quantification/image-20211020135459185.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020135459185.png"
        title="image-20211020135459185" /></p>
<h4 id="salmon的使用方法">Salmon的使用方法</h4>
<ol>
<li>构建索引
salmon index -t  all_transcripts.fa   -i all_transcripts_index</li>
</ol>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020135541711.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020135541711.png, image-20211020135541711.png 1.5x, /posts/rnaseq-quantification/image-20211020135541711.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020135541711.png"
        title="image-20211020135541711" /></p>
<ol start="2">
<li>定量分析
salmon quant<br>
&ndash;gcBias
-l A 
-1  RASshCON.R1.fastq.gz <br>
-2  RASshCON.R2.fastq.gz  <br>
-i   all_transcripts_index <br>
-g  all_transcripts.gtf<br>
-o  RASshCON.count 
-p 10</li>
</ol>
<p>&ndash;gcBias: 校正测序片段GC含量
&ndash;seqBias：矫正测序偏差
-l: A自动判断文库类型
-g  提供转录本对应gene的信息，如果提供，则输出quant.sf and quant.genes.sf 两份结果</p>
<p>高级参数介绍</p>
<p>&ndash;writeMappings  将mapping结果以SAM格式输出</p>
<p>&ndash;validateMappings 启用选择性比对到转录本。能够提高比对的灵敏度和特异性从而提高了定量的准确度。</p>
<p>&ndash;mimicBT2 和 &ndash;mimicStrictBT2 让比对模仿一些 Bowtie2 行为。</p>
<p>&ndash;recoverOrphans 这个参数应与 &ndash;validateMappings 一起用。如果只有一端 reads 能比对或者两端 reads 比对不在同一个转录本，Salmon 会在比对上位置的上下游寻找另一端 reads 的比对。</p>
<p>&ndash;hardFilter 这个参数仅在 &ndash;validateMappings 模式使用，关闭软过滤(soft filter)。</p>
<p>&ndash;allowDovetail 默认丢弃“燕尾比对”(Dovetailing mapping)，开启这参数后不丢弃。“燕尾比对”见下面 Bowtie 的示意。</p>
<p>-p/&ndash;threads 线程数，默认会使用所有线程，想要用少些线程就设置。</p>
<p>&ndash;incompatPrior 与文库类型(library type)不兼容的比对先验概率。Salmon 默认设置了一个非常小但非0数值，如果一个插入片段(fragment)的唯一比对是这种非兼容比对，Salmon 会把这个比对计数。如果希望不计入，将这个参数设置数值 0.0 即可。</p>
<p>&ndash;minScoreFraction 最低比对得分限制，低于此比对被丢弃。</p>
<p>&ndash;rangeFactorizationBins 提高对“困难”转录本的定量准度。配合比对模式的 &ndash;useErrorModel 参数或者 fastq 模式的 &ndash;validateMappings 参数使用。</p>
<p>&ndash;seqBias 从输入数据学习然后矫正测序偏差。</p>
<p>&ndash;gcBias 进行 GC 偏差矫正。建议根据质控的 GC 偏差情况设置。</p>
<p>-g/&ndash;geneMap 提供基因注释文件 Salmon 就额外生成 quant.genes.sf 文件给出基因的定量</p>
<h4 id="salmon输出结果">salmon输出结果</h4>
<p>quant.sf文件，其记录了每个转录本的count数以及对应的TPM值
ambig_info.tsv文件可以看每个转录本对应的Unique and ambiguous count
meta_info.json 参与计算的reads的信息如mapping上的总reads数</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020140156587.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020140156587.png, image-20211020140156587.png 1.5x, /posts/rnaseq-quantification/image-20211020140156587.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020140156587.png"
        title="image-20211020140156587" /></p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020140220593.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020140220593.png, image-20211020140220593.png 1.5x, /posts/rnaseq-quantification/image-20211020140220593.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020140220593.png"
        title="image-20211020140220593" /></p>
<p>name:转录本id
length:转录本长度
effectivelength:考虑了基因长度分布和序列特异性和 GC片段偏差
tpm:每千个碱基的转录每百万映射读取的Transcripts
numreads:  readcount值</p>
<h4 id="salmon-的总结">salmon 的总结</h4>
<p>salmon 和 kallisto 都是不基于比对而直接对基因进行定量的工具</p>
<p>salmon 相比 kallisto，具有丰富的偏差模型可以矫正测序GC-bias和低质量reads的影响，有丰富的参数控制，参数解释说明见网站（https://salmon.readthedocs.io/en/latest/salmon.html）</p>
<p>kallisto官方推荐的适用基因组是小麦，建议小麦使用kallisto</p>
<h3 id="rsemrnaseq-by-expectation-maximization定量软件简介">RSEM(RNASeq by Expectation Maximization)定量软件简介</h3>
<h4 id="rsem官网">RSEM官网</h4>
<p><a href="http://deweylab.biostat.wisc.edu/rsem/">http://deweylab.biostat.wisc.edu/rsem/</a></p>
<h4 id="rsem简介">RSEM简介</h4>
<p>2011年问世的老牌比对/定量/差异分析软件（Li, B 2011）
trinity官方推荐的转录本定量软件</p>
<p>rsem的主要参数：</p>
<ol>
<li>prepare-reference   构建转录本索引</li>
<li>calculate-expression   比对定量（可以输入bam也可以输入reads）</li>
<li>Differential Expression Analysis using EBSeq   差异分析
3.1 generate-data-matrix    构建数据矩阵
3.2 run-ebseq     调用EBseq进行检验
3.3 control_fdr    控制FDR(错误发现率)来挑选差异基因</li>
</ol>
<p>特点
<font color='red'>EM算法准确处理ambiguous reads</font>
不依赖参考基因组文件</p>
<p>Li, B., &amp; Dewey, C. N. (2011). <em>RSEM: accurate transcript quantification from RNA-Seq data with or without a reference genome. BMC Bioinformatics, 12(1), 323.</em> doi:10.1186/1471-2105-12-323</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020143321991.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020143321991.png, image-20211020143321991.png 1.5x, /posts/rnaseq-quantification/image-20211020143321991.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020143321991.png"
        title="image-20211020143321991" /></p>
<p>RSEM采用<font color='red'>转录本水平的比对</font></p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020141029768.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020141029768.png, image-20211020141029768.png 1.5x, /posts/rnaseq-quantification/image-20211020141029768.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020141029768.png"
        title="image-20211020141029768" /></p>
<p>1 QNAME 序列的名字（Reads id）2 flag 标记（81=1+16+64）3 RNAME   参考序列染色体的名字4 POS     在参考序列上的位置（染色体上的位置）5 MAPQ    mapping qulity 越高则位点特异性越好 6 CIGAR   代表比对结果的CIGAR字符串7 RNEXT   双末端测序中另一个reads比对的参考序列的名称，如果没有则用 “ * ” 表示，如果和前一个reads比对到同一个参考序列则用“ = ”表示 8  PNEXT   双末端测序中另一个reads比对到参考序列上的位置，如果没有则用0表示 9 TLEN    估计出的query模板序列的长度，当mate序列位于本序列上游时该值为负值。Template的长度，最左边得为正，最右边的为负，中间的不用定义正负，不分区段（single-segment)的比对上，或者不可用时，此处为0；10 SEQ    read的序列;序列片段的序列信息 11 QUAL   ASCII码格式的序列质量;序列的质量信息 12 可选的字段（field）</p>
<h4 id="rsem的比对定量优势">RSEM的比对定量优势</h4>
<p>RSEM对于基因组水平的比对的优势：
可变剪接和多聚腺苷酸化（ polyA）干扰基因组水平的比对
对于没有基因组的物种，转录本水平的比对是一种方法
转录本水平的比对参与的碱基少，所以快
综合考虑了测序方式（PE/SE）,测序片段长度分布和reads质量值的影响</p>
<p>RSEM的定量优势：
RSEM采用EM算法模型，相对于只计算比对到单一位置reads的Alexa-seq算法，似然函数更适合处理那些ambiguous reads
95%可信区间算法(CI)和后验均值算法(PME)以及最大似然法算法(ML)被开发应用于定量基因（转录本）
RSEM调用bowtie根据reads的具体比对情况来定量reads,不是仅判断reads的比较率高低，比如rsem在默认情况下只允许reads起始25bp有2个错配
RSEM的结果可以通过probabilistically-weighted可视化reads的深度</p>
<h4 id="rsem原理">RSEM原理</h4>
<p>1.调用bowtie寻找每个read的所有比对情况
2.EM算法找到抽取样本中θ的最大可能性（Maximum likelihood）
3.根据θ的值计算所有reads的表达量情况</p>
<p>RSEM的EM算法：</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020143212592.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020143212592.png, image-20211020143212592.png 1.5x, /posts/rnaseq-quantification/image-20211020143212592.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020143212592.png"
        title="image-20211020143212592" /></p>
<p>RSEM数学模型由n个随机变量控制
Gn/Fn/Sn/On分别代表转录本来源/长度/起始位置和方向
对于PE data,变量为L1n/L2n和Q1n/Q2n和R1n/R2n,分别代表read长度/质量分数/序列
对于SE数据，L2n,Q2n和R2n是不考虑的
这个模型的主要参数是由θ给出，它代表一个片段来自每个转录本的先验概率</p>
<p>先验概率：事情还没有发生，要求这件事情发生的可能性大小
最大似然法：当从模型总体随机抽取n组样本观测值后，最合理的参数估计量应该使得从中抽取该n组样品观测值的概率最大</p>
<h4 id="rsem结果评估">RSEM结果评估</h4>
<p>错误率：采用MPE/EF/FP得分来估计不同软件的定量结果和ensemble提供的定量结果之间的差异度</p>
<p>Cufflinks在处理短转录本序列（小于280bp）时表达量和ensemble结果比相比，异常高，不适易用于定量短转录本</p>
<p>RSEM和IsoEM都采用EM算法，结果相似，RSEM优化了poly(A) 的处理和CI/PME算法矫正，所以结果更准一些</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020143403308.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020143403308.png, image-20211020143403308.png 1.5x, /posts/rnaseq-quantification/image-20211020143403308.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020143403308.png"
        title="image-20211020143403308" /></p>
<p>AB双端测序  CD单端测序  AC转录本定量  BD基因定量
x轴：错误率阈值
y轴：大于错误阈值的转录本/gene的比例</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020143535090.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020143535090.png, image-20211020143535090.png 1.5x, /posts/rnaseq-quantification/image-20211020143535090.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020143535090.png"
        title="image-20211020143535090" /></p>
<p>上部 是reads的实际丰度分布
底部是比对到基因组唯一位置的reads的丰度分布</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020143618360.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020143618360.png, image-20211020143618360.png 1.5x, /posts/rnaseq-quantification/image-20211020143618360.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020143618360.png"
        title="image-20211020143618360" /></p>
<p>Rsem比对定量内存和时间占用情况</p>
<h4 id="rsem的使用方法">RSEM的使用方法</h4>
<p>1.构建索引</p>
<p>rsem-prepare-reference &ndash;gtf genome.gtf genome.fa reference_name -p 8</p>
<p>&ndash;gtf genome.gtf：输入基因组GTF注释文件
genome.fa：基因组文件
reference_name：索引名称
-p：线程数</p>
<p>2.定量
STAR/bowtie—RSEM
reads&mdash;RSEM</p>
<p>rsem-calculate-expression RASshETV4_2.clean.1.fastq.gz  RASshETV4_2.clean.2.fastq.gz  all_transcripts   RASshETV4_2    &ndash;paired-end &ndash;bowtie2 -p 8</p>
<p>&ndash;paired-end：表示输入的数据为双端测序数据
-no-bam-output：不输出BAM文件
&ndash;alignments：输入文件为BAM文件
-p：线程数</p>
<h4 id="rsem输出结果">RSEM输出结果</h4>
<p>转录本定量的bam
转录本/gene的定量结果</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020144432129.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020144432129.png, image-20211020144432129.png 1.5x, /posts/rnaseq-quantification/image-20211020144432129.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020144432129.png"
        title="image-20211020144432129" /></p>
<p>isoforms.results中包含了转录本ID，基因ID，转录本长度，有效长度，expected_count，TPM，FPKM和IsoPct（该转录本表达量占基因总表达量的百分比）
genes.results中的内容与之类似，只是少了IsoPct</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020144444350.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020144444350.png, image-20211020144444350.png 1.5x, /posts/rnaseq-quantification/image-20211020144444350.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020144444350.png"
        title="image-20211020144444350" /></p>
<h4 id="基因标准化常用方法">基因标准化常用方法</h4>
<p>RPKM（Reads Per Kilobase Million）：每百万reads中来自于某基因每千碱基长度的reads数
FPKM（Fragments Per Kilobase Million）：每百万 fragments 中来自某一基因每千碱基长度的 fragments 数目</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020144609091.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020144609091.png, image-20211020144609091.png 1.5x, /posts/rnaseq-quantification/image-20211020144609091.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020144609091.png"
        title="image-20211020144609091" /></p>
<p>TPM（Transcripts Per Kilobase of exon model per Million mapped reads ）每千个碱基的转录每百万映射读取的Transcripts</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020144736012.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020144736012.png, image-20211020144736012.png 1.5x, /posts/rnaseq-quantification/image-20211020144736012.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020144736012.png"
        title="image-20211020144736012" /></p>
<p>nr：mapping到目标基因上的read数（readcount值）；
Lr：目标基因的外显子长度的总和（转录本长度）</p>
<p>特点：可以用于同一物种不同组织间的比较，因为Sum(TMPs)的值总是唯一的</p>
<h4 id="rsem总结">RSEM总结</h4>
<p>RSEM是准确转录本定量的首选</p>
<p>转录本比对定量和基因比对定量的突出优缺点：
优点：可变剪接和多聚腺苷酸化（ polyA）干扰基因组水平的比对
缺点：转录本相对于基因长度短，导致ambiguous reads多，干扰准确性（Charlotte，2015）</p>
<p>[1] Charlotte, Soneson, Michael, I, Love, &amp; Mark, et al. (2015). Differential analyses for rna-seq: transcript-level estimates improve gene-level inferences. <em>F1000research</em>.</p>
<h3 id="stringtie定量软件简介">StringTie定量软件简介</h3>
<h4 id="stringtie官网">stringtie官网</h4>
<p><a href="https://ccb.jhu.edu/software/stringtie/index.shtml">https://ccb.jhu.edu/software/stringtie/index.shtml</a></p>
<h4 id="stringtie简介">stringtie简介</h4>
<p>Stringtie应用了起源于<font color='red'>最优化理论的网络流</font>算法与从头组装策略一起来将这些短读段组装成转录本（cufflinks的升级版）（Pertea, M. 2015）</p>
<p>作用：
转录本组装/定量</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020150538879.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020150538879.png, image-20211020150538879.png 1.5x, /posts/rnaseq-quantification/image-20211020150538879.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020150538879.png"
        title="image-20211020150538879" /></p>
<p>Pertea, M., Pertea, G. M., Antonescu, C. M., Chang, T.-C., Mendell, J. T., &amp; Salzberg, S. L. (2015). <em>StringTie enables improved reconstruction of a transcriptome from RNA-seq reads. Nature Biotechnology, 33(3), 290–295.</em> doi:10.1038/nbt.3122</p>
<h4 id="stringtie原理和优势">StringTie原理和优势</h4>
<p>人类基因组中90%的多外显子蛋白编码基因和30%的ncRNA都具有可变剪切体
其他组装软件有专注于转录本确定的Trinity、专注于表达定量的RSEM以及二者兼顾的Cufflink等
cufflinks用的是overlap graph，该模型中nodes代表fragment，如果两个fragment存在overlap并存在兼容的剪切模式，则对应的node连接起来。他利用吝啬法利用read来产生最少的转录本，尽管效果还行，但是异构体过多时，效果会变差，还有就是它没有考虑到转录本的丰度并且对于某些isoform来说该方法没有办法组装</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020150705255.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020150705255.png, image-20211020150705255.png 1.5x, /posts/rnaseq-quantification/image-20211020150705255.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020150705255.png"
        title="image-20211020150705255" /><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020150716422.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020150716422.png, image-20211020150716422.png 1.5x, /posts/rnaseq-quantification/image-20211020150716422.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020150716422.png"
        title="image-20211020150716422" /></p>
<p>Stringtie通过使用genome指导的组装的方法与从头组装的概念结合<font color='red'>（String+SR）</font>的新方法来改善转录组组装，利用了read 和super read 两种数据
stringtie采用了组装转录本和估计表达量同步进行的方法，这不同于cufflinks的先组装后定量的策略
首先，stringtie将read聚成cluster，然后采用了splice graph，其中node代表外显子或外显子的一部分path，将graph中可能的剪切现象都连起来，最终对每个转录本通过创建一个网络流的方法，利用最大流算法（maximum flow algorithm）估计转录本的表达量
与其它组装软件相比，stringtie具有很高的准确性和新型isoform的发现能力，其优势在于使用了网络流算法，同时stringtie也支持将read从头组装成更长的片段，这进一步提高了其组装的正确性
其另一个优势在于它的最优化策略，<font color='red'>它平衡了每次组装中每条转录本的覆盖度</font>，这样可以对组装算法产生一定的限制，因为在组装基因组时，覆盖度是很重要的一个参数因为它需要被用来限制算法，否则组装器可能将重复的片段错误地堆叠到一起，相似地转录组装也是如此，在isoform中的每一个外显子需要有相似的覆盖度，如果忽略这个参数可能会产生一些保守但是错误的转录本，其中含有大量剪切位点的基因组装起来尤其困难</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020150958523.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020150958523.png, image-20211020150958523.png 1.5x, /posts/rnaseq-quantification/image-20211020150958523.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020150958523.png"
        title="image-20211020150958523" /></p>
<p>相对其他转绿本组装软件比如Cufflinks，IsoLasso，Scripture，Traph等，StringTie的敏感性和阳性检测率往往更准确，更加完整，同时对于基因表水平的评估也更为准确。
举例，对于人血液中90百万的read数据，它成功的组装出10990个正确的转录本，而cufflinks（排第二）只有区区7187个，相对而言有53%的提升。对于模拟数据，组装出7559相对cufflinks 6310 有20%的提升。另外在处理同一批数据的时候，StringTie相对其他软件要快的多</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020151025657.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020151025657.png, image-20211020151025657.png 1.5x, /posts/rnaseq-quantification/image-20211020151025657.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020151025657.png"
        title="image-20211020151025657" /></p>
<p>在表达定量方面，评估了真实表达值和各个软件估计表达值的情况，cufflinks 相似性最高，但是StringTie 效果也可以，可以说定量准确水平旗鼓相当
定量时间复杂度和空间复杂度上，StringTie处理模拟数据是耗时31分钟，而cufflinks 用时81分钟，在真实数据上，耗时在35-76分钟，速度大概是其他软件的3倍左右。在内存使用方面，处理真实数据时，大约利用了1.6~12g，cufflinks等软件利用了6.4~26.6g</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020151125384.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020151125384.png, image-20211020151125384.png 1.5x, /posts/rnaseq-quantification/image-20211020151125384.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020151125384.png"
        title="image-20211020151125384" /></p>
<h4 id="stingtie定量方法">StingTie定量方法</h4>
<p>stringtie -p 10 -G all_transcripts.gtf -o output.gtf  -b  RASshCON  -e RASshCON.bam</p>
<p>-G参数指定参考基因组的gtf文件</p>
<p>-o指定输出的文件，格式也为gtf</p>
<p>-b指定ballgown的输出结果目录，这个参数是为了方便下游进行ballgown差异分析</p>
<p>-e参数要求软件只输出已知转录本的定量结果</p>
<h4 id="stingtie输出文件">StingTie输出文件</h4>
<p>stringtie的定量结果输出在一个gtf格式的文件中，如下：</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020151422095.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020151422095.png, image-20211020151422095.png 1.5x, /posts/rnaseq-quantification/image-20211020151422095.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020151422095.png"
        title="image-20211020151422095" /></p>
<p>seqname:该转录本所在的染色体号，contig或scaffold;
source:GTF文件来源；
feature:特征类型，例：外显子，转录本，mRNA和5’UTR；
start:起始位置；
end:终止位置；
score: 组装转录本的可信度打分；
strand:转录本所在的正负链信息；
frame:CDS特征，StringTie不使用该信息，所以其结果用”.”表示；
attributes:该特征的属性，包括基因id，转录本id，外显子个数，read coverage，FPKM和TPM等
官网提供了计算count值的脚本：prepDE.py</p>
<h4 id="stingtie总结">StingTie总结</h4>
<p>优点：genome组装准确性远超Cufflinks等常用软件
缺点：定量结果不直接提供count值</p>
<h3 id="htseq-high-throughput-sequencing定量软件简介">HTSeq (high-throughput sequencing)定量软件简介</h3>
<h4 id="htseq官网">HTSeq官网</h4>
<p><a href="https://htseq.readthedocs.io/en/master/count.html">https://htseq.readthedocs.io/en/master/count.html</a></p>
<h4 id="htseq简介">HTSeq简介</h4>
<p>特征：采用python语言开发的基因定量软件（Simon, A, 2015）</p>
<p>HTSeq是转录组定量分析的软件，其输入文件必须有bam（sorted）和GTF文件。</p>
<p>一般情况下HTSeq得到的Counts结果会用于下一步不同样品间的基因表达量差异分析，而不是一个样品内部基因的表达量比较。因此，HTSeq设置了-a参数的默认值10，来忽略掉比对到多个位置的reads信息，其结果有利于后续的差异分析。</p>
<p>输入的GTF文件中不能包含可变剪接信息，否则HTSeq会认为每个可变剪接都是单独的基因，导致能比对到多个可变剪接转录本上的reads的计算结果是ambiguous，从而不能计算到基因的count中。即使设置-i参数的值为transcript_id，其结果一样是不准确的，只是得到transcripts的表达量。</p>
<p>[1] Simon, A. , Theodor, P. P. , &amp; Wolfgang, H. . (2015). Htseq—a python framework to work with high-throughput sequencing data. <em>Bioinformatics</em>(2), 166-9.</p>
<h4 id="htseq原理">HTSeq原理</h4>
<p>由于序列读长的限制和基因组的同源性，一条reads可能比对到多个基因上，而且基因之间也存在overlap, 在对这些特殊情况进行处理时，htseq-count内置了以下3种模式</p>
<p>union</p>
<p>interp-strict</p>
<p>interp-nonempty</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020152650974.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020152650974.png, image-20211020152650974.png 1.5x, /posts/rnaseq-quantification/image-20211020152650974.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020152650974.png"
        title="image-20211020152650974" /></p>
<p>在BAM文件，包含了比对上的reads和没有比对上的reads, 只有比对上的reads 会用来计数，htseq-count默认会根据mapping的质量值对BAM文件进行过滤，默认值为10, 意味着只有mapping quality &gt; 10的reads才会用来计数，当然可以通过-a参数来修改这个阈值。</p>
<p>能够明确reads属于一个featurer时，比如示意图种的第一种情况，reads完全是gene_A的一个片段，该feature的计数就加1；能明确reads不属于一个feature时，称之为no_feature, 比如示意图种的第二种情况，reads的一部分比对上了gene_A, 在interp_strict模式下，判定该reads不属于gene_A, 就是no_feature。</p>
<p>当不明确一条reads是否属于某个feature时，通常是由于reads在两个gene的overlap区，比如示意图中的第六和第七种情况，这样的reads被标记为ambigous。</p>
<p>当一条reads比对上了两个feature时，会被标记为alignment_not_unique。</p>
<p>在统计属于某个基因的reads数时，需要重点关注对 ambiguous 和 alignment_not_unique 的reads的处理， 通过&ndash;nonunique参数来指定，取值有以下两种</p>
<p>none</p>
<p>all</p>
<p>默认值为none时，这两种reads被忽略掉，不进行任何的计数；取值为all时，对应的所有feature的计数都会加1。</p>
<p>除了&ndash;mode和&ndash;nonunique两个参数外，还需要关注&ndash;stranded参数，这个参数指定文库的类型，默认值为yes, 代表文库为链特异性文库，no代表为非链特异性文库。对于非链特异性文库文库，在判断一条reads是否属于一个基因时，只需要关注比对位置，而链特异性文库还需要关注比对的正负链和基因的正负链是否一致，只有一致时，才会计数。</p>
<h4 id="htseq定量方法">HTSeq定量方法</h4>
<p>HTSeq是对有参考基因组的转录组测序数据进行表达量分析的，其输入文件必须有SAM/bam和GTF文件
hisat2比对软件得到的bam文件，sort by pos, 所以需要重新sort，命令如下：</p>
<p>samtools sort -n yourfile.bam &gt; yourfile_name.bam</p>
<p>定量命令：
python3 -m HTSeq.scripts.count -f bam  -s no  -t exon -i transcript_id -m intersection-nonempty  RASshCON.sort.bam all_transcripts.gtf  &gt; RASshCON.count</p>
<p>-f | &ndash;format default: sam 设置输入文件的格式，该参数的值可以是sam或bam。
-r | &ndash;order default: name 设置sam或bam文件的排序方式，该参数的值可以是name或pos。
-s | &ndash;stranded default: yes 设置是否是链特异性测序。该参数的值可以是yes,no或reverse。
-a | &ndash;a default: 10 忽略比对质量低于此值的比对结果。在0.5.4版本以前该参数默认值是0。
-t | &ndash;type default: exon 程序会对该指定的feature（gtf/gff文件第三列）进行表达量计算，而gtf/gff文件中其它的feature都会被忽略。
-i | &ndash;idattr default: gene_id 设置feature ID是由gtf/gff文件第9列那个标签决定的；若gtf/gff文件多行具有相同的feature ID，则它们来自同一个feature，程序会计算这些features的表达量之和赋给相应的feature ID。
-m | &ndash;mode default: union 设置表达量计算模式。该参数的值可以有union, intersection-strict and intersection-nonempty
-o | &ndash;samout 输出一个sam文件，该sam文件的比对结果中多了一个XF标签，表示该read比对到了某个feature上。
-q | &ndash;quiet 不输出程序运行的状态信息和警告信息。</p>
<h4 id="htseq输出结果">HTSeq输出结果</h4>
<p>输出的count里就2列，第一列是geneid第二列是count</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020153531706.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020153531706.png, image-20211020153531706.png 1.5x, /posts/rnaseq-quantification/image-20211020153531706.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020153531706.png"
        title="image-20211020153531706" /></p>
<p>-o 参数输出一个sam文件，该sam文件的比对结果中多了一个XF标签，表示该read比对到了某个feature上的具体特征，如下：</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020153602190.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020153602190.png, image-20211020153602190.png 1.5x, /posts/rnaseq-quantification/image-20211020153602190.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020153602190.png"
        title="image-20211020153602190" /></p>
<h4 id="htseq总结">HTSeq总结</h4>
<p>是一个非常经典的gene-level定量软件</p>
<p>运行速度和内存占用情况不如featurecount</p>
<h3 id="featurecountssubread-package定量软件简介">featurecounts（Subread package）定量软件简介</h3>
<h4 id="featurecounts官网">featurecounts官网</h4>
<p><a href="http://subread.sourceforge.net/">http://subread.sourceforge.net/</a></p>
<p><a href="http://bioinf.wehi.edu.au/featureCounts/">http://bioinf.wehi.edu.au/featureCounts/</a></p>
<h4 id="featurecounts简介">featurecounts简介</h4>
<p>一个C语言编写的基因定量软件（Y Liao 2013）
特点：应用了高效率的染色体哈希算法和feature区块技术
突出优点：快</p>
<p>featurecounts属于subread
subread是一个能快速进行序列比对/定量/SNP分析的工具</p>
<p>[1] Y Liao, Smyth, G. K. , &amp; Shi, W. . (2013). Featurecounts: an efficient general-purpose program for assigning sequence reads to genomic features.</p>
<h4 id="featurecounts原理">featurecounts原理</h4>
<p>每条染色体分为128 kb 的bins，根据feature（染色体下的实线如exon）的起始位置，将其分配到各自对应的bin中, 然后这些feature后又被平均分为不同的block（灰色框),尽量保证一个block中的feature的数目与bin中的block的数目近似相等</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020154220063.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020154220063.png, image-20211020154220063.png 1.5x, /posts/rnaseq-quantification/image-20211020154220063.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020154220063.png"
        title="image-20211020154220063" /></p>
<p>Reads先与不同的基因组bin进行比对，定位到其bin中后，再与不同的特征block（虚线箭头）进行对比，最后与feature进行比较（实线箭头）.在maping过程中，featurecount只使用这一套bin和block,  在最后的统计过程中，featurecount会根据不同的参数要求来统计是否去除multimapping reads.
图中，该reads与bin I 的第一个特征块中的两个特征区域重叠。</p>
<h4 id="featurecounts结果评估">featurecounts结果评估</h4>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020154302210.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020154302210.png, image-20211020154302210.png 1.5x, /posts/rnaseq-quantification/image-20211020154302210.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020154302210.png"
        title="image-20211020154302210" /><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020154310611.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020154310611.png, image-20211020154310611.png 1.5x, /posts/rnaseq-quantification/image-20211020154310611.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020154310611.png"
        title="image-20211020154310611" /></p>
<p>htseq-count 把gff文件中的特征区域的终止位置的碱基不算在该特征区域内
featureCounts 和 summarizeOverlaps 软件则把终止位置的碱基统计在特征区域内
桑格研究所规定gff文件中特征区域的起始和终止碱基是包含在特征区域内的，所以featureCounts 比htseq的定量结果更准确一些</p>
<p>双端测序时，summorizeOverlops软件统计到的fragments数比其他两个软件少的原因主要是因为summarizeOverlaps要求双端测序的两个end reads 都成功的mapped 后才会进行分配到基因, 而 featureCounts and htseq-count 只要有一端reads 成功的mapped后，其就可以将它分配到基因上</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020154533143.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020154533143.png, image-20211020154533143.png 1.5x, /posts/rnaseq-quantification/image-20211020154533143.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020154533143.png"
        title="image-20211020154533143" /></p>
<p>如果某一fragment能够同时匹配上两个基因，但是其中一个基因（a)可以被这一fragment的两端reads匹配上，而另一个基因(b)只被一个end read 匹配上，那么featurecount就会认为该fragment来自a基因, 而htseq-count遇到这样的问题时，就会认为这个fragment发生了multmaped，从而舍弃此基因</p>
<h4 id="featurecounts使用方法">featurecounts使用方法</h4>
<p>/mnt/ilustre/users/wanyun.yang/script/quant/featurecount/subread-1.6.0-Linux-x86_64/bin/featureCounts -p -t exon -g gene_id -a all_transcripts.gtf -o RASshCON.count.txt  RASshCON.bam</p>
<p>主要的参数：</p>
<table>
<thead>
<tr>
<th>input file</th>
<th>输入的bam/sam文件，支持多个文件输入</th>
</tr>
</thead>
<tbody>
<tr>
<td>-a &lt; string &gt;</td>
<td>参考gtf文件名，支持Gzipped文件格式</td>
</tr>
<tr>
<td>-F</td>
<td>参考文件的格式，一般为GTF/SAF，C语言版本默认的格式为GTF格式</td>
</tr>
<tr>
<td>-A</td>
<td>提供一个逗号分割为两列的文件，一列为gtf中的染色体名，另一列为read中对应的染色体名，用于将gtf和read中的名称进行统一匹配，注意该文件提交时不需要列名</td>
</tr>
<tr>
<td>-J</td>
<td>对可变剪切进行计数</td>
</tr>
<tr>
<td>-G &lt; string &gt;</td>
<td>当-J设置的时候，通过-G提供一个比对的时候使用的参考基因组文件，辅助寻找可变剪切</td>
</tr>
<tr>
<td>-M</td>
<td>如果设置-M，多重map的read将会被统计到</td>
</tr>
<tr>
<td>-o &lt; string &gt;</td>
<td>输出文件的名字，输出文件的内容为read 的统计数目</td>
</tr>
<tr>
<td>-O</td>
<td>允许多重比对，即当一个read比对到多个feature或多个metafeature的时候，这条read会被统计多次</td>
</tr>
<tr>
<td>-T</td>
<td>线程数目，1~32</td>
</tr>
<tr>
<td>下面是有关featrue/metafeature选择的参数</td>
<td>参数说明</td>
</tr>
<tr>
<td>-p</td>
<td>只能用在paired-end的情况中，会统计fragment而不统计read</td>
</tr>
<tr>
<td>-B</td>
<td>在-p选择的条件下，只有两端read都比对上的fragment才会被统计</td>
</tr>
<tr>
<td>-C</td>
<td>如果-C被设置，那融合的fragment（比对到不同染色体上的fragment）就不会被计数，这个只有在-p被设置的条件下使用</td>
</tr>
<tr>
<td>-d &lt; int &gt;</td>
<td>最短的fragment，默认是50</td>
</tr>
<tr>
<td>-D &lt; int &gt;</td>
<td>最长的fragmen，默认是600</td>
</tr>
<tr>
<td>-f</td>
<td>如果-f被设置，那将会统计feature层面的数据，如exon-level，否则会统计meta-feature层面的数据，如gene-levels</td>
</tr>
<tr>
<td>-g &lt; string &gt;</td>
<td>当参考的gtf提供的时候，我们需要提供一个id identifier 来将feature水平的统计汇总为meta-feature水平的统计，默认为gene_id，注意！选择gtf中提供的id identifier！！！</td>
</tr>
<tr>
<td>-t &lt; string &gt;</td>
<td>设置feature-type，-t指定的必须是gtf中有的feature，同时read只有落到这些feature上才会被统计到，默认是“exon”</td>
</tr>
</tbody>
</table>
<h4 id="featurecounts输出结果">featurecounts输出结果</h4>
<p>feature_count.txt</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020154938873.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020154938873.png, image-20211020154938873.png 1.5x, /posts/rnaseq-quantification/image-20211020154938873.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020154938873.png"
        title="image-20211020154938873" /></p>
<p>文件中包含基因ID，定量到该基因的所有外显子的染色体编号，和所有外显子的起始 ，终止位置，链的特征，外显子长度总和   和  各个样品中该基因的表达量</p>
<p>RASshCON.count.txt.summary</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020155005873.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020155005873.png, image-20211020155005873.png 1.5x, /posts/rnaseq-quantification/image-20211020155005873.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020155005873.png"
        title="image-20211020155005873" /></p>
<p>输出mapping的总体情况</p>
<h4 id="featurecounts总结">featurecounts总结</h4>
<p>featurecounts软件内部编程大多使用c语言编写而htseq使用python,所以featurecounts快，featurecounts对genome的bin和block区块化，更加速了定量速度</p>
<p>在对gtf的理解上和处理ambiguous reads时，featurecounts比htseq参数控制多</p>
<h2 id="kallistosalmonrsemstringtiefeaturecounthtseq定量结果的差异分析比较">kallisto,salmon,rsem,stringtie,featurecount,htseq定量结果的差异分析比较</h2>
<h3 id="测试数据不同软件定量结果分析">测试数据不同软件定量结果分析</h3>
<p>从云平台上随机下载了一个项目的数据（MJ20200513073-FX2020061700029），针对该项目的2个样品分别使用kallisto,salmon,rsem,stringtie,featurecount,htseq进行了定量分析，并对定量结果的count值和tpm值进行了整理，tpm值整理结果如下:</p>
<h4 id="tpm值的相关性分析">tpm值的相关性分析</h4>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020155516258.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020155516258.png, image-20211020155516258.png 1.5x, /posts/rnaseq-quantification/image-20211020155516258.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020155516258.png"
        title="image-20211020155516258" /></p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020155526985.png"
        data-srcset="image-20211020155526985.png, image-20211020155526985.png 1.5x, image-20211020155526985.png 2x"
        data-sizes="auto"
        alt="image-20211020155526985.png"
        title="image-20211020155526985" /></p>
<h4 id="tpm值的pca">tpm值的pca</h4>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020155547681.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020155547681.png, image-20211020155547681.png 1.5x, /posts/rnaseq-quantification/image-20211020155547681.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020155547681.png"
        title="image-20211020155547681" /></p>
<h4 id="相关性和pca总结">相关性和pca总结：</h4>
<p>featurecount和htseq定量结果相似
salmon和rsem定量结果相似</p>
<p>featurecount和htseq和stringtie定量结果相似可能是由于他们输入文件都是相同的bam</p>
<p>salmon和rsem和kallisto定量结果相似可能是由于他们都是基于转录本定量的结果并且他们都是基于EM算法模型</p>
<h4 id="tpm值的大小情况">tpm值的大小情况</h4>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020155613544.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020155613544.png, image-20211020155613544.png 1.5x, /posts/rnaseq-quantification/image-20211020155613544.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020155613544.png"
        title="image-20211020155613544" /></p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020155622625.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020155622625.png, image-20211020155622625.png 1.5x, /posts/rnaseq-quantification/image-20211020155622625.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020155622625.png"
        title="image-20211020155622625" /></p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020155629687.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020155629687.png, image-20211020155629687.png 1.5x, /posts/rnaseq-quantification/image-20211020155629687.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020155629687.png"
        title="image-20211020155629687" /></p>
<h4 id="tpm值大小总结">tpm值大小总结：</h4>
<p>stringtie的定量结果总体偏高
rsem的定量结果总体偏低
rsem和salmon对低表达水平gene比较敏感</p>
<h3 id="测试数据不同定量结果的差异分析">测试数据不同定量结果的差异分析</h3>
<p>从云平台上随机下载了一个项目的数据（MJ20200513073-FX2020061700029），针对该项目的2个样品分别使用kallisto,salmon,rsem,stringtie,featurecount,htseq进行了定量分析，之后使用edgR进行差异分析，结果如下：</p>
<table>
<thead>
<tr>
<th>compare</th>
<th>all</th>
<th>up</th>
<th>down</th>
<th>software</th>
<th>padj</th>
<th>fc</th>
</tr>
</thead>
<tbody>
<tr>
<td>featurecount_1vsfeaturecount_2</td>
<td>1336</td>
<td>640</td>
<td>696</td>
<td>edgeR</td>
<td>0.05</td>
<td>2</td>
</tr>
<tr>
<td>htseq_1vshtseq_2</td>
<td>1340</td>
<td>642</td>
<td>698</td>
<td>edgeR</td>
<td>0.05</td>
<td>2</td>
</tr>
<tr>
<td>kallisto_1vskallisto_2</td>
<td>1525</td>
<td>743</td>
<td>782</td>
<td>edgeR</td>
<td>0.05</td>
<td>2</td>
</tr>
<tr>
<td>rsem_1vsrsem_2</td>
<td>1463</td>
<td>728</td>
<td>735</td>
<td>edgeR</td>
<td>0.05</td>
<td>2</td>
</tr>
<tr>
<td>salmon_1vssalmon_2</td>
<td>1467</td>
<td>709</td>
<td>758</td>
<td>edgeR</td>
<td>0.05</td>
<td>2</td>
</tr>
<tr>
<td>stringtie_1vsstringtie_2</td>
<td>3640</td>
<td>1707</td>
<td>1933</td>
<td>edgeR</td>
<td>0.05</td>
<td>2</td>
</tr>
</tbody>
</table>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020160855476.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020160855476.png, image-20211020160855476.png 1.5x, /posts/rnaseq-quantification/image-20211020160855476.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020160855476.png"
        title="image-20211020160855476" /></p>
<h4 id="差异gene的venn图">差异gene的Venn图</h4>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020160914333.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020160914333.png, image-20211020160914333.png 1.5x, /posts/rnaseq-quantification/image-20211020160914333.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020160914333.png"
        title="image-20211020160914333" /></p>
<p><img src="image-20211020161052880.png" alt="image-20211020161052880"  /><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020161122595.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020161122595.png, image-20211020161122595.png 1.5x, /posts/rnaseq-quantification/image-20211020161122595.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020161122595.png"
        title="image-20211020161122595" /><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020161129706.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020161129706.png, image-20211020161129706.png 1.5x, /posts/rnaseq-quantification/image-20211020161129706.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020161129706.png"
        title="image-20211020161129706" /></p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020161140333.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020161140333.png, image-20211020161140333.png 1.5x, /posts/rnaseq-quantification/image-20211020161140333.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020161140333.png"
        title="image-20211020161140333" /><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020161146447.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020161146447.png, image-20211020161146447.png 1.5x, /posts/rnaseq-quantification/image-20211020161146447.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020161146447.png"
        title="image-20211020161146447" /><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020161152579.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020161152579.png, image-20211020161152579.png 1.5x, /posts/rnaseq-quantification/image-20211020161152579.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020161152579.png"
        title="image-20211020161152579" /></p>
<h4 id="差异基因聚类分析">差异基因聚类分析</h4>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020161827695.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020161827695.png, image-20211020161827695.png 1.5x, /posts/rnaseq-quantification/image-20211020161827695.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020161827695.png"
        title="image-20211020161827695" /></p>
<table>
<thead>
<tr>
<th>compare</th>
<th>all</th>
<th>up</th>
<th>down</th>
<th>software</th>
<th>padj</th>
<th>fc</th>
</tr>
</thead>
<tbody>
<tr>
<td>featurecount_1vsfeaturecount_2</td>
<td>103</td>
<td>26</td>
<td>77</td>
<td>edgeR</td>
<td>0.01</td>
<td>2</td>
</tr>
<tr>
<td>htseq_1vshtseq_2</td>
<td>100</td>
<td>27</td>
<td>73</td>
<td>edgeR</td>
<td>0.01</td>
<td>2</td>
</tr>
<tr>
<td>kallisto_1vskallisto_2</td>
<td>131</td>
<td>37</td>
<td>94</td>
<td>edgeR</td>
<td>0.01</td>
<td>2</td>
</tr>
<tr>
<td>rsem_1vsrsem_2</td>
<td>164</td>
<td>56</td>
<td>108</td>
<td>edgeR</td>
<td>0.01</td>
<td>2</td>
</tr>
<tr>
<td>salmon_1vssalmon_2</td>
<td>171</td>
<td>60</td>
<td>111</td>
<td>edgeR</td>
<td>0.01</td>
<td>2</td>
</tr>
<tr>
<td>stringtie_1vsstringtie_2</td>
<td>793</td>
<td>331</td>
<td>462</td>
<td>edgeR</td>
<td>0.01</td>
<td>2</td>
</tr>
</tbody>
</table>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020161224463.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020161224463.png, image-20211020161224463.png 1.5x, /posts/rnaseq-quantification/image-20211020161224463.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020161224463.png"
        title="image-20211020161224463" /></p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020161231691.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020161231691.png, image-20211020161231691.png 1.5x, /posts/rnaseq-quantification/image-20211020161231691.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020161231691.png"
        title="image-20211020161231691" /></p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020161240823.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020161240823.png, image-20211020161240823.png 1.5x, /posts/rnaseq-quantification/image-20211020161240823.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020161240823.png"
        title="image-20211020161240823" /><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020161246482.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020161246482.png, image-20211020161246482.png 1.5x, /posts/rnaseq-quantification/image-20211020161246482.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020161246482.png"
        title="image-20211020161246482" /><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020161250637.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020161250637.png, image-20211020161250637.png 1.5x, /posts/rnaseq-quantification/image-20211020161250637.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020161250637.png"
        title="image-20211020161250637" /></p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020161257836.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020161257836.png, image-20211020161257836.png 1.5x, /posts/rnaseq-quantification/image-20211020161257836.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020161257836.png"
        title="image-20211020161257836" /><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020161303892.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020161303892.png, image-20211020161303892.png 1.5x, /posts/rnaseq-quantification/image-20211020161303892.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020161303892.png"
        title="image-20211020161303892" /><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="image-20211020161308022.png"
        data-srcset="/posts/rnaseq-quantification/image-20211020161308022.png, image-20211020161308022.png 1.5x, /posts/rnaseq-quantification/image-20211020161308022.png 2x"
        data-sizes="auto"
        alt="/posts/rnaseq-quantification/image-20211020161308022.png"
        title="image-20211020161308022" /></p>
<h4 id="差异分析总结">差异分析总结</h4>
<p>featurecount和htseq的差异分析结果比较相似
salmon和rsem的差异分析结果比较相似
kallisto的差异分析结果和其他定量软件相比差异较大</p>
</div><div class="post-footer" id="post-footer">
    <div class="post-info">
        <div class="post-info-line">
            <div class="post-info-mod">
                <span>更新于 2021-10-16</span>
            </div>
            <div class="post-info-license"></div>
        </div>
        <div class="post-info-line">
            <div class="post-info-md"><span>
                            <a class="link-to-markdown" href="/posts/rnaseq-quantification/index.md" target="_blank">阅读原始文档</a>
                        </span></div>
            <div class="post-info-share">
                <span><a href="javascript:void(0);" title="分享到 Twitter" data-sharer="twitter" data-url="http://cb-majorbiolab.work/posts/rnaseq-quantification/" data-title="RNAseq定量软件比较" data-via="https://twitter.com/czheluo" data-hashtags="RNAseq,DESeq2,差异分析,count,表达量"><i class="fab fa-twitter fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Facebook" data-sharer="facebook" data-url="http://cb-majorbiolab.work/posts/rnaseq-quantification/" data-hashtag="RNAseq"><i class="fab fa-facebook-square fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Linkedin" data-sharer="linkedin" data-url="http://cb-majorbiolab.work/posts/rnaseq-quantification/"><i class="fab fa-linkedin fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Hacker News" data-sharer="hackernews" data-url="http://cb-majorbiolab.work/posts/rnaseq-quantification/" data-title="RNAseq定量软件比较"><i class="fab fa-hacker-news fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Line" data-sharer="line" data-url="http://cb-majorbiolab.work/posts/rnaseq-quantification/" data-title="RNAseq定量软件比较"><i data-svg-src="/lib/simple-icons/icons/line.min.svg"></i></a><a href="javascript:void(0);" title="分享到 微博" data-sharer="weibo" data-url="http://cb-majorbiolab.work/posts/rnaseq-quantification/" data-title="RNAseq定量软件比较" data-ralateuid="https://weibo.com/2161302013/profile"><i class="fab fa-weibo fa-fw"></i></a></span>
            </div>
        </div>
    </div>

    <div class="post-info-more">
        <section class="post-tags"><i class="fas fa-tags fa-fw"></i>&nbsp;<a href="/tags/rnaseq/">RNAseq</a>,&nbsp;<a href="/tags/deseq2/">DESeq2</a>,&nbsp;<a href="/tags/%E5%B7%AE%E5%BC%82%E5%88%86%E6%9E%90/">差异分析</a>,&nbsp;<a href="/tags/count/">count</a>,&nbsp;<a href="/tags/%E8%A1%A8%E8%BE%BE%E9%87%8F/">表达量</a></section>
        <section>
            <span><a href="javascript:void(0);" onclick="window.history.back();">返回</a></span>&nbsp;|&nbsp;<span><a href="/">主页</a></span>
        </section>
    </div>

    <div class="post-nav"><a href="/posts/differential-analysis-of-circrna-count-data-circrna/" class="prev" rel="prev" title="circRNA差异表达分析软件"><i class="fas fa-angle-left fa-fw"></i>circRNA差异表达分析软件</a>
            <a href="/posts/dimensionality-reduction-methods-for-rna-seq-analysis/" class="next" rel="next" title="降维分析方法比较">降维分析方法比较<i class="fas fa-angle-right fa-fw"></i></a></div>
</div>
<div id="comments"><div id="valine" class="comment"></div><noscript>
                Please enable JavaScript to view the comments powered by <a href="https://valine.js.org/">Valine</a>.
            </noscript><div id="fb-root" class="comment"></div>
            <div
                class="fb-comments"
                data-href="http://cb-majorbiolab.work/posts/rnaseq-quantification/"
                data-width="100%"
                data-numposts="10"
            ></div><noscript>
                Please enable JavaScript to view the comments powered by <a href="https://developers.facebook.com/docs/plugins/comments/"></a>Facebook</a>.
            </noscript></div></article></div>
            </main><footer class="footer">
        <div class="footer-container"><div class="footer-line">MAJOR Computational Biology Research Lab</div>
                <div class="footer-line"><i class="far fa-copyright fa-fw"></i><span itemprop="copyrightYear">2020-2022, MENG LUO, All rights reserved.</span><span class="author" itemprop="copyrightHolder">&nbsp;<a href="/" target="_blank"></a></span></div>
                
        </div>
    </footer></div>

        <div id="fixed-buttons"><a href="#" id="back-to-top" class="fixed-button" title="回到顶部">
                <i class="fas fa-arrow-up fa-fw"></i>
            </a><a href="#" id="view-comments" class="fixed-button" title="查看评论">
                <i class="fas fa-comment fa-fw"></i>
            </a>
        </div><link rel="stylesheet" href="/lib/valine/valine.min.css"><link rel="stylesheet" href="/lib/lightgallery/lightgallery.min.css"><link rel="stylesheet" href="/lib/katex/katex.min.css"><link rel="stylesheet" href="/lib/katex/copy-tex.min.css"><script type="text/javascript" src="/lib/valine/Valine.min.js"></script><script type="text/javascript" src="https://connect.facebook.net/zh_CN/sdk.js#xfbml=1&amp;version=v5.0&amp;appId=&amp;autoLogAppEvents=1" defer></script><script type="text/javascript" src="/lib/smooth-scroll/smooth-scroll.min.js"></script><script type="text/javascript" src="/lib/autocomplete/autocomplete.min.js"></script><script type="text/javascript" src="/lib/algoliasearch/algoliasearch-lite.umd.min.js"></script><script type="text/javascript" src="/lib/lazysizes/lazysizes.min.js"></script><script type="text/javascript" src="/lib/lightgallery/lightgallery.min.js"></script><script type="text/javascript" src="/lib/lightgallery/lg-thumbnail.min.js"></script><script type="text/javascript" src="/lib/lightgallery/lg-zoom.min.js"></script><script type="text/javascript" src="/lib/clipboard/clipboard.min.js"></script><script type="text/javascript" src="/lib/sharer/sharer.min.js"></script><script type="text/javascript" src="/lib/katex/katex.min.js"></script><script type="text/javascript" src="/lib/katex/auto-render.min.js"></script><script type="text/javascript" src="/lib/katex/copy-tex.min.js"></script><script type="text/javascript" src="/lib/katex/mhchem.min.js"></script><script type="text/javascript">window.config={"code":{"copyTitle":"复制到剪贴板","maxShownLines":10},"comment":{"valine":{"appId":"Noyy9UKaXGwYBIUdvBTt7u76-gzGzoHsz","appKey":"15w6pTF8mFoxnm8muOFx5buM","avatar":"mp","el":"#valine","emojiCDN":"https://cdn.jsdelivr.net/npm/emoji-datasource-google@5.0.1/img/google/64/","emojiMaps":{"100":"1f4af.png","alien":"1f47d.png","anger":"1f4a2.png","angry":"1f620.png","anguished":"1f627.png","astonished":"1f632.png","black_heart":"1f5a4.png","blue_heart":"1f499.png","blush":"1f60a.png","bomb":"1f4a3.png","boom":"1f4a5.png","broken_heart":"1f494.png","brown_heart":"1f90e.png","clown_face":"1f921.png","cold_face":"1f976.png","cold_sweat":"1f630.png","confounded":"1f616.png","confused":"1f615.png","cry":"1f622.png","crying_cat_face":"1f63f.png","cupid":"1f498.png","dash":"1f4a8.png","disappointed":"1f61e.png","disappointed_relieved":"1f625.png","dizzy":"1f4ab.png","dizzy_face":"1f635.png","drooling_face":"1f924.png","exploding_head":"1f92f.png","expressionless":"1f611.png","face_vomiting":"1f92e.png","face_with_cowboy_hat":"1f920.png","face_with_hand_over_mouth":"1f92d.png","face_with_head_bandage":"1f915.png","face_with_monocle":"1f9d0.png","face_with_raised_eyebrow":"1f928.png","face_with_rolling_eyes":"1f644.png","face_with_symbols_on_mouth":"1f92c.png","face_with_thermometer":"1f912.png","fearful":"1f628.png","flushed":"1f633.png","frowning":"1f626.png","ghost":"1f47b.png","gift_heart":"1f49d.png","green_heart":"1f49a.png","grimacing":"1f62c.png","grin":"1f601.png","grinning":"1f600.png","hankey":"1f4a9.png","hear_no_evil":"1f649.png","heart":"2764-fe0f.png","heart_decoration":"1f49f.png","heart_eyes":"1f60d.png","heart_eyes_cat":"1f63b.png","heartbeat":"1f493.png","heartpulse":"1f497.png","heavy_heart_exclamation_mark_ornament":"2763-fe0f.png","hole":"1f573-fe0f.png","hot_face":"1f975.png","hugging_face":"1f917.png","hushed":"1f62f.png","imp":"1f47f.png","innocent":"1f607.png","japanese_goblin":"1f47a.png","japanese_ogre":"1f479.png","joy":"1f602.png","joy_cat":"1f639.png","kiss":"1f48b.png","kissing":"1f617.png","kissing_cat":"1f63d.png","kissing_closed_eyes":"1f61a.png","kissing_heart":"1f618.png","kissing_smiling_eyes":"1f619.png","laughing":"1f606.png","left_speech_bubble":"1f5e8-fe0f.png","love_letter":"1f48c.png","lying_face":"1f925.png","mask":"1f637.png","money_mouth_face":"1f911.png","nauseated_face":"1f922.png","nerd_face":"1f913.png","neutral_face":"1f610.png","no_mouth":"1f636.png","open_mouth":"1f62e.png","orange_heart":"1f9e1.png","partying_face":"1f973.png","pensive":"1f614.png","persevere":"1f623.png","pleading_face":"1f97a.png","pouting_cat":"1f63e.png","purple_heart":"1f49c.png","rage":"1f621.png","relaxed":"263a-fe0f.png","relieved":"1f60c.png","revolving_hearts":"1f49e.png","right_anger_bubble":"1f5ef-fe0f.png","robot_face":"1f916.png","rolling_on_the_floor_laughing":"1f923.png","scream":"1f631.png","scream_cat":"1f640.png","see_no_evil":"1f648.png","shushing_face":"1f92b.png","skull":"1f480.png","skull_and_crossbones":"2620-fe0f.png","sleeping":"1f634.png","sleepy":"1f62a.png","slightly_frowning_face":"1f641.png","slightly_smiling_face":"1f642.png","smile":"1f604.png","smile_cat":"1f638.png","smiley":"1f603.png","smiley_cat":"1f63a.png","smiling_face_with_3_hearts":"1f970.png","smiling_imp":"1f608.png","smirk":"1f60f.png","smirk_cat":"1f63c.png","sneezing_face":"1f927.png","sob":"1f62d.png","space_invader":"1f47e.png","sparkling_heart":"1f496.png","speak_no_evil":"1f64a.png","speech_balloon":"1f4ac.png","star-struck":"1f929.png","stuck_out_tongue":"1f61b.png","stuck_out_tongue_closed_eyes":"1f61d.png","stuck_out_tongue_winking_eye":"1f61c.png","sunglasses":"1f60e.png","sweat":"1f613.png","sweat_drops":"1f4a6.png","sweat_smile":"1f605.png","thinking_face":"1f914.png","thought_balloon":"1f4ad.png","tired_face":"1f62b.png","triumph":"1f624.png","two_hearts":"1f495.png","unamused":"1f612.png","upside_down_face":"1f643.png","weary":"1f629.png","white_frowning_face":"2639-fe0f.png","white_heart":"1f90d.png","wink":"1f609.png","woozy_face":"1f974.png","worried":"1f61f.png","yawning_face":"1f971.png","yellow_heart":"1f49b.png","yum":"1f60b.png","zany_face":"1f92a.png","zipper_mouth_face":"1f910.png","zzz":"1f4a4.png"},"enableQQ":false,"highlight":true,"lang":"zh-cn","pageSize":10,"placeholder":"你的评论 ...","recordIP":true,"visitor":true}},"lightGallery":{"actualSize":false,"exThumbImage":"data-thumbnail","hideBarsDelay":2000,"selector":".lightgallery","speed":400,"thumbContHeight":80,"thumbWidth":80,"thumbnail":true},"math":{"delimiters":[{"display":true,"left":"$$","right":"$$"},{"display":true,"left":"\\[","right":"\\]"},{"display":false,"left":"$","right":"$"},{"display":false,"left":"\\(","right":"\\)"}],"strict":false},"search":{"algoliaAppID":"VX4BXOJBC5","algoliaIndex":"majorbio.zh-cn","algoliaSearchKey":"53790f167edb548b217bfe91dd891ac0","highlightTag":"em","maxResultLength":10,"noResultsFound":"没有找到结果","snippetLength":50,"type":"algolia"}};</script><script type="text/javascript" src="/js/theme.min.js"></script></body>
</html>
