<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="robots" content="noodp" />
        <meta http-equiv="X-UA-Compatible" content="IE=edge, chrome=1">
        <title>The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data - MAJOR Computational Biology Research Lab</title><meta name="Description" content="The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data"><meta property="og:title" content="The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data" />
<meta property="og:description" content="The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data" />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://cb-majorbiolab.work/posts/sthdwgcna-pipeline/" />
<meta property="og:image" content="http://cb-majorbiolab.work/posts/sthdwgcna-pipeline/hdwgcna.png"/>
<meta property="article:published_time" content="2023-06-19T17:29:59+08:00" />
<meta property="article:modified_time" content="2023-06-19T16:29:59+08:00" />
<meta name="twitter:card" content="summary_large_image"/>
<meta name="twitter:image" content="http://cb-majorbiolab.work/posts/sthdwgcna-pipeline/hdwgcna.png"/>
<meta name="twitter:title" content="The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data"/>
<meta name="twitter:description" content="The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data"/>
<meta name="application-name" content="MAJOR Computational Biology Research Lab">
<meta name="apple-mobile-web-app-title" content="MAJOR Computational Biology Research Lab"><meta name="theme-color" content="#ffffff"><meta name="msapplication-TileColor" content="#da532c"><link rel="icon" href="/images/MBIL.svg"><link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png"><link rel="mask-icon" href="/safari-pinned-tab.svg" color="#5bbad5"><link rel="manifest" href="/site.webmanifest"><link rel="canonical" href="http://cb-majorbiolab.work/posts/sthdwgcna-pipeline/" /><link rel="prev" href="http://cb-majorbiolab.work/posts/twas-pipeline/" /><link rel="stylesheet" href="/lib/normalize/normalize.min.css"><link rel="stylesheet" href="/css/style.min.css"><link rel="stylesheet" href="/lib/fontawesome-free/all.min.css"><link rel="stylesheet" href="/lib/animate/animate.min.css"><script type="application/ld+json">
    {
        "@context": "http://schema.org",
        "@type": "BlogPosting",
        "headline": "The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data",
        "inLanguage": "en",
        "mainEntityOfPage": {
            "@type": "WebPage",
            "@id": "http:\/\/cb-majorbiolab.work\/posts\/sthdwgcna-pipeline\/"
        },"image": [{
                            "@type": "ImageObject",
                            "url": "http:\/\/cb-majorbiolab.work\/posts\/sthdwgcna-pipeline\/hdwgcna.png",
                            "width":  1340 ,
                            "height":  679 
                        }],"genre": "posts","keywords": "SC, ST, workflow, SCRNAseq","wordcount":  3366 ,
        "url": "http:\/\/cb-majorbiolab.work\/posts\/sthdwgcna-pipeline\/","datePublished": "2023-06-19T17:29:59+08:00","dateModified": "2023-06-19T16:29:59+08:00","publisher": {
            "@type": "Organization",
            "name": "美吉生物信息团队"},"author": {
                "@type": "Person",
                "name": "Meng Luo"
            },"description": "The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data"
    }
    </script></head>
    <body header-desktop="" header-mobile=""><script type="text/javascript">(window.localStorage && localStorage.getItem('theme') ? localStorage.getItem('theme') === 'dark' : ('' === 'auto' ? window.matchMedia('(prefers-color-scheme: dark)').matches : '' === 'dark')) && document.body.setAttribute('theme', 'dark');</script>

        <div id="mask"></div><div class="wrapper"><header class="desktop" id="header-desktop">
    <div class="header-wrapper">
        <div class="header-title">
            <a href="/" title="MAJOR Computational Biology Research Lab"><img
        class="lazyload logo"
        src="/svg/loading.min.svg"
        data-src="/images/MBIL.svg"
        data-srcset="/images/MBIL.svg, /images/MBIL.svg 1.5x, /images/MBIL.svg 2x"
        data-sizes="auto"
        alt="/images/MBIL.svg"
        title="/images/MBIL.svg" /></a>
        </div>
        <div class="menu">
            <div class="menu-inner"><a class="menu-item" href="/posts/" title="Home"> Home </a><a class="menu-item" href="/workshop/" title=" "> Workshop/Talk </a><a class="menu-item" href="/publication/" title="Publications"> Publications </a><a class="menu-item" href="/team/" title="Team"> Team </a><a class="menu-item" href="/resource/" title="Resource"> Resource </a><a class="menu-item" href="/software/" title="Software"> Software </a><a class="menu-item" href="/ne/" title="News/Events"> News/Events </a><a class="menu-item" href="/joinus/" title="Join/Contact"> Join/Contact </a><a class="menu-item" href="https://github.com/MajorBio" title="GitHub" rel="noopener noreffer" target="_blank"><i class='fab fa-github fa-fw'></i>  </a><span class="menu-item delimiter"></span><span class="menu-item search" id="search-desktop">
                        <input type="text" placeholder="搜索文章标题或内容..." id="search-input-desktop">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-desktop" title="搜索">
                            <i class="fas fa-search fa-fw"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-desktop" title="清空">
                            <i class="fas fa-times-circle fa-fw"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-desktop">
                            <i class="fas fa-spinner fa-fw fa-spin"></i>
                        </span>
                    </span><a href="javascript:void(0);" class="menu-item theme-switch" title="切换主题">
                    <i class="fas fa-adjust fa-fw"></i>
                </a>
            </div>
        </div>
    </div>
</header><header class="mobile" id="header-mobile">
    <div class="header-container">
        <div class="header-wrapper">
            <div class="header-title">
                <a href="/" title="MAJOR Computational Biology Research Lab"><img
        class="lazyload logo"
        src="/svg/loading.min.svg"
        data-src="/images/MBIL.svg"
        data-srcset="/images/MBIL.svg, /images/MBIL.svg 1.5x, /images/MBIL.svg 2x"
        data-sizes="auto"
        alt="/images/MBIL.svg"
        title="/images/MBIL.svg" /></a>
            </div>
            <div class="menu-toggle" id="menu-toggle-mobile">
                <span></span><span></span><span></span>
            </div>
        </div>
        <div class="menu" id="menu-mobile"><div class="search-wrapper">
                    <div class="search mobile" id="search-mobile">
                        <input type="text" placeholder="搜索文章标题或内容..." id="search-input-mobile">
                        <a href="javascript:void(0);" class="search-button search-toggle" id="search-toggle-mobile" title="搜索">
                            <i class="fas fa-search fa-fw"></i>
                        </a>
                        <a href="javascript:void(0);" class="search-button search-clear" id="search-clear-mobile" title="清空">
                            <i class="fas fa-times-circle fa-fw"></i>
                        </a>
                        <span class="search-button search-loading" id="search-loading-mobile">
                            <i class="fas fa-spinner fa-fw fa-spin"></i>
                        </span>
                    </div>
                    <a href="javascript:void(0);" class="search-cancel" id="search-cancel-mobile">
                        取消
                    </a>
                </div><a class="menu-item" href="/posts/" title="Home">Home</a><a class="menu-item" href="/workshop/" title=" ">Workshop/Talk</a><a class="menu-item" href="/publication/" title="Publications">Publications</a><a class="menu-item" href="/team/" title="Team">Team</a><a class="menu-item" href="/resource/" title="Resource">Resource</a><a class="menu-item" href="/software/" title="Software">Software</a><a class="menu-item" href="/ne/" title="News/Events">News/Events</a><a class="menu-item" href="/joinus/" title="Join/Contact">Join/Contact</a><a class="menu-item" href="https://github.com/MajorBio" title="GitHub" rel="noopener noreffer" target="_blank"><i class='fab fa-github fa-fw'></i></a><a href="javascript:void(0);" class="menu-item theme-switch" title="切换主题">
                <i class="fas fa-adjust fa-fw"></i>
            </a></div>
    </div>
</header>
<div class="search-dropdown desktop">
    <div id="search-dropdown-desktop"></div>
</div>
<div class="search-dropdown mobile">
    <div id="search-dropdown-mobile"></div>
</div><main class="main">
                <div class="container"><div class="toc" id="toc-auto">
            <h2 class="toc-title">目录</h2>
            <div class="toc-content" id="toc-content-auto"></div>
        </div><article class="page single"><h1 class="single-title animated flipInX">The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data</h1><div class="post-meta">
            <div class="post-meta-line"><span class="post-author"><a href="https://github.com/czheluo" title="Author" target="_blank" rel="noopener noreffer author" class="author"><i class="fas fa-user-circle fa-fw"></i>Meng Luo</a></span>&nbsp;<span class="post-category">收录于 <a href="/categories/scrnaseq/"><i class="far fa-folder fa-fw"></i>scRNAseq</a>&nbsp;<a href="/categories/pipeline/"><i class="far fa-folder fa-fw"></i>Pipeline</a>&nbsp;<a href="/categories/workflow/"><i class="far fa-folder fa-fw"></i>workflow</a></span></div>
            <div class="post-meta-line"><i class="far fa-calendar-alt fa-fw"></i>&nbsp;<time datetime="2023-06-19">2023-06-19</time>&nbsp;<i class="fas fa-pencil-alt fa-fw"></i>&nbsp;约 3366 字&nbsp;
                <i class="far fa-clock fa-fw"></i>&nbsp;预计阅读 7 分钟&nbsp;<span id="/posts/sthdwgcna-pipeline/" class="leancloud_visitors" data-flag-title="The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data">
                        <i class="far fa-eye fa-fw"></i>&nbsp;<span class=leancloud-visitors-count></span>&nbsp;次阅读
                    </span>&nbsp;</div>
        </div><div class="featured-image"><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="/posts/sthdwgcna-pipeline/hdwgcna.png"
        data-srcset="/posts/sthdwgcna-pipeline/hdwgcna.png, /posts/sthdwgcna-pipeline/hdwgcna.png 1.5x, /posts/sthdwgcna-pipeline/hdwgcna.png 2x"
        data-sizes="auto"
        alt="/posts/sthdwgcna-pipeline/hdwgcna.png"
        title="The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data" /></div><div class="details toc" id="toc-static"  kept="">
                <div class="details-summary toc-title">
                    <span>目录</span>
                    <span><i class="details-icon fas fa-angle-right"></i></span>
                </div>
                <div class="details-content toc-content" id="toc-content-static"><nav id="TableOfContents">
  <ul>
    <li><a href="#the-workflow-of-hdwgcna-analysis-for-single-cell-spatial-transcriptomics-data">The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data</a>
      <ul>
        <li>
          <ul>
            <li><a href="#learning-video">learning Video</a></li>
            <li><a href="#references">References</a></li>
          </ul>
        </li>
      </ul>
    </li>
  </ul>
</nav></div>
            </div><div class="content" id="content"><h2 id="the-workflow-of-hdwgcna-analysis-for-single-cell-spatial-transcriptomics-data">The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data</h2>
<p><figure><a class="lightgallery" href="hdwgcna.png" title="sc/nRNA-seq Schematic" data-thumbnail="hdwgcna.png" data-sub-html="<h2>RNA-seq Schematic</h2><p>sc/nRNA-seq Schematic</p>">
        <img
            class="lazyload"
            src="/svg/loading.min.svg"
            data-src="hdwgcna.png"
            data-srcset="hdwgcna.png, hdwgcna.png 1.5x, hdwgcna.png 2x"
            data-sizes="auto"
            alt="hdwgcna.png" />
    </a><figcaption class="image-caption">RNA-seq Schematic</figcaption>
    </figure></p>
<p>空间转录共表达网络分析流程</p>
<p>分析思路：针对数据的不同，分两种情况，当前流程不能进行多个样本分析，如果要分析需要针对过个样本的重新修改脚本（可以直接合并或者单独分析），目前的思路是分析单个样本的同单细胞分析一样的做法，多个样本需要（merge rds），再进行分析：</p>
<p>1，如果从原始count开始处理（针对spaceranger获得的结果），则只能按照cluster进行WGCNA分析，我们前期数据的过滤及分析目前写的脚本只能分析人和鼠的物种；</p>
<p>2，提供rds分析的结果及细胞注释结果，如果提供合并后的结果，同时有细胞注释的结果则也可以按照细胞注释的结果进行分析。</p>
<p>分析内容：数据预处理；模块识别；模块表型相关；hub基因可视化；差异模块分析</p>
<p><em><strong>*基于以上做好的数据准备：*</strong></em></p>
<p><em><strong>*1，包含多个样本（样本得必须单独分开分析，只不过需要吧几个样本都放在一个object里面），数据已经进行常规的分析，但是没有细胞注释。*</strong></em></p>
<p><strong>2，</strong><em><strong>*包含多个样本，数据已经进行常规的分析，同时有注释信息的结果。*</strong></em></p>
<p><strong>3，</strong><em><strong>*分析则可以按照样本来进行分析，不依赖细胞类型及聚类分析来进行分析，这个根据给的分组情况来选择（分组中也有不同的细胞类型，虽然是一起分析的，但是可以按照样本来划分从而进行分析）。*</strong></em></p>
<p><strong>4，</strong><em><strong>*先过滤样本，然后再使用metaspot（同metacell）的方法来构建所有的关系，但是这里如果构建得基于你的数据，比如聚类的结果，或者不同样本间的关系，然后再往下构建WGCNA的分析。*</strong></em></p>
<p><strong>5，</strong><em><strong>*如果是空间转录的常规分析推荐使用SCT的方法做标准化处理前期的数据*</strong></em><em><strong>*，和后面不同标准化的方法不同，根据空间的分析，最好使用SCT的处理来进行分析。Metaspot（可以按照每个细胞类型或者聚类的分类结果来进行分析） 之后，总的原本spot会减少很多，后续的获得的spot也会变得更少或许就几十个（小于50则会剔除）。*</strong></em></p>
<p><strong>6，</strong><em><strong>*空间转录用SCT的数据没有问题，seurat官方推荐使用，做WGCNA没有像单细胞那种做power遇到问题，就是power值，很难能到达0.8。*</strong></em><em><strong>*（kME计算时间很长和单细胞的不同，多个循环harmony计算的时间很长（最多10个），难于收敛（超过一天的计算都有可能）*</strong></em><em><strong>*）*</strong></em></p>
<p><strong>7，</strong><em><strong>*作图说明：如果分的模块很多的情况下，有些可视化的图需要修改输出的size，加大*</strong></em></p>
<p>一、运行流程</p>
<p>Rscript4.2 /mnt/ilustre/users/meng.luo/project/RNA/scRNA/hdWGCNA/spatial/hdWGCNA_ST_pipeline.R</p>
<p>参考分析路径：/mnt/ilustre/users/meng.luo/project/RNA/scRNA/hdWGCNA/spatial</p>
<p>根据不同的分析模块，给出对应不同的灵活分析参数，正常情况按照默认的运行没有问题。</p>
<p>具体参数（运行需要加载相应的代码（网络图做了更改）bin路径下，所以运行脚本时需要给脚本的绝对路径）：</p>
<div class="highlight"><div class="chroma">
<table class="lntable"><tr><td class="lntd">
<pre class="chroma"><code><span class="lnt">  1
</span><span class="lnt">  2
</span><span class="lnt">  3
</span><span class="lnt">  4
</span><span class="lnt">  5
</span><span class="lnt">  6
</span><span class="lnt">  7
</span><span class="lnt">  8
</span><span class="lnt">  9
</span><span class="lnt"> 10
</span><span class="lnt"> 11
</span><span class="lnt"> 12
</span><span class="lnt"> 13
</span><span class="lnt"> 14
</span><span class="lnt"> 15
</span><span class="lnt"> 16
</span><span class="lnt"> 17
</span><span class="lnt"> 18
</span><span class="lnt"> 19
</span><span class="lnt"> 20
</span><span class="lnt"> 21
</span><span class="lnt"> 22
</span><span class="lnt"> 23
</span><span class="lnt"> 24
</span><span class="lnt"> 25
</span><span class="lnt"> 26
</span><span class="lnt"> 27
</span><span class="lnt"> 28
</span><span class="lnt"> 29
</span><span class="lnt"> 30
</span><span class="lnt"> 31
</span><span class="lnt"> 32
</span><span class="lnt"> 33
</span><span class="lnt"> 34
</span><span class="lnt"> 35
</span><span class="lnt"> 36
</span><span class="lnt"> 37
</span><span class="lnt"> 38
</span><span class="lnt"> 39
</span><span class="lnt"> 40
</span><span class="lnt"> 41
</span><span class="lnt"> 42
</span><span class="lnt"> 43
</span><span class="lnt"> 44
</span><span class="lnt"> 45
</span><span class="lnt"> 46
</span><span class="lnt"> 47
</span><span class="lnt"> 48
</span><span class="lnt"> 49
</span><span class="lnt"> 50
</span><span class="lnt"> 51
</span><span class="lnt"> 52
</span><span class="lnt"> 53
</span><span class="lnt"> 54
</span><span class="lnt"> 55
</span><span class="lnt"> 56
</span><span class="lnt"> 57
</span><span class="lnt"> 58
</span><span class="lnt"> 59
</span><span class="lnt"> 60
</span><span class="lnt"> 61
</span><span class="lnt"> 62
</span><span class="lnt"> 63
</span><span class="lnt"> 64
</span><span class="lnt"> 65
</span><span class="lnt"> 66
</span><span class="lnt"> 67
</span><span class="lnt"> 68
</span><span class="lnt"> 69
</span><span class="lnt"> 70
</span><span class="lnt"> 71
</span><span class="lnt"> 72
</span><span class="lnt"> 73
</span><span class="lnt"> 74
</span><span class="lnt"> 75
</span><span class="lnt"> 76
</span><span class="lnt"> 77
</span><span class="lnt"> 78
</span><span class="lnt"> 79
</span><span class="lnt"> 80
</span><span class="lnt"> 81
</span><span class="lnt"> 82
</span><span class="lnt"> 83
</span><span class="lnt"> 84
</span><span class="lnt"> 85
</span><span class="lnt"> 86
</span><span class="lnt"> 87
</span><span class="lnt"> 88
</span><span class="lnt"> 89
</span><span class="lnt"> 90
</span><span class="lnt"> 91
</span><span class="lnt"> 92
</span><span class="lnt"> 93
</span><span class="lnt"> 94
</span><span class="lnt"> 95
</span><span class="lnt"> 96
</span><span class="lnt"> 97
</span><span class="lnt"> 98
</span><span class="lnt"> 99
</span><span class="lnt">100
</span><span class="lnt">101
</span><span class="lnt">102
</span><span class="lnt">103
</span><span class="lnt">104
</span><span class="lnt">105
</span><span class="lnt">106
</span><span class="lnt">107
</span><span class="lnt">108
</span><span class="lnt">109
</span><span class="lnt">110
</span><span class="lnt">111
</span><span class="lnt">112
</span><span class="lnt">113
</span><span class="lnt">114
</span><span class="lnt">115
</span><span class="lnt">116
</span><span class="lnt">117
</span><span class="lnt">118
</span><span class="lnt">119
</span></code></pre></td>
<td class="lntd">
<pre class="chroma"><code class="language-r" data-lang="r"><span class="n">Rscript4.2</span> <span class="o">/</span><span class="n">mnt</span><span class="o">/</span><span class="n">ilustre</span><span class="o">/</span><span class="n">users</span><span class="o">/</span><span class="n">meng.luo</span><span class="o">/</span><span class="n">project</span><span class="o">/</span><span class="n">RNA</span><span class="o">/</span><span class="n">scRNA</span><span class="o">/</span><span class="n">hdWGCNA</span><span class="o">/</span><span class="n">spatial</span><span class="o">/</span><span class="n">hdWGCNA_ST_pipeline.R</span> <span class="o">--</span><span class="n">rds</span> <span class="n">mouse_brain_ST_anterior.rds</span> <span class="o">--</span><span class="n">cellfile</span> <span class="n">cells.csv</span> 
详细参数：
<span class="n">Options</span><span class="o">:</span>
	<span class="o">-</span><span class="n">P</span> <span class="n">PREDATA</span><span class="p">,</span> <span class="o">--</span><span class="n">predata</span><span class="o">=</span><span class="n">PREDATA</span>
		<span class="n">PROCESSED</span> <span class="n">spatial</span> <span class="n">data</span>

	<span class="o">-</span><span class="n">H</span> <span class="n">OUTSPATH</span><span class="p">,</span> <span class="o">--</span><span class="n">outsPath</span><span class="o">=</span><span class="n">OUTSPATH</span>
		<span class="n">Input</span> <span class="n">spaceRanger</span> <span class="n">outs</span> <span class="n">pathway</span>

	<span class="o">-</span><span class="n">j</span> <span class="n">SLICE</span><span class="p">,</span> <span class="o">--</span><span class="n">slice</span><span class="o">=</span><span class="n">SLICE</span>
		<span class="n">spatial</span> <span class="n">image</span> <span class="n">name</span>

	<span class="o">-</span><span class="n">J</span> <span class="n">SPECIES</span><span class="p">,</span> <span class="o">--</span><span class="n">species</span><span class="o">=</span><span class="n">SPECIES</span>
		<span class="n">species</span> <span class="nf">name </span><span class="p">(</span><span class="n">human</span> <span class="n">or</span> <span class="n">mouse</span><span class="p">)</span>

	<span class="o">-</span><span class="n">Z</span> <span class="n">SCTRANSFORM</span><span class="p">,</span> <span class="o">--</span><span class="n">SCTransform</span><span class="o">=</span><span class="n">SCTRANSFORM</span>
		<span class="n">SCTransform</span> <span class="n">on</span> <span class="n">single</span><span class="o">-</span><span class="n">cell</span> <span class="n">data</span>

	<span class="o">-</span><span class="n">r</span> <span class="n">RDS</span><span class="p">,</span> <span class="o">--</span><span class="n">rds</span><span class="o">=</span><span class="n">RDS</span>
		<span class="n">Input</span> <span class="n">rds</span> <span class="n">file</span>

	<span class="o">-</span><span class="n">g</span> <span class="n">GENESELECT</span><span class="p">,</span> <span class="o">--</span><span class="n">geneselect</span><span class="o">=</span><span class="n">GENESELECT</span>
		 <span class="n">the</span> <span class="n">gene</span> <span class="n">selection</span> <span class="n">approach</span> <span class="nf">variable</span><span class="p">(</span> <span class="n">VariableFeatures</span><span class="p">)</span> <span class="nf">fraction </span><span class="p">(</span><span class="n">default</span><span class="p">)</span> <span class="n">all</span>  <span class="n">or</span>  <span class="nf">custom</span><span class="p">(</span><span class="n">gene_list</span><span class="p">)</span>

	<span class="o">-</span><span class="n">f</span> <span class="n">FRACTION</span><span class="p">,</span> <span class="o">--</span><span class="n">fraction</span><span class="o">=</span><span class="n">FRACTION</span>
		 <span class="n">fraction</span> <span class="n">of</span> <span class="n">cells</span> <span class="n">that</span> <span class="n">a</span> <span class="n">gene</span> <span class="n">needs</span> <span class="n">to</span> <span class="n">be</span> <span class="n">expressed</span> <span class="n">in</span> <span class="n">order</span> <span class="n">to</span> <span class="n">be</span> <span class="n">included</span>

	<span class="o">-</span><span class="n">l</span> <span class="n">LIST</span><span class="p">,</span> <span class="o">--</span><span class="n">list</span><span class="o">=</span><span class="n">LIST</span>
		 <span class="n">gene</span> <span class="n">selection</span> <span class="n">approach</span> <span class="n">custom</span> <span class="n">and</span> <span class="n">give</span> <span class="n">the</span> <span class="n">custom</span> <span class="n">gene</span> <span class="n">list</span>

	<span class="o">-</span><span class="n">c</span> <span class="n">CELLFILE</span><span class="p">,</span> <span class="o">--</span><span class="n">cellfile</span><span class="o">=</span><span class="n">CELLFILE</span>
		<span class="n">cell</span> <span class="n">annotation</span> <span class="n">file</span>

	<span class="o">-</span><span class="n">e</span> <span class="n">CELL</span><span class="p">,</span> <span class="o">--</span><span class="n">cell</span><span class="o">=</span><span class="n">CELL</span>
		<span class="n">cell</span> <span class="n">name</span>

	<span class="o">-</span><span class="n">t</span> <span class="n">SEURATCLUSTER</span><span class="p">,</span> <span class="o">--</span><span class="n">seuratcluster</span><span class="o">=</span><span class="n">SEURATCLUSTER</span>
		<span class="n">use</span> <span class="n">the</span> <span class="n">default</span> <span class="n">seurat</span> <span class="n">clusters</span>

	<span class="o">-</span><span class="n">u</span> <span class="n">CLUSTER</span><span class="p">,</span> <span class="o">--</span><span class="n">cluster</span><span class="o">=</span><span class="n">CLUSTER</span>
		<span class="n">seurat</span> <span class="n">clusters</span> <span class="n">number</span>

	<span class="o">-</span><span class="n">o</span> <span class="n">OUTDIR</span><span class="p">,</span> <span class="o">--</span><span class="n">outdir</span><span class="o">=</span><span class="n">OUTDIR</span>
		<span class="n">output</span> <span class="n">folder</span> <span class="n">name</span>

	<span class="o">-</span><span class="n">d</span> <span class="n">REDUCTION</span><span class="p">,</span> <span class="o">--</span><span class="n">reduction</span><span class="o">=</span><span class="n">REDUCTION</span>
		<span class="n">reduction</span> <span class="n">A</span> <span class="n">dimensionality</span> <span class="n">reduction</span> <span class="n">stored</span> <span class="n">in</span> <span class="n">the</span> <span class="n">Seurat</span> <span class="n">object.</span> <span class="n">Default</span> <span class="n">umap</span>

	<span class="o">-</span><span class="n">m</span> <span class="n">MODE</span><span class="p">,</span> <span class="o">--</span><span class="n">mode</span><span class="o">=</span><span class="n">MODE</span>
		<span class="n">determines</span> <span class="n">how</span> <span class="n">to</span> <span class="n">make</span> <span class="n">gene</span> <span class="n">expression</span> <span class="n">profiles</span> <span class="n">for</span> <span class="n">metaspots</span> <span class="n">from</span> <span class="n">their</span> <span class="n">constituent</span> <span class="n">single</span> <span class="n">cells.</span> <span class="n">Options</span> <span class="n">are</span> <span class="s">&#34;average&#34;</span> <span class="n">or</span> <span class="s">&#34;sum&#34;</span><span class="n">.
</span><span class="n"></span>
	<span class="o">-</span><span class="n">i</span> <span class="n">MIN_SPOTS</span><span class="p">,</span> <span class="o">--</span><span class="n">min_spots</span><span class="o">=</span><span class="n">MIN_SPOTS</span>
		<span class="n">the</span> <span class="n">minimum</span> <span class="n">number</span> <span class="n">of</span> <span class="n">spots</span> <span class="n">in</span> <span class="n">a</span> <span class="n">particular</span> <span class="n">grouping</span> <span class="n">to</span> <span class="n">construct</span> <span class="n">metaspots</span>

	<span class="o">-</span><span class="n">y</span> <span class="n">METACELLS</span><span class="p">,</span> <span class="o">--</span><span class="n">metacells</span><span class="o">=</span><span class="n">METACELLS</span>
		<span class="n">use</span> <span class="n">metacell</span> <span class="n">or</span> <span class="n">not</span>

	<span class="o">-</span><span class="n">v</span> <span class="n">SOFT</span><span class="p">,</span> <span class="o">--</span><span class="n">soft</span><span class="o">=</span><span class="n">SOFT</span>
		<span class="n">softpower</span> <span class="n">VALUE</span>

	<span class="o">-</span><span class="n">x</span> <span class="n">MAXBLOCKSIZE</span><span class="p">,</span> <span class="o">--</span><span class="n">maxBlockSize</span><span class="o">=</span><span class="n">MAXBLOCKSIZE</span>
		<span class="n">integer</span> <span class="n">giving</span> <span class="n">maximum</span> <span class="n">block</span> <span class="n">size</span> <span class="n">for</span> <span class="n">module</span> <span class="n">detection</span> 

	<span class="o">-</span><span class="bp">T</span> <span class="n">CORTYPE</span><span class="p">,</span> <span class="o">--</span><span class="n">corType</span><span class="o">=</span><span class="n">CORTYPE</span>
		<span class="n">pearson</span> <span class="n">and</span> <span class="n">bicor</span>  <span class="n">corresponding</span> <span class="n">to</span> <span class="n">Pearson</span> <span class="n">and</span> <span class="n">bidweight</span> <span class="n">midcorrelation</span>  <span class="n">respectively</span>

	<span class="o">-</span><span class="n">N</span> <span class="n">NETWORKTYPE</span><span class="p">,</span> <span class="o">--</span><span class="n">networkType</span><span class="o">=</span><span class="n">NETWORKTYPE</span>
		<span class="n">network</span> <span class="n">type</span> <span class="n">Allowed</span> <span class="n">values</span> <span class="nf">are </span><span class="p">(</span><span class="n">unique</span> <span class="n">abbreviations</span> <span class="n">of</span><span class="p">)</span> <span class="n">unsigned</span> <span class="n">signed</span> <span class="n">signed</span> <span class="n">hybrid</span>

	<span class="o">-</span><span class="n">M</span> <span class="n">TOMTYPE</span><span class="p">,</span> <span class="o">--</span><span class="n">TOMType</span><span class="o">=</span><span class="n">TOMTYPE</span>
		<span class="n">Allowed</span> <span class="n">values</span> <span class="nf">are </span><span class="p">(</span><span class="n">unique</span> <span class="n">abbreviations</span> <span class="n">of</span><span class="p">)</span>  <span class="n">unsigned</span> <span class="n">signed</span> <span class="n">signed</span> <span class="n">hybrid</span>

	<span class="o">-</span><span class="n">U</span> <span class="n">MINMODULESIZE</span><span class="p">,</span> <span class="o">--</span><span class="n">minModuleSize</span><span class="o">=</span><span class="n">MINMODULESIZE</span>
		<span class="n">minimum</span> <span class="n">module</span> <span class="n">size</span> <span class="n">for</span> <span class="n">module</span> <span class="n">detection</span>

	<span class="o">-</span><span class="n">I</span> <span class="n">MERGECUTHEIGHT</span><span class="p">,</span> <span class="o">--</span><span class="n">mergeCutHeight</span><span class="o">=</span><span class="n">MERGECUTHEIGHT</span>
		<span class="n">dendrogram</span> <span class="n">cut</span> <span class="n">height</span> <span class="n">for</span> <span class="n">module</span> <span class="n">merging</span>

	<span class="o">-</span><span class="n">L</span> <span class="n">DETECTCUTHEIGHT</span><span class="p">,</span> <span class="o">--</span><span class="n">detectCutHeight</span><span class="o">=</span><span class="n">DETECTCUTHEIGHT</span>
		<span class="n">dendrogram</span> <span class="n">cut</span> <span class="n">height</span> <span class="n">for</span> <span class="n">module</span> <span class="n">detection</span>

	<span class="o">-</span><span class="n">V</span> <span class="n">GBV</span><span class="p">,</span> <span class="o">--</span><span class="n">gbv</span><span class="o">=</span><span class="n">GBV</span>
		<span class="n">chose</span> <span class="n">the</span> <span class="nf">hME</span><span class="p">(</span><span class="n">default</span><span class="p">)</span> <span class="n">or</span> <span class="n">ME</span>

	<span class="o">-</span><span class="n">S</span> <span class="n">SCALE.MODEL.USE</span><span class="p">,</span> <span class="o">--</span><span class="n">scale.model.use</span><span class="o">=</span><span class="n">SCALE.MODEL.USE</span>
		 <span class="n">linear</span> <span class="n">poisson</span> <span class="n">or</span>  <span class="n">negbinom</span> 

	<span class="o">-</span><span class="bp">F</span> <span class="n">CORFNC</span><span class="p">,</span> <span class="o">--</span><span class="n">corFnc</span><span class="o">=</span><span class="n">CORFNC</span>
		<span class="n">character</span> <span class="n">string</span> <span class="n">specifying</span> <span class="n">the</span> <span class="n">function</span> <span class="n">to</span> <span class="n">be</span> <span class="n">used</span> <span class="n">to</span> <span class="n">calculate</span> <span class="n">co</span><span class="o">-</span><span class="n">expression</span> <span class="n">similarity</span>  <span class="n">Defaults</span> <span class="n">to</span> <span class="n">Pearson</span> <span class="n">correlation</span>

	<span class="o">-</span><span class="n">C</span> <span class="n">COROPTIONS</span><span class="p">,</span> <span class="o">--</span><span class="n">corOptions</span><span class="o">=</span><span class="n">COROPTIONS</span>
		<span class="n">Use</span> <span class="n">use</span> <span class="o">=</span>  <span class="n">p</span>   <span class="n">method</span> <span class="o">=</span>  <span class="n">spearman</span>  <span class="n">to</span> <span class="n">obtain</span> <span class="n">Spearman</span> <span class="n">correlation</span> 

	<span class="o">-</span><span class="n">E</span> <span class="n">SCORES</span><span class="p">,</span> <span class="o">--</span><span class="n">scores</span><span class="o">=</span><span class="n">SCORES</span>
		<span class="n">Compute</span> <span class="n">hub</span> <span class="n">gene</span> <span class="n">signature</span> <span class="n">scores</span>

	<span class="o">-</span><span class="n">D</span> <span class="n">METHOD</span><span class="p">,</span> <span class="o">--</span><span class="n">method</span><span class="o">=</span><span class="n">METHOD</span>
		<span class="n">compute</span> <span class="n">gene</span> <span class="n">scores</span> <span class="n">for</span> <span class="n">a</span> <span class="n">give</span> <span class="n">number</span> <span class="n">of</span> <span class="n">genes</span> <span class="n">for</span> <span class="n">each</span> <span class="n">module</span> <span class="o">||</span> <span class="n">using</span> <span class="n">either</span> <span class="n">the</span> <span class="n">Seurat</span> <span class="n">or</span> <span class="n">UCell</span> <span class="n">algorithm</span>

	<span class="o">-</span><span class="n">W</span> <span class="n">NETWORK</span><span class="p">,</span> <span class="o">--</span><span class="n">network</span><span class="o">=</span><span class="n">NETWORK</span>
		<span class="n">Network</span> <span class="n">Visualization</span> <span class="n">OR</span> <span class="n">NOT</span> 

	<span class="o">-</span><span class="n">w</span> <span class="n">TRAIT</span><span class="p">,</span> <span class="o">--</span><span class="n">trait</span><span class="o">=</span><span class="n">TRAIT</span>
		<span class="n">input</span> <span class="n">trait.txt</span> <span class="n">file</span>

	<span class="o">-</span><span class="n">X</span> <span class="n">TRT</span><span class="p">,</span> <span class="o">--</span><span class="n">trt</span><span class="o">=</span><span class="n">TRT</span>
		<span class="n">Module</span> <span class="n">Trait</span> <span class="n">Correlation</span> <span class="n">or</span> <span class="n">not</span>

	<span class="o">-</span><span class="n">G</span> <span class="n">DME</span><span class="p">,</span> <span class="o">--</span><span class="n">dme</span><span class="o">=</span><span class="n">DME</span>
		<span class="n">Differential</span> <span class="n">module</span> <span class="nf">eigengene </span><span class="p">(</span><span class="n">DME</span><span class="p">)</span> <span class="n">analysis</span>

	<span class="o">-</span><span class="n">R</span> <span class="n">CHARACTER</span><span class="p">,</span> <span class="o">--</span><span class="n">group</span><span class="o">=</span><span class="n">CHARACTER</span>
		<span class="n">sample</span> <span class="n">group</span> <span class="n">information</span> <span class="n">file</span> <span class="n">name</span> <span class="n">with</span> <span class="n">header</span><span class="o">:</span> <span class="n">sample</span><span class="o">&lt;</span><span class="n">tab</span><span class="o">&gt;</span><span class="n">group</span>

	<span class="o">-</span><span class="n">A</span> <span class="n">CHARACTER</span><span class="p">,</span> <span class="o">--</span><span class="n">compare</span><span class="o">=</span><span class="n">CHARACTER</span>
		<span class="n">group</span> <span class="n">comparison</span> <span class="n">information</span> <span class="n">file</span> <span class="n">name</span> <span class="n">with</span> <span class="n">header</span><span class="o">:</span> <span class="n">control</span><span class="o">&lt;</span><span class="n">tab</span><span class="o">&gt;</span><span class="n">case</span>

	<span class="o">-</span><span class="n">h</span><span class="p">,</span> <span class="o">--</span><span class="n">help</span>
		<span class="n">Show</span> <span class="n">this</span> <span class="n">help</span> <span class="n">message</span> <span class="n">and</span> <span class="n">exit</span>
</code></pre></td></tr></table>
</div>
</div><p>二、分析模式（当前是按照单细胞的模型来进行）</p>
<p>当前数据分析的选择（只有以下两种，无subcluster（这里也就是cell Type 的小分类，如果分析结果文件中有对应的注释信息，也直接根据cell type来选就行，我觉得这个不用加都可以））当前只针对一个样本的，数据进行分析，如果是多个样本，最好还是直接提供rds合并好的数据进行分析，如果提供多个样本，分析时也需要按照样本分析的话，需要更新脚本。</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="wps1.jpg"
        data-srcset="/posts/sthdwgcna-pipeline/wps1.jpg, wps1.jpg 1.5x, /posts/sthdwgcna-pipeline/wps1.jpg 2x"
        data-sizes="auto"
        alt="/posts/sthdwgcna-pipeline/wps1.jpg"
        title="img" /></p>
<p>1，分析所有的细胞类型（进行常规分析，默认用hme的方法进行）：</p>
<p>数据路径：/mnt/ilustre/users/meng.luo/project/RNA/scRNA/hdWGCNA/spatial/brain</p>
<p>Rscript4.2 /mnt/ilustre/users/meng.luo/project/RNA/scRNA/hdWGCNA/spatial/hdWGCNA_ST_pipeline.R &ndash;rds mouse_brain_ST_anterior.rds &ndash;cellfile cells.csv</p>
<p>2, 用me的方法构建：</p>
<p>数据路径：/mnt/ilustre/users/meng.luo/project/RNA/scRNA/hdWGCNA/spatial/brain</p>
<p>Rscript4.2 /mnt/ilustre/users/meng.luo/project/RNA/scRNA/hdWGCNA/spatial/hdWGCNA_ST_pipeline.R &ndash;rds mouse_brain_ST_anterior.rds &ndash;cellfile cells.csv &ndash;gbv F</p>
<p>3, 用me的方法构建,只做某一类细胞，比如“Caudoputamen”：</p>
<p>数据路径：/mnt/ilustre/users/meng.luo/project/RNA/scRNA/hdWGCNA/spatial/brain</p>
<p>Rscript4.2 /mnt/ilustre/users/meng.luo/project/RNA/scRNA/hdWGCNA/spatial/hdWGCNA_ST_pipeline.R &ndash;rds mouse_brain_ST_anterior.rds &ndash;cellfile cells.csv &ndash;gbv F &ndash;cell Caudoputamen</p>
<p>4, 使用seurat的分类结果（默认也是“hME”），全部一起分析：</p>
<p>数据路径：</p>
<p>/mnt/ilustre/users/meng.luo/project/RNA/scRNA/hdWGCNA/spatial/DD073R_A1_processed</p>
<p>Rscript4.2 /mnt/ilustre/users/meng.luo/project/RNA/scRNA/hdWGCNA/spatial/hdWGCNA_ST_pipeline.R &ndash;outsPath outs/ &ndash;seuratcluster T &ndash;predata T</p>
<p>5, 使用seurat的分类结果,某一类的结果比如，“0”（“ME”）：</p>
<p>Rscript4.2 /mnt/ilustre/users/meng.luo/project/RNA/scRNA/hdWGCNA/hdWGCNA_SC_pipeline.R &ndash;rds Merged_Seurat.rds &ndash;gbv F &ndash;seuratcluster T &ndash;cluster 0 &ndash;predata T</p>
<p>数据预处理，以下是默认参数，可以自己调整（参考单细胞）</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="wps2.jpg"
        data-srcset="/posts/sthdwgcna-pipeline/wps2.jpg, wps2.jpg 1.5x, /posts/sthdwgcna-pipeline/wps2.jpg 2x"
        data-sizes="auto"
        alt="/posts/sthdwgcna-pipeline/wps2.jpg"
        title="img" /></p>
<p>模块构建，以下是默认参数，可以自己调整</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="wps3.jpg"
        data-srcset="/posts/sthdwgcna-pipeline/wps3.jpg, wps3.jpg 1.5x, /posts/sthdwgcna-pipeline/wps3.jpg 2x"
        data-sizes="auto"
        alt="/posts/sthdwgcna-pipeline/wps3.jpg"
        title="img" /></p>
<p>如果不给表型，默认做RNA的基因数及count值的关联</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="wps4.jpg"
        data-srcset="/posts/sthdwgcna-pipeline/wps4.jpg, wps4.jpg 1.5x, /posts/sthdwgcna-pipeline/wps4.jpg 2x"
        data-sizes="auto"
        alt="/posts/sthdwgcna-pipeline/wps4.jpg"
        title="img" /></p>
<p>Hub基因可视化，这个结果都有，只不过需要看，你前期运行的是当个细胞或者是多个细胞的结果，数据体层需要提前</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="wps5.jpg"
        data-srcset="/posts/sthdwgcna-pipeline/wps5.jpg, wps5.jpg 1.5x, /posts/sthdwgcna-pipeline/wps5.jpg 2x"
        data-sizes="auto"
        alt="/posts/sthdwgcna-pipeline/wps5.jpg"
        title="img" /></p>
<p>模块差异分析，指的是每个细胞中的模块差异，如果不提供分组信息，则默认按照所有的样本来进行比较分析（不同细胞类型），需要看选择的细胞类型，</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="wps6.jpg"
        data-srcset="/posts/sthdwgcna-pipeline/wps6.jpg, wps6.jpg 1.5x, /posts/sthdwgcna-pipeline/wps6.jpg 2x"
        data-sizes="auto"
        alt="/posts/sthdwgcna-pipeline/wps6.jpg"
        title="img" /></p>
<p>三、分析流程分析的一些详情</p>
<p>1, 数据情况</p>
<p>不管几个样本都可以做，几个样本对应的就是spot，还有就是针对的是样本来做WGCNA，使用 Metaspot（也和metacell是一样的，但是每个spot可能包含1-10个cell）。可以使用不同样本之间进行分析（单个样本进行），或者就是合并数据之后得数据得分析结果比如聚类分析结果或细胞类型结果（不管是几个样本，都是一起分析的）。结果不好的项目的情况，获得的细胞注释的结果很少，比如一个细胞注释的结果是不到50个spot，那都是直接剔除的，但是我们参数可以设置。这个的分析思路和metacell一样，使用metacell最小是10个（最大的share 100，小于200以下可以考虑不做，做完的metacell可能没有结果，如果少于10个就不用进行分析了(不建议用SCT的方法，这个构建模型的时候获得的power很难达到0.8)。</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="wps7.png"
        data-srcset="/posts/sthdwgcna-pipeline/wps7.png, wps7.png 1.5x, /posts/sthdwgcna-pipeline/wps7.png 2x"
        data-sizes="auto"
        alt="/posts/sthdwgcna-pipeline/wps7.png"
        title="img" /><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="wps8.png"
        data-srcset="/posts/sthdwgcna-pipeline/wps8.png, wps8.png 1.5x, /posts/sthdwgcna-pipeline/wps8.png 2x"
        data-sizes="auto"
        alt="/posts/sthdwgcna-pipeline/wps8.png"
        title="img" /><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="wps9.png"
        data-srcset="/posts/sthdwgcna-pipeline/wps9.png, wps9.png 1.5x, /posts/sthdwgcna-pipeline/wps9.png 2x"
        data-sizes="auto"
        alt="/posts/sthdwgcna-pipeline/wps9.png"
        title="img" /><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="wps10.jpg"
        data-srcset="/posts/sthdwgcna-pipeline/wps10.jpg, wps10.jpg 1.5x, /posts/sthdwgcna-pipeline/wps10.jpg 2x"
        data-sizes="auto"
        alt="/posts/sthdwgcna-pipeline/wps10.jpg"
        title="img" /></p>
<p>2, soft-power</p>
<p>根据文献中的要求，默认选择的阈值是R2只超过0.8以上的第一个值作为软阈值。</p>
<p>3, Construct co-expression network</p>
<p>当前的分析都是按照，单个细胞进行共表达网络构建，这里的参数可以考虑如下，默认的参数都是按照文章中的一些参数，所以这些也是可以调整，其中明显影响的参数是merge cut hgiht.</p>
<p><img
        class="lazyload"
        src="/svg/loading.min.svg"
        data-src="wps11.jpg"
        data-srcset="/posts/sthdwgcna-pipeline/wps11.jpg, wps11.jpg 1.5x, /posts/sthdwgcna-pipeline/wps11.jpg 2x"
        data-sizes="auto"
        alt="/posts/sthdwgcna-pipeline/wps11.jpg"
        title="img" /></p>
<p>4, Module Eigengenes and Connectivity</p>
<p>模块的特征基因和连通性分析注意分两种情况，模块相关的算法：hme和me（目前这两种方法结果会有出入，最好的选择是hme，使用harmany的包进行矫正后的数据进行分析，结果获得的R2值比me的会更高），根据hme或者me的结果，我们可以获得hub基因，这里提供了两种计算方法，来计算hub基因的得分。这里也可以获得模块间的相关。（针对空间转录的结果，harmany很难收敛，计算的时间会很长）</p>
<p>5, Module Trait Correlation</p>
<p>如果不提供性状，默认分析'&lsquo;nCount_Spatial&rsquo;和 &lsquo;nFeature_Spatial&quot;。</p>
<p>6, Differential module eigengene (DME) analysis</p>
<p>差异模块分析，可以根据不同样本（比如，control和condition），此外如果不提供比较差异的分组信息，默认做所有的样本进行分析。</p>
<h4 id="learning-video">learning Video</h4>
<iframe src="//player.bilibili.com/player.html?aid=491204967&bvid=BV1SN411H7Yq&cid=1272089296&p=1" height="600" width="800"scrolling="no" border="0" frameborder="no" framespacing="0" allowfullscreen="true"> </iframe>
<h4 id="references">References</h4>
<blockquote>
<p>Gusev, A., Ko, A., Shi, H. <em>et al.</em> Integrative approaches for large-scale transcriptome-wide association studies. <em>Nat Genet</em> <strong>48</strong>, 245–252 (2016). <a href="https://doi.org/10.1038/ng.3506">https://doi.org/10.1038/ng.3506</a></p>
</blockquote>
</div><div class="post-footer" id="post-footer">
    <div class="post-info">
        <div class="post-info-line">
            <div class="post-info-mod">
                <span>更新于 2023-06-19</span>
            </div>
            <div class="post-info-license"></div>
        </div>
        <div class="post-info-line">
            <div class="post-info-md"><span>
                            <a class="link-to-markdown" href="/posts/sthdwgcna-pipeline/index.md" target="_blank">阅读原始文档</a>
                        </span></div>
            <div class="post-info-share">
                <span><a href="javascript:void(0);" title="分享到 Twitter" data-sharer="twitter" data-url="http://cb-majorbiolab.work/posts/sthdwgcna-pipeline/" data-title="The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data" data-via="https://twitter.com/czheluo" data-hashtags="SC,ST,workflow,SCRNAseq"><i class="fab fa-twitter fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Facebook" data-sharer="facebook" data-url="http://cb-majorbiolab.work/posts/sthdwgcna-pipeline/" data-hashtag="SC"><i class="fab fa-facebook-square fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Linkedin" data-sharer="linkedin" data-url="http://cb-majorbiolab.work/posts/sthdwgcna-pipeline/"><i class="fab fa-linkedin fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Hacker News" data-sharer="hackernews" data-url="http://cb-majorbiolab.work/posts/sthdwgcna-pipeline/" data-title="The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data"><i class="fab fa-hacker-news fa-fw"></i></a><a href="javascript:void(0);" title="分享到 Line" data-sharer="line" data-url="http://cb-majorbiolab.work/posts/sthdwgcna-pipeline/" data-title="The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data"><i data-svg-src="/lib/simple-icons/icons/line.min.svg"></i></a><a href="javascript:void(0);" title="分享到 微博" data-sharer="weibo" data-url="http://cb-majorbiolab.work/posts/sthdwgcna-pipeline/" data-title="The workflow of hdWGCNA analysis for Single-cell Spatial Transcriptomics data" data-ralateuid="https://weibo.com/2161302013/profile"><i class="fab fa-weibo fa-fw"></i></a></span>
            </div>
        </div>
    </div>

    <div class="post-info-more">
        <section class="post-tags"><i class="fas fa-tags fa-fw"></i>&nbsp;<a href="/tags/sc/">SC</a>,&nbsp;<a href="/tags/st/">ST</a>,&nbsp;<a href="/tags/workflow/">workflow</a>,&nbsp;<a href="/tags/scrnaseq/">SCRNAseq</a></section>
        <section>
            <span><a href="javascript:void(0);" onclick="window.history.back();">返回</a></span>&nbsp;|&nbsp;<span><a href="/">主页</a></span>
        </section>
    </div>

    <div class="post-nav"><a href="/posts/twas-pipeline/" class="prev" rel="prev" title="Integrative approaches for large-scale transcriptome-wide association studies"><i class="fas fa-angle-left fa-fw"></i>Integrative approaches for large-scale transcriptome-wide association studies</a></div>
</div>
<div id="comments"><div id="valine" class="comment"></div><noscript>
                Please enable JavaScript to view the comments powered by <a href="https://valine.js.org/">Valine</a>.
            </noscript><div id="fb-root" class="comment"></div>
            <div
                class="fb-comments"
                data-href="http://cb-majorbiolab.work/posts/sthdwgcna-pipeline/"
                data-width="100%"
                data-numposts="10"
            ></div><noscript>
                Please enable JavaScript to view the comments powered by <a href="https://developers.facebook.com/docs/plugins/comments/"></a>Facebook</a>.
            </noscript></div></article></div>
            </main><footer class="footer">
        <div class="footer-container"><div class="footer-line">MAJOR Computational Biology Research Lab</div>
                <div class="footer-line"><i class="far fa-copyright fa-fw"></i><span itemprop="copyrightYear">2020-2022, MENG LUO, All rights reserved.</span><span class="author" itemprop="copyrightHolder">&nbsp;<a href="/" target="_blank"></a></span></div>
                
        </div>
    </footer></div>

        <div id="fixed-buttons"><a href="#" id="back-to-top" class="fixed-button" title="回到顶部">
                <i class="fas fa-arrow-up fa-fw"></i>
            </a><a href="#" id="view-comments" class="fixed-button" title="查看评论">
                <i class="fas fa-comment fa-fw"></i>
            </a>
        </div><link rel="stylesheet" href="/lib/valine/valine.min.css"><link rel="stylesheet" href="/lib/lightgallery/lightgallery.min.css"><link rel="stylesheet" href="/lib/katex/katex.min.css"><link rel="stylesheet" href="/lib/katex/copy-tex.min.css"><script type="text/javascript" src="/lib/valine/Valine.min.js"></script><script type="text/javascript" src="https://connect.facebook.net/zh_CN/sdk.js#xfbml=1&amp;version=v5.0&amp;appId=&amp;autoLogAppEvents=1" defer></script><script type="text/javascript" src="/lib/smooth-scroll/smooth-scroll.min.js"></script><script type="text/javascript" src="/lib/autocomplete/autocomplete.min.js"></script><script type="text/javascript" src="/lib/algoliasearch/algoliasearch-lite.umd.min.js"></script><script type="text/javascript" src="/lib/lazysizes/lazysizes.min.js"></script><script type="text/javascript" src="/lib/lightgallery/lightgallery.min.js"></script><script type="text/javascript" src="/lib/lightgallery/lg-thumbnail.min.js"></script><script type="text/javascript" src="/lib/lightgallery/lg-zoom.min.js"></script><script type="text/javascript" src="/lib/clipboard/clipboard.min.js"></script><script type="text/javascript" src="/lib/sharer/sharer.min.js"></script><script type="text/javascript" src="/lib/katex/katex.min.js"></script><script type="text/javascript" src="/lib/katex/auto-render.min.js"></script><script type="text/javascript" src="/lib/katex/copy-tex.min.js"></script><script type="text/javascript" src="/lib/katex/mhchem.min.js"></script><script type="text/javascript">window.config={"code":{"copyTitle":"复制到剪贴板","maxShownLines":10},"comment":{"valine":{"appId":"Noyy9UKaXGwYBIUdvBTt7u76-gzGzoHsz","appKey":"15w6pTF8mFoxnm8muOFx5buM","avatar":"mp","el":"#valine","emojiCDN":"https://cdn.jsdelivr.net/npm/emoji-datasource-google@5.0.1/img/google/64/","emojiMaps":{"100":"1f4af.png","alien":"1f47d.png","anger":"1f4a2.png","angry":"1f620.png","anguished":"1f627.png","astonished":"1f632.png","black_heart":"1f5a4.png","blue_heart":"1f499.png","blush":"1f60a.png","bomb":"1f4a3.png","boom":"1f4a5.png","broken_heart":"1f494.png","brown_heart":"1f90e.png","clown_face":"1f921.png","cold_face":"1f976.png","cold_sweat":"1f630.png","confounded":"1f616.png","confused":"1f615.png","cry":"1f622.png","crying_cat_face":"1f63f.png","cupid":"1f498.png","dash":"1f4a8.png","disappointed":"1f61e.png","disappointed_relieved":"1f625.png","dizzy":"1f4ab.png","dizzy_face":"1f635.png","drooling_face":"1f924.png","exploding_head":"1f92f.png","expressionless":"1f611.png","face_vomiting":"1f92e.png","face_with_cowboy_hat":"1f920.png","face_with_hand_over_mouth":"1f92d.png","face_with_head_bandage":"1f915.png","face_with_monocle":"1f9d0.png","face_with_raised_eyebrow":"1f928.png","face_with_rolling_eyes":"1f644.png","face_with_symbols_on_mouth":"1f92c.png","face_with_thermometer":"1f912.png","fearful":"1f628.png","flushed":"1f633.png","frowning":"1f626.png","ghost":"1f47b.png","gift_heart":"1f49d.png","green_heart":"1f49a.png","grimacing":"1f62c.png","grin":"1f601.png","grinning":"1f600.png","hankey":"1f4a9.png","hear_no_evil":"1f649.png","heart":"2764-fe0f.png","heart_decoration":"1f49f.png","heart_eyes":"1f60d.png","heart_eyes_cat":"1f63b.png","heartbeat":"1f493.png","heartpulse":"1f497.png","heavy_heart_exclamation_mark_ornament":"2763-fe0f.png","hole":"1f573-fe0f.png","hot_face":"1f975.png","hugging_face":"1f917.png","hushed":"1f62f.png","imp":"1f47f.png","innocent":"1f607.png","japanese_goblin":"1f47a.png","japanese_ogre":"1f479.png","joy":"1f602.png","joy_cat":"1f639.png","kiss":"1f48b.png","kissing":"1f617.png","kissing_cat":"1f63d.png","kissing_closed_eyes":"1f61a.png","kissing_heart":"1f618.png","kissing_smiling_eyes":"1f619.png","laughing":"1f606.png","left_speech_bubble":"1f5e8-fe0f.png","love_letter":"1f48c.png","lying_face":"1f925.png","mask":"1f637.png","money_mouth_face":"1f911.png","nauseated_face":"1f922.png","nerd_face":"1f913.png","neutral_face":"1f610.png","no_mouth":"1f636.png","open_mouth":"1f62e.png","orange_heart":"1f9e1.png","partying_face":"1f973.png","pensive":"1f614.png","persevere":"1f623.png","pleading_face":"1f97a.png","pouting_cat":"1f63e.png","purple_heart":"1f49c.png","rage":"1f621.png","relaxed":"263a-fe0f.png","relieved":"1f60c.png","revolving_hearts":"1f49e.png","right_anger_bubble":"1f5ef-fe0f.png","robot_face":"1f916.png","rolling_on_the_floor_laughing":"1f923.png","scream":"1f631.png","scream_cat":"1f640.png","see_no_evil":"1f648.png","shushing_face":"1f92b.png","skull":"1f480.png","skull_and_crossbones":"2620-fe0f.png","sleeping":"1f634.png","sleepy":"1f62a.png","slightly_frowning_face":"1f641.png","slightly_smiling_face":"1f642.png","smile":"1f604.png","smile_cat":"1f638.png","smiley":"1f603.png","smiley_cat":"1f63a.png","smiling_face_with_3_hearts":"1f970.png","smiling_imp":"1f608.png","smirk":"1f60f.png","smirk_cat":"1f63c.png","sneezing_face":"1f927.png","sob":"1f62d.png","space_invader":"1f47e.png","sparkling_heart":"1f496.png","speak_no_evil":"1f64a.png","speech_balloon":"1f4ac.png","star-struck":"1f929.png","stuck_out_tongue":"1f61b.png","stuck_out_tongue_closed_eyes":"1f61d.png","stuck_out_tongue_winking_eye":"1f61c.png","sunglasses":"1f60e.png","sweat":"1f613.png","sweat_drops":"1f4a6.png","sweat_smile":"1f605.png","thinking_face":"1f914.png","thought_balloon":"1f4ad.png","tired_face":"1f62b.png","triumph":"1f624.png","two_hearts":"1f495.png","unamused":"1f612.png","upside_down_face":"1f643.png","weary":"1f629.png","white_frowning_face":"2639-fe0f.png","white_heart":"1f90d.png","wink":"1f609.png","woozy_face":"1f974.png","worried":"1f61f.png","yawning_face":"1f971.png","yellow_heart":"1f49b.png","yum":"1f60b.png","zany_face":"1f92a.png","zipper_mouth_face":"1f910.png","zzz":"1f4a4.png"},"enableQQ":false,"highlight":true,"lang":"zh-cn","pageSize":10,"placeholder":"你的评论 ...","recordIP":true,"visitor":true}},"lightGallery":{"actualSize":false,"exThumbImage":"data-thumbnail","hideBarsDelay":2000,"selector":".lightgallery","speed":400,"thumbContHeight":80,"thumbWidth":80,"thumbnail":true},"math":{"delimiters":[{"display":true,"left":"$$","right":"$$"},{"display":true,"left":"\\[","right":"\\]"},{"display":false,"left":"$","right":"$"},{"display":false,"left":"\\(","right":"\\)"}],"strict":false},"search":{"algoliaAppID":"VX4BXOJBC5","algoliaIndex":"majorbio.zh-cn","algoliaSearchKey":"53790f167edb548b217bfe91dd891ac0","highlightTag":"em","maxResultLength":10,"noResultsFound":"没有找到结果","snippetLength":50,"type":"algolia"}};</script><script type="text/javascript" src="/js/theme.min.js"></script></body>
</html>
